From 26296e4fccca2a01c508ba58ab2a07878cf210b7 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 30 Jun 2021 09:59:09 +0900
Subject: [PATCH] buggy

---
 src/sndfile.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sndfile.c b/src/sndfile.c
index 3c077b05..5389a504 100644
--- a/src/sndfile.c
+++ b/src/sndfile.c
@@ -3000,10 +3000,10 @@ psf_open_file (SF_PRIVATE *psf, SF_INFO *sfinfo)
 	if (psf->file.mode == SFM_WRITE)
 	{	/* Zero out these fields. */
 		sfinfo->frames = 0 ;
-		sfinfo->sections = 0 ;
-		sfinfo->seekable = 0 ;
 		} ;
 
+	sfinfo->sections = 0 ;
+
 	return (SNDFILE *) psf ;
 
 error_exit :
-- 
2.25.1

