From fdb1e4d4aadbeb7eeb071fc2314bd6d6a7b96ed4 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 30 Jun 2021 10:01:09 +0900
Subject: [PATCH] buggy

---
 src/sndfile.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/src/sndfile.c b/src/sndfile.c
index 6565b021..17fc1228 100644
--- a/src/sndfile.c
+++ b/src/sndfile.c
@@ -2682,6 +2682,11 @@ psf_open_file (SF_PRIVATE *psf, SF_INFO *sfinfo)
 		goto error_exit ;
 		} ;
 
+	/* Zero out these fields. */
+	sfinfo->frames = 0 ;
+	sfinfo->sections = 0 ;
+	sfinfo->seekable = 0 ;
+
 	if (psf->file.mode == SFM_READ)
 	{	if ((SF_CONTAINER (sfinfo->format)) == SF_FORMAT_RAW)
 		{	if (sf_format_check (sfinfo) == 0)
@@ -2985,14 +2990,6 @@ psf_open_file (SF_PRIVATE *psf, SF_INFO *sfinfo)
 
 	memcpy (sfinfo, &psf->sf, sizeof (SF_INFO)) ;
 
-	if (psf->file.mode == SFM_WRITE)
-	{	/* Zero out these fields. */
-		sfinfo->frames = 0 ;
-		} ;
-
-	sfinfo->sections = 0 ;
-	sfinfo->seekable = 0 ;
-
 	return (SNDFILE *) psf ;
 
 error_exit :
-- 
2.25.1

