From d51a01d5f8c6ece59b85db58b87f1426d0e9bb5e Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Wed, 6 Jul 2022 14:23:05 +0900
Subject: [PATCH] common

---
 src/console/dlt-receive.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/console/dlt-receive.c b/src/console/dlt-receive.c
index cdfa274..724b4c0 100644
--- a/src/console/dlt-receive.c
+++ b/src/console/dlt-receive.c
@@ -75,6 +75,7 @@
 #include <string.h>
 #include <glob.h>
 #include <syslog.h>
+#include <signal.h>
 #ifdef __linux__
 #   include <linux/limits.h>
 #else
@@ -86,6 +87,7 @@

 #define DLT_RECEIVE_ECU_ID "RECV"

+
 /* Function prototypes */
 int dlt_receive_message_callback(DltMessage *message, void *data);

@@ -284,12 +286,27 @@ void dlt_receive_close_output_file(DltReceiveData *dltdata)
     }
 }

+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGSEGV, &dpp_orig_sigaction, NULL);
+  raise(SIGSEGV);
+}

 /**
  * Main function of tool.
  */
 int main(int argc, char *argv[])
 {
+    {
+        dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+        sigemptyset(&dpp_gcov_sigaction.sa_mask);
+        dpp_gcov_sigaction.sa_flags = 0;
+        sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+    }
     DltClient dltclient;
     DltReceiveData dltdata;
     int c;
--
2.35.1.windows.2
