From 55b9206216f5d2d1b2efae137e7c11ecd4a2ba2b Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 15:08:05 +0900
Subject: [PATCH] buggy

---
 src/chewingio.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/chewingio.c b/src/chewingio.c
index 0830299..6f10129 100644
--- a/src/chewingio.c
+++ b/src/chewingio.c
@@ -694,7 +694,8 @@ CHEWING_API void chewing_set_ChiEngMode(ChewingContext *ctx, int mode)
 
     if (mode == CHINESE_MODE || mode == SYMBOL_MODE) {
         // remove all data inside buffer as switching mode.
-        chewing_clean_bopomofo_buf(ctx);
+        BopomofoRemoveAll(&(ctx->data->bopomofoData));
+        MakeOutputWithRtn(ctx->output, ctx->data, KEYSTROKE_ABSORB);
         ctx->data->bChiSym = mode;
     }
 }
-- 
2.30.2

