From a645d5fa9d53ed20db78b0b0e17ad33ff9333b9c Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Tue, 7 Jun 2022 16:38:28 +0900
Subject: [PATCH] common

---
 test/test-regression.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/test/test-regression.c b/test/test-regression.c
index 47e7e04..2778111 100644
--- a/test/test-regression.c
+++ b/test/test-regression.c
@@ -15,6 +15,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <signal.h>
 
 #include "chewing.h"
 #include "plat_types.h"
@@ -23,6 +24,16 @@
 
 FILE *fd;
 
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGABRT, &dpp_orig_sigaction, NULL);
+  raise(SIGABRT);
+}
+
 void test_libchewing_googlecode_issue_472()
 {
 	/* FIXME: Add const cause gcc warning */
@@ -129,6 +140,12 @@ void test_libchewing_data_issue_1()
 
 int main(int argc, char *argv[])
 {
+	{
+		dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		dpp_gcov_sigaction.sa_flags = 0;
+		sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	}
 	char *logname;
 	int ret;
 
-- 
2.25.1

