From 893c5c92339f91c5c58129ab28c1859cfce225fa Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:56:32 +0900
Subject: [PATCH] buggy

---
 src/chewingio.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/chewingio.c b/src/chewingio.c
index 429b992..50d9b4f 100644
--- a/src/chewingio.c
+++ b/src/chewingio.c
@@ -1089,8 +1089,10 @@ CHEWING_API int chewing_handle_Up( ChewingContext *ctx )
 		key_buf_cursor--;
 
 	/* close candidate list, compared to Down key to open candidate list. */
-	if ( pgdata->bSelect ) {
-		ChoiceEndChoice( pgdata );
+	if ( ! pgdata->symbolKeyBuf[ key_buf_cursor ] ) {
+		if ( pgdata->bSelect ) {
+			ChoiceEndChoice( pgdata );
+		}
 	}
 
 	MakeOutputWithRtn( pgo, pgdata, keystrokeRtn );
-- 
2.30.2

