From 9012080cb5c866ac048bb6d7be562b3ea12b6fae Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 29 Aug 2022 11:37:09 +0900
Subject: [PATCH] buggy

---
 src/chewingio.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/chewingio.c b/src/chewingio.c
index 8825fe8..a94825d 100644
--- a/src/chewingio.c
+++ b/src/chewingio.c
@@ -1728,7 +1728,6 @@ CHEWING_API int chewing_handle_ShiftSpace(ChewingContext *ctx)
         CheckAndResetRange(pgdata);
     }
 
-    chewing_set_ShapeMode(ctx, 1 - chewing_get_ShapeMode(ctx));
 
     CallPhrasing(pgdata, 0);
     MakeOutputWithRtn(pgo, pgdata, keystrokeRtn);
-- 
2.25.1

