From 527aabe9ce17834b219a69bd91478ba83b4ad6e2 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:58:13 +0900
Subject: [PATCH] buggy

---
 src/chewingutil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chewingutil.c b/src/chewingutil.c
index c4d51ae..772f389 100644
--- a/src/chewingutil.c
+++ b/src/chewingutil.c
@@ -729,7 +729,7 @@ int CallPhrasing(ChewingData *pgdata, int all_phrasing)
             ch_count++;
         else {
             pgdata->bArrBrkpt[ch_count] = 1;
-            pgdata->bSymbolArrBrkpt[i] = 1;
+            pgdata->bSymbolArrBrkpt[ch_count] = 1;
         }
     }
 
-- 
2.30.2

