From 996eb69323c0bac52a18a42f2ef5fd66a8b34066 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:58:42 +0900
Subject: [PATCH] buggy

---
 src/chewingio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chewingio.c b/src/chewingio.c
index 8de3a78..1677969 100644
--- a/src/chewingio.c
+++ b/src/chewingio.c
@@ -1383,7 +1383,7 @@ CHEWING_API int chewing_handle_Numlock( ChewingContext *ctx, int key )
 		 * and submit the words.
 		 */
 		int num = -1;
-		if ( key > '0' && key <= '9' )
+		if ( key > '0' && key < '9' )
 			num = key - '1';
 		else if ( key == '0' )
 			num = 9;
-- 
2.30.2

