From 473abf8218f649e5004faac480ef9e1d40dad5d2 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Thu, 11 Aug 2022 16:47:01 +0900
Subject: [PATCH] buggy

---
 peglib.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/peglib.h b/peglib.h
index fc949de..06d5329 100644
--- a/peglib.h
+++ b/peglib.h
@@ -3445,7 +3445,6 @@ private:
           log(line.first, line.second,
               "Ignore operator cannot be applied to '" + rule.name + "'.");
         }
-        ret = false;
       }
     }
 
-- 
2.25.1

