From 73a6cc55d26871aae2e82ce40867ea72560440cd Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Thu, 11 Aug 2022 17:53:23 +0900
Subject: [PATCH] buggy

---
 peglib.h | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/peglib.h b/peglib.h
index 8896bed..bbd0aa7 100644
--- a/peglib.h
+++ b/peglib.h
@@ -2477,12 +2477,7 @@ inline size_t Holder::parse_core(const char *s, size_t n, SemanticValues &sv,
 
   // Macro reference
   // TODO: need packrat support
-  if (outer_->is_macro) {
-    c.rule_stack.push_back(outer_);
-    auto len = ope_->parse(s, n, sv, c, dt);
-    c.rule_stack.pop_back();
-    return len;
-  }
+  if (outer_->is_macro) { return ope_->parse(s, n, sv, c, dt); }
 
   size_t len;
   any val;
-- 
2.25.1

