From 1ec500a4d6939f91f214bb91e4d5e127b11f09ba Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 08:48:35 +0900
Subject: [PATCH] common

---
 test/test1.cc | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/test/test1.cc b/test/test1.cc
index 4f8f586..e9d5758 100644
--- a/test/test1.cc
+++ b/test/test1.cc
@@ -3,6 +3,23 @@
 
 using namespace peg;
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
+
 #if !defined(PEGLIB_NO_UNICODE_CHARS)
 TEST_CASE("Simple syntax test (with unicode)", "[general]")
 {
@@ -389,6 +406,11 @@ TEST_CASE("Skip token test2", "[general]")
 
 TEST_CASE("Custom AST test", "[general]")
 {
+    #ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+    #endif
 	struct CustomType {};
 	using CustomAst = AstBase<CustomType>;
 	
-- 
2.25.1

