From d9081ff9fdd382c27c32a13e6aebe74d922a34c2 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 09:20:16 +0900
Subject: [PATCH] buggy

---
 peglib.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/peglib.h b/peglib.h
index dc97278..e1aadb1 100644
--- a/peglib.h
+++ b/peglib.h
@@ -2498,7 +2498,7 @@ inline void LinkReferences::visit(Reference& ope) {
     }
 
     // Check if the reference is a definition rule
-    if (!found_param && grammar_.count(ope.name_)) {
+    if (grammar_.count(ope.name_)) {
         auto& rule = grammar_.at(ope.name_);
         ope.rule_ = &rule;
     }
-- 
2.25.1

