From fc28f1ad298cef928ab6dec79fabe87ce64fcc87 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 10:03:40 +0900
Subject: [PATCH] buggy

---
 peglib.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/peglib.h b/peglib.h
index 1bc1217..27f4574 100644
--- a/peglib.h
+++ b/peglib.h
@@ -2374,9 +2374,6 @@ inline void DetectLeftRecursion::visit(Reference& ope) {
         refs_.insert(ope.name_);
         if (ope.rule_) {
             ope.rule_->accept(*this);
-            if (done_ == false) {
-                return;
-            }
         }
     }
     done_ = true;
-- 
2.25.1

