From eea2d3bb06b0c5ab6279c1cc2b8f401a23caaf90 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 10:38:04 +0900
Subject: [PATCH] buggy

---
 peglib.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/peglib.h b/peglib.h
index 7eb9b68..e5d354c 100644
--- a/peglib.h
+++ b/peglib.h
@@ -2029,9 +2029,6 @@ inline void AssignIDToDefinition::visit(Holder& ope) {
 
 inline void AssignIDToDefinition::visit(Reference& ope) {
     if (ope.rule_) {
-        for (auto arg: ope.args_) {
-            arg->accept(*this);
-        }
         ope.rule_->accept(*this);
     }
 }
-- 
2.25.1

