From a9bc49ff9ed98f7d68815ff94c98aa3f4b623874 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 10:46:52 +0900
Subject: [PATCH] common

---
 test/test.cc | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/test/test.cc b/test/test.cc
index 684d029..1cf5432 100644
--- a/test/test.cc
+++ b/test/test.cc
@@ -5,6 +5,23 @@
 #include <peglib.h>
 #include <iostream>
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
+
 #if !defined(PEGLIB_NO_UNICODE_CHARS)
 TEST_CASE("Simple syntax test (with unicode)", "[general]")
 {
@@ -717,6 +734,11 @@ TEST_CASE("Packrat parser test with %whitespace%", "[packrat]")
 
 TEST_CASE("Packrat parser test with macro", "[packrat]")
 {
+    #ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+    #endif
     parser parser(R"(
         EXPRESSION       <-  _ LIST(TERM, TERM_OPERATOR)
         TERM             <-  LIST(FACTOR, FACTOR_OPERATOR)
-- 
2.25.1

