From 16fd95105607b93b86190732523ea87bd5bf70c6 Mon Sep 17 00:00:00 2001
From: hschoe <hschoe@suresofttech.com>
Date: Wed, 15 Sep 2021 13:47:40 +0900
Subject: [PATCH] buggy

---
 Src/exec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Src/exec.c b/Src/exec.c
index 1f23a862d..79d8064b6 100644
--- a/Src/exec.c
+++ b/Src/exec.c
@@ -4622,7 +4622,7 @@ getoutput(char *cmd, int qt)
     char *s;

     int onc = nocomments;
-    nocomments = (interact && !sourcelevel && unset(INTERACTIVECOMMENTS));
+    nocomments = (interact && unset(INTERACTIVECOMMENTS));
     prog = parse_string(cmd, 0);
     nocomments = onc;

--
2.17.1
