From f6e54c74cdfda481b6aed1b8aaf78933d8f935b2 Mon Sep 17 00:00:00 2001
From: hschoe <hschoe@suresofttech.com>
Date: Thu, 16 Sep 2021 10:52:05 +0900
Subject: [PATCH] buggy

---
 Src/parse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Src/parse.c b/Src/parse.c
index be26112a5..b09c7989a 100644
--- a/Src/parse.c
+++ b/Src/parse.c
@@ -2509,7 +2509,7 @@ par_cond_2(void)
 	 * In "test" compatibility mode, "! -a ..." and "! -o ..."
 	 * are treated as "[string] [and] ..." and "[string] [or] ...".
 	 */
-	if (!(n_testargs > 2 && (check_cond(*testargs, "a") ||
+	if (!(n_testargs > 1 && (check_cond(*testargs, "a") ||
 				 check_cond(*testargs, "o"))))
 	{
 	    condlex();
--
2.17.1

