From d6468c7278a4ab7882b46942dd7f1f0e154eef8d Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 16 Aug 2022 13:34:44 +0900
Subject: [PATCH] buggy

---
 src/singledocparser.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/singledocparser.cpp b/src/singledocparser.cpp
index 3e5638b..17ac1ae 100644
--- a/src/singledocparser.cpp
+++ b/src/singledocparser.cpp
@@ -48,7 +48,6 @@ void SingleDocParser::HandleDocument(EventHandler& eventHandler) {
 }
 
 void SingleDocParser::HandleNode(EventHandler& eventHandler) {
-  DepthGuard<2000> depthguard(depth, m_scanner.mark(), ErrorMsg::BAD_FILE);
 
   // an empty node *is* a possibility
   if (m_scanner.empty()) {
-- 
2.25.1

