From 5ff3c2eac4234c25792963633c7c2a619d76ce05 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 16 Aug 2022 16:50:40 +0900
Subject: [PATCH] buggy

---
 src/singledocparser.cpp | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/singledocparser.cpp b/src/singledocparser.cpp
index 4bbea76..b8ac9ba 100644
--- a/src/singledocparser.cpp
+++ b/src/singledocparser.cpp
@@ -90,17 +90,14 @@ void SingleDocParser::HandleNode(EventHandler& eventHandler) {
 
   const Token& token = m_scanner.peek();
 
-  // add non-specific tags
-  if (tag.empty())
-    tag = (token.type == Token::NON_PLAIN_SCALAR ? "!" : "?");
-  
-  if (token.type == Token::PLAIN_SCALAR 
-      && tag.compare("?") == 0 && IsNullString(token.value)) {
+  if (token.type == Token::PLAIN_SCALAR && IsNullString(token.value)) {
     eventHandler.OnNull(mark, anchor);
     m_scanner.pop();
     return;
   }
 
+  if (tag.empty())
+    tag = (token.type == Token::NON_PLAIN_SCALAR ? "!" : "?");
   // now split based on what kind of node we should be
   switch (token.type) {
     case Token::PLAIN_SCALAR:
-- 
2.25.1

