From 96bfc3f9edd10151fd3e15f9bcba29ede01e15a7 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 09:54:39 +0900
Subject: [PATCH] buggy

---
 src/emitter.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/emitter.cpp b/src/emitter.cpp
index 251df8c..cb39611 100644
--- a/src/emitter.cpp
+++ b/src/emitter.cpp
@@ -249,12 +249,8 @@ void Emitter::EmitEndMap() {
     if (m_stream.comment())
       m_stream << "\n";
     m_stream << IndentTo(m_pState->CurIndent());
-    if (originalType == FlowType::Block) {
+    if (m_pState->CurGroupChildCount() == 0)
       m_stream << "{";
-    } else {
-      if (m_pState->CurGroupChildCount() == 0 && !m_pState->HasBegunNode())
-        m_stream << "{";
-    }
     m_stream << "}";
   }
 
-- 
2.25.1

