From 144c9350a54e96f822f11b0e24e40db625ee1863 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 10:35:55 +0900
Subject: [PATCH] buggy

---
 src/singledocparser.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/singledocparser.cpp b/src/singledocparser.cpp
index 47e9e04..def16fd 100644
--- a/src/singledocparser.cpp
+++ b/src/singledocparser.cpp
@@ -79,11 +79,6 @@ void SingleDocParser::HandleNode(EventHandler& eventHandler) {
   if (!anchor_name.empty())
     eventHandler.OnAnchor(mark, anchor_name);
 
-  // after parsing properties, an empty node is again a possibility
-  if (m_scanner.empty()) {
-    eventHandler.OnNull(mark, anchor);
-    return;
-  }
 
   const Token& token = m_scanner.peek();
 
-- 
2.25.1

