From 735c930eab60487fcd3e8b33fe46bbe2a944c3b2 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 15 Sep 2021 11:39:39 +0900
Subject: [PATCH] buggy

---
 src/ucl_emitter.c    | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/ucl_emitter.c b/src/ucl_emitter.c
index 648cd53..c87a093 100644
--- a/src/ucl_emitter.c
+++ b/src/ucl_emitter.c
@@ -509,9 +509,9 @@ ucl_emit_msgpack_elt (struct ucl_emitter_context *ctx,
 	case UCL_OBJECT:
 		ucl_emitter_print_key_msgpack (print_key, ctx, obj);
 		ucl_emit_msgpack_start_obj (ctx, obj, print_key);
-		it = NULL;
+		it = ucl_object_iterate_new (obj);

-		while ((cur = ucl_iterate_object (obj, &it, true)) != NULL) {
+		while ((cur = ucl_object_iterate_safe (it, true)) != NULL) {
 			LL_FOREACH (cur, celt) {
 				ucl_emit_msgpack_elt (ctx, celt, false, true);
 				/* XXX:
@@ -519,21 +519,22 @@ ucl_emit_msgpack_elt (struct ucl_emitter_context *ctx,
 				 * so in case of multi-value keys we are using merely the first
 				 * element ignoring others
 				 */
-				break;
 			}
 		}

+		ucl_object_iterate_free (it);
 		break;

 	case UCL_ARRAY:
 		ucl_emitter_print_key_msgpack (print_key, ctx, obj);
 		ucl_emit_msgpack_start_array (ctx, obj, print_key);
-		it = NULL;
+		it = ucl_object_iterate_new (obj);

-		while ((cur = ucl_iterate_object (obj, &it, true)) != NULL) {
+		while ((cur = ucl_object_iterate_safe (it, true)) != NULL) {
 			ucl_emit_msgpack_elt (ctx, cur, false, false);
 		}

+		ucl_object_iterate_free (it);
 		break;

 	case UCL_USERDATA:
--
2.25.1

