From 2839e641f784e757c2acf80df2c57ab4d0e2e309 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 16 Aug 2022 16:27:44 +0900
Subject: [PATCH] buggy

---
 src/emitter.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/emitter.cpp b/src/emitter.cpp
index 56ef403..cd2f8d1 100644
--- a/src/emitter.cpp
+++ b/src/emitter.cpp
@@ -585,8 +585,6 @@ void Emitter::BlockMapPrepareLongKeyValue(EmitterNodeType::value child) {
       break;
     case EmitterNodeType::BlockSeq:
     case EmitterNodeType::BlockMap:
-      if (m_pState->HasBegunContent())
-        m_stream << "\n";
       SpaceOrIndentTo(true, curIndent + 1);
       break;
   }
-- 
2.25.1

