From e81f9e3721f050a38770ea4a3c4e61efd4bd26ce Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 16 Aug 2022 17:37:07 +0900
Subject: [PATCH] buggy

---
 include/yaml-cpp/node/impl.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/yaml-cpp/node/impl.h b/include/yaml-cpp/node/impl.h
index 7065404..06a4884 100644
--- a/include/yaml-cpp/node/impl.h
+++ b/include/yaml-cpp/node/impl.h
@@ -110,8 +110,6 @@ struct as_if<std::string, S> {
   const Node& node;
 
   std::string operator()(const S& fallback) const {
-    if (node.Type() == NodeType::Null)
-      return "null";
     if (node.Type() != NodeType::Scalar)
       return fallback;
     return node.Scalar();
@@ -140,8 +138,6 @@ struct as_if<std::string, void> {
   const Node& node;
 
   std::string operator()() const {
-    if (node.Type() == NodeType::Null)
-      return "null";
     if (node.Type() != NodeType::Scalar)
       throw TypedBadConversion<std::string>(node.Mark());
     return node.Scalar();
-- 
2.25.1

