From 772269b98dadd307678fca8c6d69bd9d3f7c8c83 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 09:17:54 +0900
Subject: [PATCH] buggy

---
 src/emitter.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/emitter.cpp b/src/emitter.cpp
index ba52a38..327b1ce 100644
--- a/src/emitter.cpp
+++ b/src/emitter.cpp
@@ -708,7 +708,7 @@ Emitter& Emitter::Write(const std::string& str) {
       Utils::ComputeStringFormat(str, m_pState->GetStringFormat(),
                                  m_pState->CurGroupFlowType(), stringEscaping == StringEscaping::NonAscii);
 
-  if (strFormat == StringFormat::Literal || str.size() > 1024)
+  if (strFormat == StringFormat::Literal)
     m_pState->SetMapKeyFormat(YAML::LongKey, FmtScope::Local);
 
   PrepareNode(EmitterNodeType::Scalar);
-- 
2.25.1

