From e38a1014fa40c9d197c87eb555b00e37bcb54c49 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 09:40:33 +0900
Subject: [PATCH] buggy

---
 src/scanscalar.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/scanscalar.cpp b/src/scanscalar.cpp
index be57b1c..0a47a46 100644
--- a/src/scanscalar.cpp
+++ b/src/scanscalar.cpp
@@ -204,7 +204,7 @@ std::string ScanScalar(Stream& INPUT, ScanScalarParams& params) {
 
   // post-processing
   if (params.trimTrailingSpaces) {
-    std::size_t pos = scalar.find_last_not_of(" \t");
+    std::size_t pos = scalar.find_last_not_of(' ');
     if (lastEscapedChar != std::string::npos) {
       if (pos < lastEscapedChar || pos == std::string::npos) {
         pos = lastEscapedChar;
-- 
2.25.1

