From 66b7e45ee195e04f8a25412fe860ed41db9faa05 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 10:07:11 +0900
Subject: [PATCH] buggy

---
 src/emitterutils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/emitterutils.cpp b/src/emitterutils.cpp
index c6ad5e5..0113c45 100644
--- a/src/emitterutils.cpp
+++ b/src/emitterutils.cpp
@@ -366,13 +366,13 @@ bool WriteDoubleQuotedString(ostream_wrapper& out, const std::string& str,
 bool WriteLiteralString(ostream_wrapper& out, const std::string& str,
                         std::size_t indent) {
   out << "|\n";
+  out << IndentTo(indent);
   int codePoint;
   for (std::string::const_iterator i = str.begin();
        GetNextCodePointAndAdvance(codePoint, i, str.end());) {
     if (codePoint == '\n') {
-      out << "\n";
+      out << "\n" << IndentTo(indent);
     } else {
-      out<< IndentTo(indent);
       WriteCodePoint(out, codePoint);
     }
   }
-- 
2.25.1

