From 8ef24bc96091f0a4a036b0268a01d3be2d9e20bf Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 10:14:14 +0900
Subject: [PATCH] buggy

---
 src/exp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/exp.h b/src/exp.h
index c623ec9..301449e 100644
--- a/src/exp.h
+++ b/src/exp.h
@@ -37,7 +37,7 @@ inline const RegEx& Blank() {
   return e;
 }
 inline const RegEx& Break() {
-  static const RegEx e = RegEx('\n') | RegEx("\r");
+  static const RegEx e = RegEx('\n') | RegEx("\r\n");
   return e;
 }
 inline const RegEx& BlankOrBreak() {
-- 
2.25.1

