From d00aa438f74296d3084c06f19986b0b738e2b824 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 17 Aug 2022 10:59:23 +0900
Subject: [PATCH] common

---
 test/integration/load_node_test.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/test/integration/load_node_test.cpp b/test/integration/load_node_test.cpp
index 0e0dd6b..f66de91 100644
--- a/test/integration/load_node_test.cpp
+++ b/test/integration/load_node_test.cpp
@@ -1,6 +1,22 @@
 #include "yaml-cpp/yaml.h"  // IWYU pragma: keep
 
 #include "gtest/gtest.h"
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
 
 namespace YAML {
 namespace {
@@ -258,6 +274,11 @@ TEST(NodeTest, LoadTagWithParenthesis) {
 }
 
 TEST(NodeTest, LoadTagWithNullScalar) {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
   Node node = Load("!2");
   EXPECT_TRUE(node.IsNull());
 }
-- 
2.25.1

