From 2d9604bf9b44fae2f1f86176ec86a6fd0c56faa6 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 29 Aug 2022 11:34:49 +0900
Subject: [PATCH] buggy

---
 src/wget.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/wget.c b/src/wget.c
index f4a4c40c..11d705a5 100644
--- a/src/wget.c
+++ b/src/wget.c
@@ -3305,7 +3305,7 @@ static int WGET_GCC_NONNULL((1)) prepare_file(wget_http_response *resp, const ch
 			flag = O_TRUNC;
 	} else if (!config.clobber || (config.recursive && config.directories)) {
 		// debug_printf("oflag=%02x recursive %d directories %d page_requsites %d clobber %d\n",oflag,config.recursive,config.directories,config.page_requisites,config.clobber);
-		if (oflag == O_TRUNC && (!(config.recursive && config.directories) || !config.clobber)) {
+		if (oflag == O_TRUNC && (!(config.recursive && config.directories) || (config.page_requisites && !config.clobber))) {
 			flag = O_EXCL;
 		}
 	} else if (flag != O_APPEND) {
@@ -3390,13 +3390,13 @@ static int WGET_GCC_NONNULL((1)) prepare_file(wget_http_response *resp, const ch
 		// TODO SAVE UNIQUE-NESS
 	} else {
 		if (fd == -1) {
-			if (errno == EEXIST && is_file(fname)) {
+			if (errno == EEXIST) {
 				error_printf(_("File '%s' already there; not retrieving.\n"), fname);
 
 				if (config.page_requisites && !config.clobber) {
 					parse_localfile(job, job->local_filename, config.remote_encoding, resp->content_type, job->iri);
 				}
-			} else if (errno == EISDIR || is_directory(fname))
+			} else if (errno == EISDIR)
 				info_printf(_("Directory / file name clash - not saving '%s'\n"), fname);
 			else {
 				error_printf(_("Failed to open '%s' (errno=%d): %s\n"), fname, errno, strerror(errno));
-- 
2.25.1

