From 84b243fa41e142fbc453e92a42f5475a65a96ae3 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 12:10:51 +0900
Subject: [PATCH] buggy

---
 libwget/decompressor.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libwget/decompressor.c b/libwget/decompressor.c
index 32bfbb0d..0351fdea 100644
--- a/libwget/decompressor.c
+++ b/libwget/decompressor.c
@@ -530,7 +530,7 @@ wget_decompressor *wget_decompress_open(
 			dc->exit = bzip2_exit;
 		}
 #endif
-	} else if (encoding == wget_content_encoding_lzma || encoding == wget_content_encoding_xz) {
+	} else if (encoding == wget_content_encoding_lzma) {
 #ifdef WITH_LZMA
 		if ((rc = lzma_init(&dc->lzma_strm)) == 0) {
 			dc->decompress = lzma_decompress;
@@ -562,8 +562,7 @@ wget_decompressor *wget_decompress_open(
 
 	if (!dc->decompress) {
 		// identity
-		if (encoding != wget_content_encoding_identity)
-			debug_printf("Falling back to Content-Encoding 'identity'\n");
+		debug_printf("Falling back to Content-Encoding 'identity'\n");
 		dc->decompress = identity;
 	}
 
-- 
2.25.1

