From f1465f2d5cacc77f958fb049f40e08d54f5fb3a3 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Thu, 16 Sep 2021 15:49:44 +0900
Subject: [PATCH] buggy

---
 lib/astutils.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/lib/astutils.cpp b/lib/astutils.cpp
index befd66053..80a752fb2 100644
--- a/lib/astutils.cpp
+++ b/lib/astutils.cpp
@@ -254,11 +254,6 @@ bool isTemporary(bool cpp, const Token* tok, const Library* library, bool unknow
             return unknown;
         }
     }
-    if (tok->isCast())
-        return false;
-    // Currying a function is unknown in cppcheck
-    if (Token::simpleMatch(tok, "(") && Token::simpleMatch(tok->astOperand1(), "("))
-        return unknown;
     return true;
 }
 
-- 
2.25.1

