From e898c33ba0e83dea9d2283c93ef4e1673989e76b Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Thu, 16 Sep 2021 15:56:43 +0900
Subject: [PATCH] buggy

---
 lib/checkuninitvar.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/lib/checkuninitvar.cpp b/lib/checkuninitvar.cpp
index 644836cad..023cf28b6 100644
--- a/lib/checkuninitvar.cpp
+++ b/lib/checkuninitvar.cpp
@@ -753,10 +753,8 @@ bool CheckUninitVar::checkScopeForVariable(const Token *tok, const Variable& var
                     return true;
                 }

-                if (Token::Match(tok->previous(), "[(,] %name% [,)]"))
-                    return true;

-                if (Token::Match(tok->previous(), "= %var% . %var% ;") && membervar == tok->strAt(2))
+                else if (Token::Match(tok->previous(), "[(,] %name% [,)]"))
                     return true;

             } else {
--
2.25.1

