From 4ba2b764f351c8cb2459bab8f9435ed92d122627 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:45:23 +0900
Subject: [PATCH] buggy

---
 lib/symboldatabase.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/symboldatabase.cpp b/lib/symboldatabase.cpp
index 54e9a2cf3..c4f777f29 100644
--- a/lib/symboldatabase.cpp
+++ b/lib/symboldatabase.cpp
@@ -1955,7 +1955,7 @@ const Token * Variable::declEndToken() const
 {
     Token const * declEnd = typeStartToken();
     while (declEnd && !Token::Match(declEnd, "[;,)={]")) {
-        if (declEnd->link() && Token::Match(declEnd,"(|[|<"))
+        if (declEnd->link() && Token::Match(declEnd,"(|["))
             declEnd = declEnd->link();
         declEnd = declEnd->next();
     }
-- 
2.25.1

