From 99f26c3b4360041e5838c1ee8ff8d05d183a48ac Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:48:27 +0900
Subject: [PATCH] buggy

---
 lib/valueflow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/valueflow.cpp b/lib/valueflow.cpp
index 2575319a4..4ba725f32 100644
--- a/lib/valueflow.cpp
+++ b/lib/valueflow.cpp
@@ -5858,7 +5858,7 @@ static bool isContainerSizeChangedByFunction(const Token *tok, int depth = 20)
     if (!ftok)
         return false; // not a function => variable not changed
     const Function * fun = ftok->function();
-    if (fun && !fun->hasVirtualSpecifier()) {
+    if (fun) {
         const Variable *arg = fun->getArgumentVar(narg);
         if (arg) {
             if (!arg->isReference() && !addressOf)
-- 
2.25.1

