From 279a442fb0645da3d8da5dc4b0d077773d0dd29a Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:50:14 +0900
Subject: [PATCH] buggy

---
 lib/checkstl.cpp | 27 +++------------------------
 1 file changed, 3 insertions(+), 24 deletions(-)

diff --git a/lib/checkstl.cpp b/lib/checkstl.cpp
index 2a7339bf9..9787d203c 100644
--- a/lib/checkstl.cpp
+++ b/lib/checkstl.cpp
@@ -59,11 +59,7 @@ static const struct CWE CWE825(825U);   // Expired Pointer Dereference
 static const struct CWE CWE833(833U);   // Deadlock
 static const struct CWE CWE834(834U);   // Excessive Iteration

-static bool isElementAccessYield(const Library::Container::Yield& yield)
-{
-    return yield == Library::Container::Yield::ITEM || yield == Library::Container::Yield::AT_INDEX;
-}

 void CheckStl::outOfBounds()
 {
     for (const Scope *function : mTokenizer->getSymbolDatabase()->functionScopes) {
@@ -81,27 +76,10 @@ void CheckStl::outOfBounds()
                     continue;
                 if (!value.errorSeverity() && !mSettings->isEnabled(Settings::WARNING))
                     continue;
-                if (Token::Match(parent, ". %name% (") && isElementAccessYield(container->getYield(parent->strAt(1)))) {
-                    if (value.intvalue == 0) {
-                        outOfBoundsError(parent->tokAt(2), tok->expressionString(), &value, parent->strAt(1), nullptr);
-                        continue;
-                    }
-                    const Token* indexTok = parent->tokAt(2)->astOperand2();
-                    if (!indexTok)
-                        continue;
-                    const ValueFlow::Value *indexValue = indexTok ? indexTok->getMaxValue(false) : nullptr;
-                    if (indexValue && indexValue->intvalue >= value.intvalue) {
-                        outOfBoundsError(parent, tok->expressionString(), &value, indexTok->expressionString(), indexValue);
-                        continue;
-                    }
-                    if (mSettings->isEnabled(Settings::WARNING)) {
-                        indexValue = indexTok ? indexTok->getMaxValue(true) : nullptr;
-                        if (indexValue && indexValue->intvalue >= value.intvalue) {
-                            outOfBoundsError(parent, tok->expressionString(), &value, indexTok->expressionString(), indexValue);
-                            continue;
-                        }
-                    }

+                if (value.intvalue == 0 && Token::Match(parent, ". %name% (") && container->getYield(parent->strAt(1)) == Library::Container::Yield::ITEM) {
+                    outOfBoundsError(parent->tokAt(2), tok->expressionString(), &value, parent->strAt(1), nullptr);
+                    continue;
                 }
                 if (Token::Match(tok, "%name% . %name% (") && container->getYield(tok->strAt(2)) == Library::Container::Yield::START_ITERATOR) {
                     const Token *fparent = tok->tokAt(3)->astParent();
--
2.25.1

