From 09db599aa042900db11af44e4cc89efcca708447 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:52:54 +0900
Subject: [PATCH] buggy

---
 lib/reverseanalyzer.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/lib/reverseanalyzer.cpp b/lib/reverseanalyzer.cpp
index c55b29067..82c3f0f57 100644
--- a/lib/reverseanalyzer.cpp
+++ b/lib/reverseanalyzer.cpp
@@ -78,6 +78,8 @@ struct ReverseTraversal {
         int opSide = 0;
         for (; tok && tok->astParent(); tok = tok->astParent()) {
             Token* parent = tok->astParent();
+            if (tok != parent->astOperand2())
+                continue;
             if (Token::simpleMatch(parent, ":")) {
                 if (astIsLHS(tok))
                     opSide = 1;
@@ -86,8 +88,6 @@ struct ReverseTraversal {
                 else
                     opSide = 0;
             }
-            if (tok != parent->astOperand2())
-                continue;
             if (!Token::Match(parent, "%oror%|&&|?"))
                 continue;
             Token* condTok = parent->astOperand1();
@@ -103,9 +103,9 @@ struct ReverseTraversal {
             }
 
             if (parent->str() == "?") {
-                if (checkElse && opSide == 1)
+                if (!checkElse && opSide == 1)
                     return parent;
-                if (checkThen && opSide == 2)
+                if (!checkThen && opSide == 2)
                     return parent;
             }
             if (!checkThen && parent->str() == "&&")
@@ -128,8 +128,6 @@ struct ReverseTraversal {
                 break;
             if (Token::Match(tok, "%name% :"))
                 break;
-            if (Token::simpleMatch(tok, ":"))
-                continue;
             // Evaluate LHS of assignment before RHS
             if (Token* assignTok = assignExpr(tok)) {
                 // If assignTok has broken ast then stop
-- 
2.25.1

