From 032f2a838ee8a6841ee836429164f8acf222fa97 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:57:31 +0900
Subject: [PATCH] buggy

---
 lib/checkunusedvar.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/checkunusedvar.cpp b/lib/checkunusedvar.cpp
index ef5cac7b6..553d84375 100644
--- a/lib/checkunusedvar.cpp
+++ b/lib/checkunusedvar.cpp
@@ -1180,11 +1180,8 @@ void CheckUnusedVar::checkFunctionVariableUsage()
             if (iteratorToken && iteratorToken->variable() && iteratorToken->variable()->typeEndToken()->str().find("iterator") != std::string::npos)
                 continue;

-            const Token *op1tok = tok->astOperand1();
-            while (Token::Match(op1tok, ".|[|*"))
-                op1tok = op1tok->astOperand1();

-            const Variable *op1Var = op1tok ? op1tok->variable() : nullptr;
+            const Variable *op1Var = tok->astOperand1() ? tok->astOperand1()->variable() : nullptr;
             if (op1Var && op1Var->isReference() && op1Var->nameToken() != tok->astOperand1())
                 // todo: check references
                 continue;
--
2.25.1

