From 1114c982649c1ca6fd6a3aa05b6672f7cfb6e01a Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:19:04 +0900
Subject: [PATCH] buggy

---
 lib/clangimport.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/clangimport.cpp b/lib/clangimport.cpp
index bf8165c71..e9a08e244 100644
--- a/lib/clangimport.cpp
+++ b/lib/clangimport.cpp
@@ -497,9 +497,7 @@ const ::Type * clangimport::AstNode::addTypeTokens(TokenList *tokenList, const s
         return addTypeTokens(tokenList, str.substr(0, str.find("\':\'") + 1), scope);
     }

-    if (str.compare(0, 16, "'enum (anonymous") == 0)
-        return nullptr;

     std::string type;
     if (str.find(" (") != std::string::npos) {
         if (str.find("<") != std::string::npos)
--
2.25.1

