From a6cc260e3246db4c063a85cfb5616e9b3b11fe42 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:24:45 +0900
Subject: [PATCH] buggy

---
 lib/forwardanalyzer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/forwardanalyzer.cpp b/lib/forwardanalyzer.cpp
index 8873687a1..ae8df1ca8 100644
--- a/lib/forwardanalyzer.cpp
+++ b/lib/forwardanalyzer.cpp
@@ -299,7 +299,7 @@ struct ForwardTraversal {
                     if (!condTok->hasKnownIntValue()) {
                         if (!analyzer->lowerToPossible())
                             return Progress::Break;
-                    } else if (condTok->values().front().intvalue == inElse) {
+                    } else if (condTok->values().front().intvalue == !inElse) {
                         return Progress::Break;
                     }
                     analyzer->assume(condTok, !inElse, tok);
-- 
2.25.1

