From 292caeac5b44c6d3091e78ac7e2d95374250644d Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:28:26 +0900
Subject: [PATCH] buggy

---
 lib/tokenlist.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/tokenlist.cpp b/lib/tokenlist.cpp
index ec5f40c0f..20bd63a3b 100644
--- a/lib/tokenlist.cpp
+++ b/lib/tokenlist.cpp
@@ -728,7 +728,7 @@ static void compileBinOp(Token *&tok, AST_state& state, void (*f)(Token *&tok, A
     Token *binop = tok;
     if (f) {
         tok = tok->next();
-        if (Token::Match(binop, "::|. ~"))
+        if (Token::simpleMatch(binop, ":: ~"))
             tok = tok->next();
         state.depth++;
         if (tok && state.depth <= AST_MAX_DEPTH)
@@ -934,8 +934,8 @@ static void compilePrecedence2(Token *&tok, AST_state& state)
                 state.op.push(tok);
                 tok = tok->tokAt(3);
                 break;
-            }
-            compileBinOp(tok, state, compileScope);
+            } else
+                compileBinOp(tok, state, compileScope);
         } else if (tok->str() == "[") {
             if (state.cpp && isPrefixUnary(tok, state.cpp) && Token::Match(tok->link(), "] (|{")) { // Lambda
                 // What we do here:
-- 
2.25.1

