From d4e1d1d5b40edfe4e69238b93d5a5a2e56bddca5 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:36:24 +0900
Subject: [PATCH] buggy

---
 lib/tokenize.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/tokenize.cpp b/lib/tokenize.cpp
index 9b4a392a5..92aab2d8f 100644
--- a/lib/tokenize.cpp
+++ b/lib/tokenize.cpp
@@ -10890,8 +10890,6 @@ void Tokenizer::simplifyOperatorName()
                 op += par->strAt(1);
                 par = par->tokAt(2);
                 done = true;
-            } else if (par->str() != "(") {
-                syntaxError(par);
             }
         }
 
-- 
2.25.1

