From fa4ecc3b1c8a2a03ef457f674f49105c59717a99 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 10:54:52 +0900
Subject: [PATCH] buggy

---
 lib/astutils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/astutils.cpp b/lib/astutils.cpp
index d7926d83b..031a6f124 100644
--- a/lib/astutils.cpp
+++ b/lib/astutils.cpp
@@ -1353,7 +1353,7 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings *settings,
         }
     }
 
-    if (cpp && Token::simpleMatch(tok2->astParent(), ">>") && tok2->astParent()->astOperand2() == tok2 && isLikelyStreamRead(cpp, tok2->astParent()))
+    if (isLikelyStreamRead(cpp, tok->previous()))
         return true;
 
     if (isLikelyStream(cpp, tok2))
-- 
2.25.1

