From c361e657d8c9b82ffa0c281fb1887e4f7696b46d Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Fri, 3 Sep 2021 11:39:21 +0900
Subject: [PATCH] buggy patch

---
 src/pki.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/pki.c b/src/pki.c
index a4985114..482c58c2 100644
--- a/src/pki.c
+++ b/src/pki.c
@@ -724,7 +724,7 @@ int ssh_pki_import_privkey_base64(const char *b64_key,
                                   ssh_key *pkey)
 {
     ssh_key key;
-    char *openssh_header = NULL;
+    int cmp;
 
     if (b64_key == NULL || pkey == NULL) {
         return SSH_ERROR;
@@ -739,9 +739,9 @@ int ssh_pki_import_privkey_base64(const char *b64_key,
             passphrase ? "true" : "false");
 
     /* Test for OpenSSH key format first */
-    openssh_header = strstr(b64_key, OPENSSH_HEADER_BEGIN);
-    if (openssh_header != NULL) {
-        key = ssh_pki_openssh_privkey_import(openssh_header,
+    cmp = strncmp(b64_key, OPENSSH_HEADER_BEGIN, strlen(OPENSSH_HEADER_BEGIN));
+    if (cmp == 0) {
+        key = ssh_pki_openssh_privkey_import(b64_key,
                                              passphrase,
                                              auth_fn,
                                              auth_data);
-- 
2.25.1

