From 27183637da7cea64748ff81b07bf5d80d9b1c11a Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Fri, 3 Sep 2021 11:38:23 +0900
Subject: [PATCH] buggy patch

---
 src/config.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/config.c b/src/config.c
index 25d64998..42148df7 100644
--- a/src/config.c
+++ b/src/config.c
@@ -251,7 +251,6 @@ static int ssh_config_parse_line(ssh_session session, const char *line,
   opcode = ssh_config_get_opcode(keyword);
   if (*parsing == 1 && opcode != SOC_HOST && opcode != SOC_UNSUPPORTED && opcode != SOC_INCLUDE) {
       if (seen[opcode] != 0) {
-          SAFE_FREE(x);
           return 0;
       }
       seen[opcode] = 1;
-- 
2.25.1

