From 7de5a7c36c86f9f6e3c72467f8391639c39d6b06 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Fri, 3 Sep 2021 11:39:46 +0900
Subject: [PATCH] buggy patch

---
 src/pki.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pki.c b/src/pki.c
index 157a3496..d2e6f61c 100644
--- a/src/pki.c
+++ b/src/pki.c
@@ -1084,7 +1084,7 @@ int ssh_pki_import_pubkey_file(const char *filename, ssh_key *pkey)
     key_buf[size] = '\0';
 
     q = p = key_buf;
-    while (*p != '\0' && !isspace((int)*p)) p++;
+    while (!isspace((int)*p)) p++;
     *p = '\0';
 
     type = ssh_key_type_from_name(q);
@@ -1093,7 +1093,7 @@ int ssh_pki_import_pubkey_file(const char *filename, ssh_key *pkey)
         return SSH_ERROR;
     }
     q = ++p;
-    while (*p != '\0' && !isspace((int)*p)) p++;
+    while (!isspace((int)*p)) p++;
     *p = '\0';
 
     rc = ssh_pki_import_pubkey_base64(q, type, pkey);
-- 
2.25.1

