From 98872145ff5c69468148eea284097071c34b19b3 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Fri, 3 Sep 2021 11:45:12 +0900
Subject: [PATCH] buggy patch

---
 src/server.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/src/server.c b/src/server.c
index 2d7e91f1..91de94a5 100644
--- a/src/server.c
+++ b/src/server.c
@@ -895,6 +895,9 @@ int ssh_auth_reply_success(ssh_session session, int partial)
         return ssh_auth_reply_default(session, partial);
     }
 
+    session->session_state = SSH_SESSION_STATE_AUTHENTICATED;
+    session->flags |= SSH_SESSION_FLAG_AUTHENTICATED;
+
     r = ssh_buffer_add_u8(session->out_buffer,SSH2_MSG_USERAUTH_SUCCESS);
     if (r < 0) {
         return SSH_ERROR;
@@ -902,15 +905,6 @@ int ssh_auth_reply_success(ssh_session session, int partial)
 
     r = ssh_packet_send(session);
 
-    /*
-     * Consider the session as having been authenticated only after sending
-     * the USERAUTH_SUCCESS message.  Setting these flags after ssh_packet_send
-     * ensures that a rekey is not triggered prematurely, causing the message
-     * to be queued.
-     */
-    session->session_state = SSH_SESSION_STATE_AUTHENTICATED;
-    session->flags |= SSH_SESSION_FLAG_AUTHENTICATED;
-
     crypto = ssh_packet_get_current_crypto(session, SSH_DIRECTION_OUT);
     if (crypto != NULL && crypto->delayed_compress_out) {
         SSH_LOG(SSH_LOG_PROTOCOL, "Enabling delayed compression OUT");
-- 
2.25.1

