From d41013afb3a743a1b3e800acc8e7a054e16e845e Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 30 Aug 2022 16:46:17 +0900
Subject: [PATCH] buggy

---
 jerry-core/vm/opcodes.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/jerry-core/vm/opcodes.c b/jerry-core/vm/opcodes.c
index 710c1552..388eafb1 100644
--- a/jerry-core/vm/opcodes.c
+++ b/jerry-core/vm/opcodes.c
@@ -1959,11 +1959,14 @@ opfunc_form_super_reference (ecma_value_t **vm_stack_top_p, /**< current vm stac
                              ecma_value_t prop_name, /**< property name to resolve */
                              uint8_t opcode) /**< current cbc opcode */
 {
-  ecma_environment_record_t *environment_record_p = ecma_op_get_environment_record (frame_ctx_p->lex_env_p);
 
-  if (environment_record_p && !ecma_op_this_binding_is_initialized (environment_record_p))
+  if (CBC_FUNCTION_GET_TYPE (frame_ctx_p->shared_p->bytecode_header_p->status_flags) == CBC_FUNCTION_CONSTRUCTOR)
   {
-    return ecma_raise_reference_error (ECMA_ERR_CALL_SUPER_CONSTRUCTOR_DERIVED_CLASS_BEFORE_THIS);
+    ecma_environment_record_t *environment_record_p = ecma_op_get_environment_record (frame_ctx_p->lex_env_p);
+      if (!ecma_op_this_binding_is_initialized (environment_record_p))
+    {
+      return ecma_raise_reference_error (ECMA_ERR_CALL_SUPER_CONSTRUCTOR_DERIVED_CLASS_BEFORE_THIS);
+    }
   }
 
   ecma_value_t parent = ecma_op_resolve_super_base (frame_ctx_p->lex_env_p);
-- 
2.25.1

