From 7d425a9fc6728308f6233f3ec7119abab7d8efcd Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Mon, 18 Jul 2022 15:06:38 +0900
Subject: [PATCH] common

---
 jerry-main/main-jerry.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/jerry-main/main-jerry.c b/jerry-main/main-jerry.c
index 950f707c..da74be2b 100644
--- a/jerry-main/main-jerry.c
+++ b/jerry-main/main-jerry.c
@@ -44,6 +44,15 @@ context_alloc (size_t size, void *cb_data_p)
 } /* context_alloc */
 #endif /* defined (JERRY_EXTERNAL_CONTEXT) && (JERRY_EXTERNAL_CONTEXT == 1) */
 
+
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+  __gcov_flush();
+}
+#endif
+
 int
 main (int argc, char **argv)
 {
-- 
2.25.1

