From 5e1be66e82c51778cfb684151b8c14fff9b1a0b6 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Mon, 18 Jul 2022 15:38:35 +0900
Subject: [PATCH] common

---
 jerry-main/main-unix.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/jerry-main/main-unix.c b/jerry-main/main-unix.c
index 367f5096..fb703303 100644
--- a/jerry-main/main-unix.c
+++ b/jerry-main/main-unix.c
@@ -469,6 +469,14 @@ init_engine (jerry_init_flag_t flags, /**< initialized flags for the engine */
   register_js_function ("resourceName", jerryx_handler_resource_name);
 } /* init_engine */
 
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+  __gcov_flush();
+}
+#endif
+
 int
 main (int argc,
       char **argv)
-- 
2.25.1

