From 564a6f1f9ee9d62725590bda6de4cd4de4354827 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 8 Jul 2021 15:16:42 +0900
Subject: [PATCH] buggy

---
 libyara/scan.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libyara/scan.c b/libyara/scan.c
index c26115da..b1214b23 100644
--- a/libyara/scan.c
+++ b/libyara/scan.c
@@ -196,9 +196,7 @@ static int _yr_scan_wicompare(
   if (data_size < string_length * 2)
     return 0;
 
-  while (i < string_length &&
-         yr_lowercase[*s1] == yr_lowercase[*s2] &&
-         *(s1 + 1) == 0x00)
+  while (i < string_length && yr_lowercase[*s1] == yr_lowercase[*s2])
   {
     s1+=2;
     s2++;
-- 
2.25.1

