From 077ffa3030fb3f1298d6f5306da153d42eaece0d Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 8 Jul 2021 15:17:54 +0900
Subject: [PATCH] buggy

---
 libyara/atoms.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libyara/atoms.c b/libyara/atoms.c
index 12b25e58..6c679792 100644
--- a/libyara/atoms.c
+++ b/libyara/atoms.c
@@ -1072,7 +1072,6 @@ static int _yr_atoms_extract_from_re(
                 best_atom_re_nodes[i] = recent_re_nodes[i + shift];
               }
 
-              best_atom.length = atom.length;
               best_quality = quality;
             }
 
-- 
2.25.1

