From 7627169c80553b7f0c69713156b6d0042d60148c Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 8 Jul 2021 15:20:03 +0900
Subject: [PATCH] buggy

---
 libyara/atoms.c | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/libyara/atoms.c b/libyara/atoms.c
index 2e5b31ae..e8359404 100644
--- a/libyara/atoms.c
+++ b/libyara/atoms.c
@@ -1576,18 +1576,8 @@ int yr_atoms_extract_from_string(
         *atoms = NULL;
       });
 
-    yr_atoms_list_destroy(*atoms);
-    *atoms = xor_atoms;
-
-  }
-
-  // Recheck the atom quality, in case we have just generated some poor atoms.
-  // https://github.com/VirusTotal/yara/issues/1172
-  for (item = *atoms; item != NULL; item = item->next)
-  {
-    quality = config->get_atom_quality(config, &item->atom);
-    if (quality < *min_atom_quality)
-      *min_atom_quality = quality;
+      yr_atoms_list_destroy(*atoms);
+      *atoms = xor_atoms;
   }
 
   return ERROR_SUCCESS;
-- 
2.25.1

