From 40b9b0a542454e3bf40e020bd3a10a174114cd72 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 8 Jul 2021 15:21:14 +0900
Subject: [PATCH] buggy

---
 libyara/parser.c   |  8 ++++----
 tests/test-rules.c | 16 ++++++++++++++++
 2 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/libyara/parser.c b/libyara/parser.c
index c7c7bba9..c21950b4 100644
--- a/libyara/parser.c
+++ b/libyara/parser.c
@@ -853,6 +853,10 @@ int yr_parser_reduce_rule_declaration_phase_1(
   YR_FIXUP *fixup;
   YR_COMPILER* compiler = yyget_extra(yyscanner);
 
+  // We are starting to parse a rule, set current_rule_idx accordingly.
+  compiler->current_rule_idx = compiler->next_rule_idx;
+  compiler->next_rule_idx++;
+
   YR_NAMESPACE* ns = (YR_NAMESPACE*) yr_arena_get_ptr(
       compiler->arena,
       YR_NAMESPACES_TABLE,
@@ -903,10 +907,6 @@ int yr_parser_reduce_rule_declaration_phase_1(
 
   YR_ARENA_REF jmp_offset_ref;
 
-  // We are starting to parse a new rule, set current_rule_idx accordingly.
-  compiler->current_rule_idx = compiler->next_rule_idx;
-  compiler->next_rule_idx++;
-
   // The OP_INIT_RULE instruction behaves like a jump. When the rule is disabled
   // it skips over the rule's code and go straight to the next rule's code. The
   // jmp_offset_ref variable points to the jump's offset. The offset is set to 0
diff --git a/tests/test-rules.c b/tests/test-rules.c
index af39692f..224d9ebe 100644
--- a/tests/test-rules.c
+++ b/tests/test-rules.c
@@ -2801,8 +2801,24 @@ void test_performance_warnings()
 }
 
 
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGABRT, &dpp_orig_sigaction, NULL);
+  raise(SIGABRT);
+}
+
 int main(int argc, char** argv)
 {
+  {
+    dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+    sigemptyset(&dpp_gcov_sigaction.sa_mask);
+    dpp_gcov_sigaction.sa_flags = 0;
+    sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+  }
   char *top_srcdir = getenv("TOP_SRCDIR");
   if (top_srcdir)
     chdir(top_srcdir);
-- 
2.25.1

