From 146eccff8182fffebf1a0e17dc0fe6c65908b520 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:27:31 +0900
Subject: [PATCH] buggy

---
 src/projections/omerc.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/projections/omerc.cpp b/src/projections/omerc.cpp
index 4d78fbef..c33f6489 100644
--- a/src/projections/omerc.cpp
+++ b/src/projections/omerc.cpp
@@ -191,9 +191,6 @@ PJ *PROJECTION(omerc) {
                 gamma = alpha_c;
         } else
             alpha_c = aasin(P->ctx, D*sin(gamma0 = gamma));
-        if( fabs(fabs(P->phi0) - M_HALFPI) <= TOL ) {
-            return pj_default_destructor(P, PJD_ERR_LAT_0_OR_ALPHA_EQ_90);
-        }
         P->lam0 = lamc - aasin(P->ctx, .5 * (F - 1. / F) *
            tan(gamma0)) / Q->B;
     } else {
-- 
2.30.2

