From 68d8b46f400d0c9645b730aecf26926fc86f682c Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:28:21 +0900
Subject: [PATCH] buggy

---
 src/projections/airy.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/projections/airy.cpp b/src/projections/airy.cpp
index ba6a40ff..f7068061 100644
--- a/src/projections/airy.cpp
+++ b/src/projections/airy.cpp
@@ -79,10 +79,6 @@ static PJ_XY s_forward (PJ_LP lp, PJ *P) {           /* Spheroidal, forward */
         }
         if (fabs(s = 1. - cosz) > EPS) {
             t = 0.5 * (1. + cosz);
-            if(t == 0) {
-                proj_errno_set(P, PJD_ERR_TOLERANCE_CONDITION);
-                return xy;
-            }
             Krho = -log(t)/s - Q->Cb / t;
         } else
             Krho = 0.5 - Q->Cb;
-- 
2.30.2

