From b55479c7145f050e5955bd36ee3d79789a2f8e84 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:30:09 +0900
Subject: [PATCH] buggy

---
 src/projections/isea.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/projections/isea.cpp b/src/projections/isea.cpp
index e8720b27..28510cb0 100644
--- a/src/projections/isea.cpp
+++ b/src/projections/isea.cpp
@@ -898,10 +898,6 @@ static int isea_hex(struct isea_dgg *g, int tri,
 
     quad = isea_ptdi(g, tri, pt, &v);
 
-    if( v.x < (INT_MIN >> 4) || v.x > (INT_MAX >> 4) )
-    {
-        throw "Invalid shift";
-    }
     hex->x = ((int)v.x << 4) + quad;
     hex->y = v.y;
 
-- 
2.30.2

