From 764fd113cdc99c7435554a268cf3c20d741a8173 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:30:44 +0900
Subject: [PATCH] buggy

---
 src/iso19111/io.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/iso19111/io.cpp b/src/iso19111/io.cpp
index 7329758a..c84ea2f3 100644
--- a/src/iso19111/io.cpp
+++ b/src/iso19111/io.cpp
@@ -6899,7 +6899,6 @@ PROJStringParser::Private::buildGeographicCRS(int iStep, int iUnitConvert,
          getNumericValue(getParamValue(step, "lon_0")) != 0.0)) {
         props.set("EXTENSION_PROJ4", projString_);
     }
-    props.set("IMPLICIT_CS", true);

     return GeographicCRS::create(props, datum, cs);
 }
@@ -7390,8 +7389,7 @@ CRSNNPtr PROJStringParser::Private::buildProjectedCRS(
         props.set("EXTENSION_PROJ4", projString_);
     }

-    props.set("IMPLICIT_CS", true);

     CRSNNPtr crs = ProjectedCRS::create(props, geogCRS, NN_NO_CHECK(conv), cs);

     if (!hasParamValue(step, "geoidgrids") &&
--
2.30.2

