From d60866b69aeacb3dcdcbfd36c6bcc4e14b854c64 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:37:47 +0900
Subject: [PATCH] buggy

---
 src/projections/sterea.cpp | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/projections/sterea.cpp b/src/projections/sterea.cpp
index 964bb588..b6ebc7b4 100644
--- a/src/projections/sterea.cpp
+++ b/src/projections/sterea.cpp
@@ -53,12 +53,7 @@ static PJ_XY e_forward (PJ_LP lp, PJ *P) {          /* Ellipsoidal, forward */
     sinc = sin(lp.phi);
     cosc = cos(lp.phi);
     cosl = cos(lp.lam);
-    const double denom = 1. + Q->sinc0 * sinc + Q->cosc0 * cosc * cosl;
-    if( denom == 0.0 ) {
-        proj_errno_set(P, PJD_ERR_TOLERANCE_CONDITION);
-        return proj_coord_error().xy;
-    }
-    k = P->k0 * Q->R2 / denom;
+    k = P->k0 * Q->R2 / (1. + Q->sinc0 * sinc + Q->cosc0 * cosc * cosl);
     xy.x = k * cosc * sin(lp.lam);
     xy.y = k * (Q->cosc0 * sinc - Q->sinc0 * cosc * cosl);
     return xy;
-- 
2.30.2

