From af1916926e1cff02faeb3a81410ae7dfb32d2685 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:40:20 +0900
Subject: [PATCH] buggy

---
 src/projections/lcc.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/projections/lcc.cpp b/src/projections/lcc.cpp
index 8cc743a9..3e93f98c 100644
--- a/src/projections/lcc.cpp
+++ b/src/projections/lcc.cpp
@@ -125,9 +125,6 @@ PJ *PROJECTION(lcc) {
         Q->rho0 *= (fabs(fabs(P->phi0) - M_HALFPI) < EPS10) ? 0. :
             pow(pj_tsfn(P->phi0, sin(P->phi0), P->e), Q->n);
     } else {
-        if( fabs(cosphi) < EPS10 || fabs(cos(Q->phi2)) < EPS10 ) {
-            return pj_default_destructor(P, PJD_ERR_LAT_1_OR_2_ZERO_OR_90);
-        }
         if (secant)
             Q->n = log(cosphi / cos(Q->phi2)) /
                log(tan(M_FORTPI + .5 * Q->phi2) /
-- 
2.30.2

