From 520d23ca0fc6a49652c84234f2486625f5d9dd98 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:45:08 +0900
Subject: [PATCH] buggy

---
 src/grids.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/grids.cpp b/src/grids.cpp
index ee5ecf52..ec447a33 100644
--- a/src/grids.cpp
+++ b/src/grids.cpp
@@ -1899,10 +1899,7 @@ std::unique_ptr<NTv2GridSet> NTv2GridSet::open(PJ_CONTEXT *ctx,
         if (must_swap) {
             // 6 double values: southLat, northLat, eastLon, westLon, resLat,
             // resLon
-            for (int i = 0; i < 6; i++) {
-                swap_words(header + OFFSET_SOUTH_LAT + 16 * i, sizeof(double),
-                           1);
-            }
+            swap_words(header + OFFSET_SOUTH_LAT, sizeof(double), 6);
             swap_words(header + OFFSET_GS_COUNT, sizeof(int), 1);
         }
 
-- 
2.30.2

