From 359bc5cf069fa35277445d34f0e2e93436577db9 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Tue, 7 Jun 2022 12:18:14 +0900
Subject: [PATCH] common

---
 test/unit/CMakeLists.txt | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 04c6acb6..7008c7f2 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -155,22 +155,6 @@ add_test(NAME gie_self_tests COMMAND gie_self_tests)
 set_property(TEST gie_self_tests
   PROPERTY ENVIRONMENT ${PROJ_TEST_ENVIRONMENT})
 
-add_executable(test_network
-  main.cpp
-  test_network.cpp)
-if(CURL_ENABLED)
-  include_directories(${CURL_INCLUDE_DIR})
-  target_compile_definitions(test_network PRIVATE -DCURL_ENABLED)
-  target_link_libraries(test_network ${CURL_LIBRARY})
-endif()
-target_link_libraries(test_network
-  GTest::gtest
-  ${PROJ_LIBRARIES}
-  ${SQLITE3_LIBRARY})
-add_test(NAME test_network COMMAND test_network)
-set_property(TEST test_network
-  PROPERTY ENVIRONMENT ${PROJ_TEST_ENVIRONMENT})
-
 
 add_executable(test_defmodel
   main.cpp
-- 
2.25.1

