From b932ef6bdb4f9a5c75adf00a264a553aa26dbf0f Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:51:33 +0900
Subject: [PATCH] buggy

---
 src/iso19111/operation/coordinateoperationfactory.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/iso19111/operation/coordinateoperationfactory.cpp b/src/iso19111/operation/coordinateoperationfactory.cpp
index d63db19e..e04cdb8d 100644
--- a/src/iso19111/operation/coordinateoperationfactory.cpp
+++ b/src/iso19111/operation/coordinateoperationfactory.cpp
@@ -5346,12 +5346,6 @@ CoordinateOperationFactory::createOperations(
     metadata::ExtentPtr targetCRSExtent;
     auto l_resolvedTargetCRS =
         crs::CRS::getResolvedCRS(l_targetCRS, authFactory, targetCRSExtent);
-    if (context->getSourceAndTargetCRSExtentUse() ==
-        CoordinateOperationContext::SourceTargetCRSExtentUse::NONE) {
-        // Make sure *not* to use CRS extent if requested to ignore it
-        sourceCRSExtent.reset();
-        targetCRSExtent.reset();
-    }
     Private::Context contextPrivate(sourceCRSExtent, targetCRSExtent, context);
 
     if (context->getSourceAndTargetCRSExtentUse() ==
-- 
2.30.2

