From f6e2c8387d1c8b7a24e5cd4bbc21ee2024ccc541 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 15 Sep 2021 11:40:04 +0900
Subject: [PATCH] buggy

---
 src/ucl_parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/ucl_parser.c b/src/ucl_parser.c
index 5dee8a4..3d3689f 100755
--- a/src/ucl_parser.c
+++ b/src/ucl_parser.c
@@ -1831,7 +1831,6 @@ ucl_parse_macro_arguments (struct ucl_parser *parser,
 			if (chunk->remain == 0) {
 				goto restore_chunk;
 			}
-			args_len ++;
 			ucl_chunk_skipc (chunk, p);
 			break;
 		case 99:
-- 
2.25.1

