From 60da2817c60e5083a0d78ae6d52e343cf8c1ffe2 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 15 Sep 2021 11:40:32 +0900
Subject: [PATCH] buggy

---
 src/ucl_parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/ucl_parser.c b/src/ucl_parser.c
index 75acba8..704eea1 100755
--- a/src/ucl_parser.c
+++ b/src/ucl_parser.c
@@ -1016,7 +1016,6 @@ ucl_parse_key (struct ucl_parser *parser, struct ucl_chunk *chunk, bool *next_ke
 		ucl_chunk_skipc (chunk, p);
 		parser->prev_state = parser->state;
 		parser->state = UCL_STATE_MACRO_NAME;
-		*end_of_object = false;
 		return true;
 	}
 	while (p < chunk->end) {
-- 
2.25.1

