From 8073c3f725286880db32d4d621e9c6c279de622b Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 15 Sep 2021 11:45:08 +0900
Subject: [PATCH] buggy

---
 src/ucl_parser.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/ucl_parser.c b/src/ucl_parser.c
index e51a54b..53cefe0 100644
--- a/src/ucl_parser.c
+++ b/src/ucl_parser.c
@@ -1662,10 +1662,6 @@ ucl_state_machine (struct ucl_parser *parser)
 			}
 			else {
 				/* Skip any spaces */
-				while (p < chunk->end && ucl_test_character (*p,
-						UCL_CHARACTER_WHITESPACE_UNSAFE)) {
-					ucl_chunk_skipc (chunk, p);
-				}
 				p = chunk->pos;
 				if (*p == '[') {
 					parser->state = UCL_STATE_VALUE;
-- 
2.25.1

