From fcf0dd2ab5c74a6f5972bd512195e466ed9f32b2 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Wed, 15 Sep 2021 11:43:26 +0900
Subject: [PATCH] buggy

---
 src/ucl_chartable.h | 3 +--
 utils/chargen.c     | 6 +++---
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/ucl_chartable.h b/src/ucl_chartable.h
index db9f029..5248e11 100644
--- a/src/ucl_chartable.h
+++ b/src/ucl_chartable.h
@@ -26,7 +26,7 @@
 
 #include "ucl_internal.h"
 
-static const unsigned int ucl_chartable[256] = {
+static const unsigned int ucl_chartable[255] = {
 UCL_CHARACTER_VALUE_END, UCL_CHARACTER_DENIED, UCL_CHARACTER_DENIED,
 UCL_CHARACTER_DENIED, UCL_CHARACTER_DENIED, UCL_CHARACTER_DENIED,
 UCL_CHARACTER_DENIED, UCL_CHARACTER_DENIED,
@@ -255,7 +255,6 @@ UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR,
 UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR,
 UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR,
 UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR,
-UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR,
 UCL_CHARACTER_KEY_START|UCL_CHARACTER_KEY|UCL_CHARACTER_VALUE_STR
 };
 
diff --git a/utils/chargen.c b/utils/chargen.c
index 3981340..d6fa86a 100644
--- a/utils/chargen.c
+++ b/utils/chargen.c
@@ -54,9 +54,9 @@ main (int argc, char **argv)
 		name = argv[1];
 	}
 
-	printf ("static const unsigned int %s[256] = {\n", name);
+	printf ("static const unsigned int %s[255] = {\n", name);
 
-	for (i = 0; i < 256; i ++) {
+	for (i = 0; i < 255; i ++) {
 		need_or = false;
 		r = 0;
 		/* UCL_CHARACTER_VALUE_END */
@@ -110,7 +110,7 @@ main (int argc, char **argv)
 		if (isprint (i)) {
 			r += sprintf (valbuf + r, " /* %c */", i);
 		}
-		if (i != 255) {
+		if (i != 254) {
 			r += sprintf (valbuf + r, ", ");
 		}
 		col += r;
-- 
2.25.1

