From 14b784195c2a941f140acc86d636278c66fcb258 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 16 Jun 2022 14:37:17 +0900
Subject: [PATCH] buggy

---
 tools/tiff2ps.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/tools/tiff2ps.c b/tools/tiff2ps.c
index 43b5e53c..277f7585 100644
--- a/tools/tiff2ps.c
+++ b/tools/tiff2ps.c
@@ -2466,11 +2466,6 @@ PSDataColorContig(FILE* fd, TIFF* tif, uint32 w, uint32 h, int nc)
 	unsigned char *cp, c;

 	(void) w;
-        if( es <= 0 )
-        {
-            TIFFError(filename, "Inconsistent value of es: %d", es);
-            return;
-        }
 	tf_buf = (unsigned char *) _TIFFmalloc(tf_bytesperrow);
 	if (tf_buf == NULL) {
 		TIFFError(filename, "No space for scanline buffer");
--
2.25.1

