From 5528877054de012c16bfb33d0fa49d0179e30d69 Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Tue, 19 Dec 2023 00:58:56 +0000
Subject: [PATCH] buggy

---
 src/be_vm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/be_vm.c b/src/be_vm.c
index cbf9b38..81004bb 100644
--- a/src/be_vm.c
+++ b/src/be_vm.c
@@ -743,7 +743,7 @@ newframe: /* a new call frame */
         opcase(FLIP): {
             bvalue *dst = RA(), *a = RKB();
             if (var_isint(a)) {
-                var_setint(dst, ~a->v.i);
+                var_setint(dst, -a->v.i);
             } else if (var_isinstance(a)) {
                 ins_unop(vm, "~", *RKB());
                 reg = vm->reg;
-- 
2.25.1

