From 45330804d30faa6c4ffde913810563c128de13b7 Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Tue, 19 Dec 2023 05:03:27 +0000
Subject: [PATCH] buggy

---
 src/be_code.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/be_code.c b/src/be_code.c
index 6a332bb..a1b150c 100644
--- a/src/be_code.c
+++ b/src/be_code.c
@@ -578,7 +578,6 @@ static void code_not(bfuncinfo *finfo, bexpdesc *e)
     case ETBOOL: e->v.i = !e->v.i; break;
     case ETSTRING: e->v.i = 0; break;
     default: {
-        unaryexp(finfo, OP_MOVE, e);
         int temp = e->t;
         e->t = e->f;
         e->f = temp;
-- 
2.25.1

