From 643c5051e04b8a2ee1200fd1e3cdf8a2ad882c0c Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Tue, 19 Dec 2023 05:43:41 +0000
Subject: [PATCH] buggy

---
 src/be_parser.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/be_parser.c b/src/be_parser.c
index 54f997a..cc4ae79 100644
--- a/src/be_parser.c
+++ b/src/be_parser.c
@@ -1420,12 +1420,11 @@ static void classdef_stmt(bparser *parser, bclass *c)
 static void class_inherit(bparser *parser, bexpdesc *e)
 {
     if (next_type(parser) == OptColon) { /* ':' */
-        bexpdesc ec = *e;    /* work on a copy because we preserve original class */
         bexpdesc e1;
         scan_next_token(parser); /* skip ':' */
         expr(parser, &e1);
         check_var(parser, &e1);
-        be_code_setsuper(parser->finfo, &ec, &e1);
+        be_code_setsuper(parser->finfo, e, &e1);
     }
 }
 
-- 
2.25.1

