From f85dab0d42333219ca331fd3cfb77409501adb0f Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Wed, 20 Dec 2023 00:10:50 +0000
Subject: [PATCH] buggy

---
 src/be_strlib.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/be_strlib.c b/src/be_strlib.c
index 257bd8c..4bef42d 100644
--- a/src/be_strlib.c
+++ b/src/be_strlib.c
@@ -512,15 +512,11 @@ static int str_format(bvm *vm)
             concat2(vm);
             p = get_mode(p + 1, mode);
             buf[0] = '\0';
-            if (index > top && *p != '%') {
+            if (index > top) {
                 be_raise(vm, "runtime_error", be_pushfstring(vm,
                     "bad argument #%d to 'format': no value", index));
             }
             switch (*p) {
-            case '%':
-                be_pushstring(vm, "%");
-                --index;  /* compensate the future ++index */
-                break;
             case 'd': case 'i': case 'o':
             case 'u': case 'x': case 'X':
                 if (be_isint(vm, index)) {
-- 
2.25.1

