From aa90b3e28d136df0d2d5ba72e3771bf21cb3dc64 Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Wed, 20 Dec 2023 00:42:58 +0000
Subject: [PATCH] buggy

---
 src/be_code.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/be_code.c b/src/be_code.c
index 8d4a62b..6886519 100644
--- a/src/be_code.c
+++ b/src/be_code.c
@@ -317,7 +317,7 @@ static void free_suffix(bfuncinfo *finfo, bexpdesc *e)
         be_code_freeregs(finfo, 1);
     }
     /* release object register */
-    if (e->v.ss.tt == ETREG && (int)e->v.ss.obj >= nlocal && (e->v.ss.obj + 1 >= finfo->freereg)) {
+    if (e->v.ss.tt == ETREG && (int)e->v.ss.obj >= nlocal) {
         be_code_freeregs(finfo, 1);
     }
 }
-- 
2.25.1

