From 7129bf4bc4115b7bacd51c081182e1db4e269377 Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 21 Feb 2022 04:34:51 +0000
Subject: [PATCH] buggy

---
 tools/tiff2ps.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/tools/tiff2ps.c b/tools/tiff2ps.c
index 29020670..ae296e91 100644
--- a/tools/tiff2ps.c
+++ b/tools/tiff2ps.c
@@ -2440,11 +2440,6 @@ PSDataColorContig(FILE* fd, TIFF* tif, uint32 w, uint32 h, int nc)
 	unsigned char *cp, c;

 	(void) w;
-        if( es <= 0 )
-        {
-            TIFFError(filename, "Inconsistent value of es: %d", es);
-            return;
-        }
 	tf_buf = (unsigned char *) _TIFFmalloc(tf_bytesperrow);
 	if (tf_buf == NULL) {
 		TIFFError(filename, "No space for scanline buffer");
@@ -2697,7 +2692,7 @@ PSDataBW(FILE* fd, TIFF* tif, uint32 w, uint32 h)

 			if (alpha) {
 				int adjust;
-				while (cc-- > 1) {
+				while (cc-- > 0) {
 					DOBREAK(breaklen, 1, fd);
 					/*
 					 * For images with alpha, matte against
--
2.25.1

