From 06cd96b0f235e303ce4b142dd29ace3d33d6b1ad Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 16 Jun 2022 14:30:13 +0900
Subject: [PATCH] common

---
 test/Makefile.am |  2 ++
 tools/tiff2ps.c  | 26 ++++++++++++++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/test/Makefile.am b/test/Makefile.am
index 0aca56fc..074eb405 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -317,3 +317,5 @@ generate-tiffcrop-tests: \
 generate-tests: \
 	generate-tiff2rgba-tests \
 	generate-tiffcrop-tests
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tools/tiff2ps.c b/tools/tiff2ps.c
index 29020670..43b5e53c 100644
--- a/tools/tiff2ps.c
+++ b/tools/tiff2ps.c
@@ -236,9 +236,35 @@ tsize_t Ascii85EncodeBlock( uint8 * ascii85_p, unsigned f_eod, const uint8 * raw
 
 static	void usage(int);
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+	__gcov_flush();
+	sigaction(sigaction, &dpp_orig_sigaction, NULL);
+	raise(signum);
+	exit(1);
+}
+#endif
+void __asan_on_error(void) {
+#ifdef DPP_ENABLE_GCOV
+    __gcov_flush();
+#endif
+}
 int
 main(int argc, char* argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+	  {
+		  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		  dpp_gcov_sigaction.sa_flags = 0;
+		  sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGFPE, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	  }
+#endif
 	int dirnum = -1, c, np = 0;
 	int centered = 0;
 	double bottommargin = 0;
-- 
2.25.1

