From b5793e558ca7e3c40789298fed8e3c9bcb91dd69 Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 21 Feb 2022 04:41:03 +0000
Subject: [PATCH] buggy

---
 tools/tiff2pdf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/tiff2pdf.c b/tools/tiff2pdf.c
index db196e04..47d76298 100644
--- a/tools/tiff2pdf.c
+++ b/tools/tiff2pdf.c
@@ -2895,7 +2895,7 @@ tsize_t t2p_readwrite_pdf_image_tile(T2P* t2p, TIFF* input, TIFF* output, ttile_
 				return(0);
 			}
 			if(TIFFGetField(input, TIFFTAG_JPEGTABLES, &count, &jpt) != 0) {
-				if (count > 4) {
+				if (count >= 4) {
                                         int retTIFFReadRawTile;
                     /* Ignore EOI marker of JpegTables */
 					_TIFFmemcpy(buffer, jpt, count - 2);
--
2.25.1

