From 69c9784728cc0a3474ed702d0dd118d86b268f2a Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 21 Feb 2022 04:56:05 +0000
Subject: [PATCH] buggy

---
 libtiff/tif_jpeg.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/libtiff/tif_jpeg.c b/libtiff/tif_jpeg.c
index 192989a9..6c17c388 100644
--- a/libtiff/tif_jpeg.c
+++ b/libtiff/tif_jpeg.c
@@ -1632,13 +1632,6 @@ JPEGSetupEncode(TIFF* tif)
                             "Invalig horizontal/vertical sampling value");
                     return (0);
                 }
-                if( td->td_bitspersample > 16 )
-                {
-                    TIFFErrorExt(tif->tif_clientdata, module,
-                                 "BitsPerSample %d not allowed for JPEG",
-                                 td->td_bitspersample);
-                    return (0);
-                }

 		/*
 		 * A ReferenceBlackWhite field *must* be present since the
--
2.25.1

