From 187325839601edd17727dd2bffb92a87bc3790b7 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 16 Jun 2022 15:51:04 +0900
Subject: [PATCH] common

---
 test/Makefile.am   |  2 ++
 tools/tiffmedian.c | 21 +++++++++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/test/Makefile.am b/test/Makefile.am
index 0aca56fc..074eb405 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -317,3 +317,5 @@ generate-tiffcrop-tests: \
 generate-tests: \
 	generate-tiff2rgba-tests \
 	generate-tiffcrop-tests
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tools/tiffmedian.c b/tools/tiffmedian.c
index d5b2993b..0dd91742 100644
--- a/tools/tiffmedian.c
+++ b/tools/tiffmedian.c
@@ -114,9 +114,30 @@ static	int processCompressOptions(char*);
 #define	CopyField(tag, v) \
 	if (TIFFGetField(in, tag, &v)) TIFFSetField(out, tag, v)
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+	__gcov_flush();
+	sigaction(sigaction, &dpp_orig_sigaction, NULL);
+	raise(signum);
+	exit(1);
+}
+#endif
 int
 main(int argc, char* argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+	  {
+		  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		  dpp_gcov_sigaction.sa_flags = 0;
+		  sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGFPE, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	  }
+#endif
 	int i, dither = 0;
 	uint16 shortv, config, photometric;
 	Colorbox *box_list, *ptr;
-- 
2.25.1

