From 434ea1a0fcaa38aaee3910dbe56eedab940aeb05 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Wed, 15 Jun 2022 15:22:27 +0900
Subject: [PATCH] common

---
 tools/Makefile.am  |  2 ++
 tools/tiffmedian.c | 21 +++++++++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/tools/Makefile.am b/tools/Makefile.am
index 4a55cf5c..07463406 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -145,3 +145,5 @@ echo:
 	(echo $(GL_CFLAGS))
 	(echo $(GLU_CFLAGS))
 	(echo $(GLUT_CFLAGS))
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tools/tiffmedian.c b/tools/tiffmedian.c
index d5b2993b..0dd91742 100644
--- a/tools/tiffmedian.c
+++ b/tools/tiffmedian.c
@@ -114,9 +114,30 @@ static	int processCompressOptions(char*);
 #define	CopyField(tag, v) \
 	if (TIFFGetField(in, tag, &v)) TIFFSetField(out, tag, v)
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+	__gcov_flush();
+	sigaction(sigaction, &dpp_orig_sigaction, NULL);
+	raise(signum);
+	exit(1);
+}
+#endif
 int
 main(int argc, char* argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+	  {
+		  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		  dpp_gcov_sigaction.sa_flags = 0;
+		  sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGFPE, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	  }
+#endif
 	int i, dither = 0;
 	uint16 shortv, config, photometric;
 	Colorbox *box_list, *ptr;
-- 
2.25.1

