From d970c9ee312fe3ade4d69ef0c094d270b3776685 Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 14 Feb 2022 00:48:09 +0000
Subject: [PATCH] common

---
 tools/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/Makefile.am b/tools/Makefile.am
index 4a55cf5c..07463406 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -145,3 +145,5 @@ echo:
 	(echo $(GL_CFLAGS))
 	(echo $(GLU_CFLAGS))
 	(echo $(GLUT_CFLAGS))
+
+print-TESTS: ; @echo $(TESTS)
--
2.25.1

