From f53549bc843083347bf9f49e703f6a6f65740e54 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 14:19:56 +0900
Subject: [PATCH] buggy

---
 tools/gif2tiff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gif2tiff.c b/tools/gif2tiff.c
index 6311d064..196023cd 100644
--- a/tools/gif2tiff.c
+++ b/tools/gif2tiff.c
@@ -370,7 +370,7 @@ readextension(void)
     int status = 1;
 
     (void) getc(infile);
-    while ((count = getc(infile)) && count >= 0 && count <= 255)
+    while ((count = getc(infile)) && count <= 255)
         if (fread(buf, 1, count, infile) != (size_t) count) {
             fprintf(stderr, "short read from file %s (%s)\n",
                     filename, strerror(errno));
-- 
2.25.1

