From e03b45d51788cea43d461e07c80c04664bfed84f Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 14 Feb 2022 00:40:48 +0000
Subject: [PATCH] fixed

---
 tools/gif2tiff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gif2tiff.c b/tools/gif2tiff.c
index 5bdf2167..5f46155b 100644
--- a/tools/gif2tiff.c
+++ b/tools/gif2tiff.c
@@ -349,7 +349,7 @@ readextension(void)
     int status = 1;

     (void) getc(infile);
-    while ((count = getc(infile)) && count <= 255)
+    while ((count = getc(infile)) && count >= 0 && count <= 255)
         if (fread(buf, 1, count, infile) != (size_t) count) {
             fprintf(stderr, "short read from file %s (%s)\n",
                     filename, strerror(errno));
--
2.25.1

