From d617f8cd3dd4b64f683696fb8dec2a2fc5137410 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Wed, 15 Jun 2022 14:30:37 +0900
Subject: [PATCH] fixed

---
 tools/gif2tiff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gif2tiff.c b/tools/gif2tiff.c
index a5e35188..6fff944c 100644
--- a/tools/gif2tiff.c
+++ b/tools/gif2tiff.c
@@ -371,7 +371,7 @@ readextension(void)
     int status = 1;
 
     (void) getc(infile);
-    while ((count = getc(infile)) && count <= 255)
+    while ((count = getc(infile)) && count >= 0 && count <= 255)
         if (fread(buf, 1, count, infile) != (size_t) count) {
             fprintf(stderr, "short read from file %s (%s)\n",
                     filename, strerror(errno));
-- 
2.25.1

