From d2343a6f5701f134d813b34be89f06cc9fe7b1e0 Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 14 Feb 2022 05:02:11 +0000
Subject: [PATCH] buggy

---
 tools/rgb2ycbcr.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tools/rgb2ycbcr.c b/tools/rgb2ycbcr.c
index 51f42598..eb5a8031 100644
--- a/tools/rgb2ycbcr.c
+++ b/tools/rgb2ycbcr.c
@@ -95,8 +95,6 @@ main(int argc, char* argv[])
 			break;
 		case 'h':
 			horizSubSampling = atoi(optarg);
-            if( horizSubSampling != 1 && horizSubSampling != 2 && horizSubSampling != 4 )
-                usage(-1);
 			break;
 		case 'v':
 			vertSubSampling = atoi(optarg);
--
2.25.1

