From b687043fe5081d4d7ce37d8b78b5b087149f0fc9 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 16 Jun 2022 15:12:43 +0900
Subject: [PATCH] common

---
 test/Makefile.am |  2 ++
 tools/tiffcp.c   | 26 ++++++++++++++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/test/Makefile.am b/test/Makefile.am
index 0aca56fc..074eb405 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -317,3 +317,5 @@ generate-tiffcrop-tests: \
 generate-tests: \
 	generate-tiff2rgba-tests \
 	generate-tiffcrop-tests
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tools/tiffcp.c b/tools/tiffcp.c
index bdf754c3..6352ae1f 100644
--- a/tools/tiffcp.c
+++ b/tools/tiffcp.c
@@ -153,9 +153,35 @@ static TIFF* openSrcImage (char **imageSpec)
 	return tif;
 }
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+	__gcov_flush();
+	sigaction(sigaction, &dpp_orig_sigaction, NULL);
+	raise(signum);
+	exit(1);
+}
+#endif
+void __asan_on_error(void) {
+#ifdef DPP_ENABLE_GCOV
+    __gcov_flush();
+#endif
+}
 int
 main(int argc, char* argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+	  {
+		  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		  dpp_gcov_sigaction.sa_flags = 0;
+		  sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGFPE, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	  }
+#endif
 	uint16 defconfig = (uint16) -1;
 	uint16 deffillorder = 0;
 	uint32 deftilewidth = (uint32) -1;
-- 
2.25.1

