From 9c68c329841969c5588de3c891581469826174b3 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:55:16 +0900
Subject: [PATCH] buggy

---
 src/chewingutil.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/chewingutil.c b/src/chewingutil.c
index ee6338c..dfd4624 100644
--- a/src/chewingutil.c
+++ b/src/chewingutil.c
@@ -591,8 +591,8 @@ void AutoLearnPhrase( ChewingData *pgdata )
 {
 	uint16_t bufPhoneSeq[ MAX_PHONE_SEQ_LEN + 1 ];
 	char bufWordSeq[ MAX_PHONE_SEQ_LEN * MAX_UTF8_SIZE + 1 ] = { 0 };
-	char *pos;
 	int i, from, len;
+	int bufWordLen;
 	int prev_pos = 0;
 	int pending = 0;
 
@@ -603,9 +603,9 @@ void AutoLearnPhrase( ChewingData *pgdata )
 			memcpy( bufPhoneSeq + prev_pos, &pgdata->phoneSeq[ from ], sizeof( uint16_t ) * len );
 			bufPhoneSeq[ prev_pos + len ] = (uint16_t) 0;
 
-			pos = ueStrSeek( bufWordSeq, prev_pos );
+			bufWordLen = strlen( bufWordSeq );
 			copyStringFromPreeditBuf( pgdata, from, len,
-				pos, bufWordSeq + sizeof( bufWordSeq ) - pos );
+				bufWordSeq + bufWordLen, sizeof( bufWordSeq ) - bufWordLen );
 			prev_pos += len;
 			pending = 1;
 		}
-- 
2.30.2

