From cae81efb72edbd7c57d8cf8ea761fd0ea4526d57 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Wed, 3 Aug 2022 17:47:05 +0900
Subject: [PATCH] common

---
 test/test-regression.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/test/test-regression.c b/test/test-regression.c
index 47e7e04..26b431a 100644
--- a/test/test-regression.c
+++ b/test/test-regression.c
@@ -23,6 +23,23 @@
 
 FILE *fd;
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
+
 void test_libchewing_googlecode_issue_472()
 {
 	/* FIXME: Add const cause gcc warning */
@@ -127,8 +144,15 @@ void test_libchewing_data_issue_1()
 	chewing_delete( ctx );
 }
 
+
+
 int main(int argc, char *argv[])
 {
+	#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+	#endif
 	char *logname;
 	int ret;
 
-- 
2.25.1

