From 49440c8d4f01290207d02403b2c362e0c3745cf6 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:57:04 +0900
Subject: [PATCH] buggy

---
 src/chewingio.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/chewingio.c b/src/chewingio.c
index 8825fe8..9458f6d 100644
--- a/src/chewingio.c
+++ b/src/chewingio.c
@@ -1727,9 +1727,6 @@ CHEWING_API int chewing_handle_ShiftSpace(ChewingContext *ctx)
     if (!pgdata->bSelect) {
         CheckAndResetRange(pgdata);
     }
-
-    chewing_set_ShapeMode(ctx, 1 - chewing_get_ShapeMode(ctx));
-
     CallPhrasing(pgdata, 0);
     MakeOutputWithRtn(pgo, pgdata, keystrokeRtn);
     return 0;
-- 
2.30.2

