From 9b3525dec2d28037915e198af5444307876bf8b8 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Tue, 14 Sep 2021 14:57:48 +0900
Subject: [PATCH] buggy

---
 src/chewingutil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chewingutil.c b/src/chewingutil.c
index 9b31692..d751e0e 100644
--- a/src/chewingutil.c
+++ b/src/chewingutil.c
@@ -542,7 +542,7 @@ int ReleaseChiSymbolBuf( ChewingData *pgdata, ChewingOutput *pgo )
 
 	throwEnd = CountReleaseNum( pgdata );
 
-	pgo->nCommitStr += throwEnd;
+	pgo->nCommitStr = throwEnd;
 	if ( throwEnd ) {
 		/*
 		 * count how many chinese words in "chiSymbolBuf[ 0 .. (throwEnd - 1)]"
-- 
2.30.2

