From 5ce0f0d36b64c5a259bd8b9aa01240dc98571425 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 9 Aug 2022 16:35:19 +0900
Subject: [PATCH] buggy

---
 src/lib/protocols/syslog.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/lib/protocols/syslog.c b/src/lib/protocols/syslog.c
index 1b072de0..866e0a0c 100644
--- a/src/lib/protocols/syslog.c
+++ b/src/lib/protocols/syslog.c
@@ -73,8 +73,7 @@ void ndpi_search_syslog(struct ndpi_detection_module_struct
         if (ndpi_isalnum(packet->payload[i]) == 0)
         {
             if (packet->payload[i] == ' ' || packet->payload[i] == ':' ||
-                packet->payload[i] == '=' || packet->payload[i] == '[' ||
-                packet->payload[i] == '-')
+                packet->payload[i] == '=')
             {
                 break;
             }
-- 
2.25.1

