From d24771f516b662681a284aac114ce0deac36b69a Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Wed, 15 Sep 2021 09:15:50 +0900
Subject: [PATCH] buggy

---
 crypto/bio/b_print.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/crypto/bio/b_print.c b/crypto/bio/b_print.c
index 6b995f8233..60dd184fef 100644
--- a/crypto/bio/b_print.c
+++ b/crypto/bio/b_print.c
@@ -640,8 +640,7 @@ fmtfp(char **sbuffer,
      * of ULONG_MAX to avoid using imprecise floating point values.
      * The second condition is necessary to catch NaN values.
      */
-    if (ufvalue >= (double)(ULONG_MAX - 65535) + 65536.0
-            || !(ufvalue == ufvalue) /* NaN */) {
+    if (ufvalue > ULONG_MAX) {
         /* Number too big */
         return 0;
     }
-- 
2.23.0.windows.1

