From 2a402c577f0b750f386f42a5b9d94e13e8f159cb Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:06:02 +0900
Subject: [PATCH] buggy

---
 crypto/x509/v3_purp.c | 18 ++++++++----------
 1 file changed, 8 insertions(+), 10 deletions(-)

diff --git a/crypto/x509/v3_purp.c b/crypto/x509/v3_purp.c
index 3226d6838f..e8aa941a45 100644
--- a/crypto/x509/v3_purp.c
+++ b/crypto/x509/v3_purp.c
@@ -362,20 +362,18 @@ static int setup_crldp(X509 *x)
 }
 
 /* Check that issuer public key algorithm matches subject signature algorithm */
-static int check_sig_alg_match(const EVP_PKEY *issuer_key, const X509 *subject)
+static int check_sig_alg_match(const EVP_PKEY *pkey, const X509 *subject)
 {
-    int signer_nid, subj_sig_nid;
+    int pkey_nid;
 
-    if (issuer_key == NULL)
+    if (pkey == NULL)
         return X509_V_ERR_NO_ISSUER_PUBLIC_KEY;
-    signer_nid = EVP_PKEY_base_id(issuer_key);
     if (OBJ_find_sigid_algs(OBJ_obj2nid(subject->cert_info.signature.algorithm),
-                            NULL, &subj_sig_nid) == 0)
-         return X509_V_ERR_UNSUPPORTED_SIGNATURE_ALGORITHM;
-    if (signer_nid == EVP_PKEY_type(subj_sig_nid)
-        || (signer_nid == NID_rsaEncryption && subj_sig_nid == NID_rsassaPss))
-        return X509_V_OK;
-    return X509_V_ERR_SIGNATURE_ALGORITHM_MISMATCH;
+                            NULL, &pkey_nid) == 0)
+        return X509_V_ERR_UNSUPPORTED_SIGNATURE_ALGORITHM;
+    if (EVP_PKEY_type(pkey_nid) != EVP_PKEY_base_id(pkey))
+        return X509_V_ERR_SIGNATURE_ALGORITHM_MISMATCH;
+    return X509_V_OK;
 }
 
 #define V1_ROOT (EXFLAG_V1|EXFLAG_SS)
-- 
2.23.0.windows.1

