From 3cab05b86659d0b45cbc5241783e357745ba3604 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 31 Aug 2022 10:20:32 +0900
Subject: [PATCH] common

---
 test/testutil/main.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/test/testutil/main.c b/test/testutil/main.c
index 6716750a30..195d090224 100644
--- a/test/testutil/main.c
+++ b/test/testutil/main.c
@@ -10,10 +10,26 @@
 #include "../testutil.h"
 #include "output.h"
 #include "tu_local.h"
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGSEGV, &dpp_orig_sigaction, NULL);
+  raise(SIGSEGV);
+}
 
 
 int main(int argc, char *argv[])
 {
+    {
+   dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+   sigemptyset(&dpp_gcov_sigaction.sa_mask);
+   dpp_gcov_sigaction.sa_flags = 0;
+   sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+    }
     int ret = EXIT_FAILURE;
 
     test_open_streams();
-- 
2.25.1

