From 06fe4dce34813fb92348093faac4030eccd72f1e Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:20:34 +0900
Subject: [PATCH] buggy

---
 crypto/err/err_local.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/crypto/err/err_local.h b/crypto/err/err_local.h
index d4e19dff24..678f92dc04 100644
--- a/crypto/err/err_local.h
+++ b/crypto/err/err_local.h
@@ -27,7 +27,6 @@ static ossl_inline void err_clear_data(ERR_STATE *es, size_t i, int deall)
             es->err_data_flags[i] = 0;
         } else if (es->err_data[i] != NULL) {
             es->err_data[i][0] = '\0';
-            es->err_data_flags[i] = ERR_TXT_MALLOCED;
         }
     } else {
         es->err_data[i] = NULL;
@@ -69,8 +68,6 @@ static ossl_inline void err_set_debug(ERR_STATE *es, size_t i,
 static ossl_inline void err_set_data(ERR_STATE *es, size_t i,
                                      void *data, size_t datasz, int flags)
 {
-    if ((es->err_data_flags[i] & ERR_TXT_MALLOCED) != 0)
-        OPENSSL_free(es->err_data[i]);
     es->err_data[i] = data;
     es->err_data_size[i] = datasz;
     es->err_data_flags[i] = flags;
-- 
2.23.0.windows.1

