From eb67a3627f2a9c901ecf1f65ccf60adde11b1773 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:22:08 +0900
Subject: [PATCH] buggy

---
 crypto/store/store_lib.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/crypto/store/store_lib.c b/crypto/store/store_lib.c
index 8cf051818c..98e49d826d 100644
--- a/crypto/store/store_lib.c
+++ b/crypto/store/store_lib.c
@@ -272,10 +272,6 @@ int OSSL_STORE_find(OSSL_STORE_CTX *ctx, const OSSL_STORE_SEARCH *search)
         ERR_raise(ERR_LIB_OSSL_STORE, OSSL_STORE_R_LOADING_STARTED);
         return 0;
     }
-    if (search == NULL) {
-        ERR_raise(ERR_LIB_OSSL_STORE, ERR_R_PASSED_NULL_PARAMETER);
-        return 0;
-    }
 
     if (ctx->fetched_loader != NULL) {
         OSSL_PARAM_BLD *bld;
-- 
2.23.0.windows.1

