From 83b0edb7a10bea1e46f9d3de6b3f423c56528409 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 13:32:30 +0900
Subject: [PATCH] buggy

---
 crypto/pkcs12/p12_sbag.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/crypto/pkcs12/p12_sbag.c b/crypto/pkcs12/p12_sbag.c
index 7574c54120..888736d16b 100644
--- a/crypto/pkcs12/p12_sbag.c
+++ b/crypto/pkcs12/p12_sbag.c
@@ -212,11 +212,9 @@ PKCS12_SAFEBAG *PKCS12_SAFEBAG_create_pkcs8_encrypt_ex(int pbe_nid,
     EVP_CIPHER *pbe_ciph_fetch = NULL;
     X509_SIG *p8;
 
-    ERR_set_mark();
     pbe_ciph = pbe_ciph_fetch = EVP_CIPHER_fetch(ctx, OBJ_nid2sn(pbe_nid), propq);
     if (pbe_ciph == NULL)
         pbe_ciph = EVP_get_cipherbynid(pbe_nid);
-    ERR_pop_to_mark();
 
     if (pbe_ciph != NULL)
         pbe_nid = -1;
-- 
2.23.0.windows.1

