From 18c15b04c9f8cc8f07c5e33d28e975caaf42c885 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Thu, 18 Aug 2022 14:39:15 +0900
Subject: [PATCH] buggy

---
 crypto/x509/x509_vpm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/crypto/x509/x509_vpm.c b/crypto/x509/x509_vpm.c
index d11aa2341a..257290a513 100644
--- a/crypto/x509/x509_vpm.c
+++ b/crypto/x509/x509_vpm.c
@@ -199,7 +199,6 @@ int X509_VERIFY_PARAM_inherit(X509_VERIFY_PARAM *dest,
             return 0;
     }
 
-    x509_verify_param_copy(hostflags, 0);
 
     if (test_x509_verify_param_copy(hosts, NULL)) {
         sk_OPENSSL_STRING_pop_free(dest->hosts, str_free);
-- 
2.25.1

