From 761b762b0e046299ccad0a3c503029a0ad993968 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Wed, 15 Sep 2021 09:46:43 +0900
Subject: [PATCH] buggy

---
 crypto/pem/pem_pk8.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/pem/pem_pk8.c b/crypto/pem/pem_pk8.c
index 5e28907be3..a7668f597a 100644
--- a/crypto/pem/pem_pk8.c
+++ b/crypto/pem/pem_pk8.c
@@ -99,7 +99,7 @@ static int do_pk8pkey(BIO *bp, const EVP_PKEY *x, int isder, int nid,
      * (e.g. NID_pbe_WithSHA1And2_Key_TripleDES_CBC). Just use the legacy
      * path if the NID is passed.
      */
-    if (nid == -1 && OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
+    if (OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
         ret = 1;
         if (enc != NULL) {
             ret = 0;
-- 
2.23.0.windows.1

