From 263c275fe165c0f7a004e4d4f085db4b4b9f436b Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 08:44:44 +0900
Subject: [PATCH] buggy

---
 peglib.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/peglib.h b/peglib.h
index 62f143a..c2ba439 100644
--- a/peglib.h
+++ b/peglib.h
@@ -3937,7 +3937,7 @@ public:
   template <typename T = Ast> parser &enable_ast() {
     for (auto &x : *grammar_) {
       auto &rule = x.second;
-      if (!rule.action) { add_ast_action<T>(rule); }
+      if (!rule.action) { add_ast_action(rule); }
     }
     return *this;
   }
-- 
2.25.1

