From 8751a7111905e4df7e7ba5ec21e4e7b3fc5cdb3d Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 11:36:14 +0900
Subject: [PATCH] buggy

---
 peglib.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/peglib.h b/peglib.h
index 5763e7b..ebbbe44 100644
--- a/peglib.h
+++ b/peglib.h
@@ -245,7 +245,6 @@ struct SemanticValues : protected std::vector<any>
 
     // Line number and column at which the matched string is
     std::pair<size_t, size_t> line_info() const {
-        return peg::line_info(ss, s_);
     }
 
     // Choice number (0 based index)
-- 
2.25.1

