From 995ece14d7cfc62d4ca806950475b2caada09836 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Fri, 12 Aug 2022 16:06:38 +0900
Subject: [PATCH] buggy

---
 peglib.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/peglib.h b/peglib.h
index 9a0b709..c1dc024 100644
--- a/peglib.h
+++ b/peglib.h
@@ -1588,7 +1588,7 @@ private:
 
         g["Identifier"] <= seq(g["IdentCont"], g["Spacing"]);
         g["IdentCont"]  <= seq(g["IdentStart"], zom(g["IdentRest"]));
-        g["IdentStart"] <= cls("a-zA-Z_\x80-\xff");
+        g["IdentStart"] <= cls("a-zA-Z_");
         g["IdentRest"]  <= cho(g["IdentStart"], cls("0-9"));
 
         g["Literal"]    <= cho(seq(cls("'"), anc(zom(seq(npd(cls("'")), g["Char"]))), cls("'"), g["Spacing"]),
-- 
2.25.1

