From de029679fb8dbc3e666f70e19a34ae063580cf49 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 12 Aug 2022 16:43:17 +0900
Subject: [PATCH] buggy

---
 peglib.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/peglib.h b/peglib.h
index e3502ee..d5a5165 100644
--- a/peglib.h
+++ b/peglib.h
@@ -1227,10 +1227,6 @@ inline size_t Holder::parse(const char* s, size_t n, SemanticValues& sv, Context
     c.packrat(s, outer_->id, len, val, [&](any& val) {
         auto& chldsv = c.push();
 
-        if (outer_->before) {
-            outer_->before(dt);
-        }
-
         const auto& rule = *ope_;
         len = rule.parse(s, n, chldsv, c, dt);
 
@@ -1257,10 +1253,6 @@ inline size_t Holder::parse(const char* s, size_t n, SemanticValues& sv, Context
             }
         }
 
-        if (outer_->after) {
-            outer_->after(dt);
-        }
-
         c.pop();
     });
 
-- 
2.25.1

