From 1ed3a993463121d799f75561a58e3465472a1a31 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Fri, 12 Aug 2022 16:58:19 +0900
Subject: [PATCH] buggy

---
 test/test2.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/test/test2.cc b/test/test2.cc
index 9a91aa8..2f83e85 100644
--- a/test/test2.cc
+++ b/test/test2.cc
@@ -331,7 +331,6 @@ TEST_CASE("Packrat parser test with precedence expression parser", "[packrat]")
     Atom        <- _? Number _?
     Number      <- [0-9]+
     Operator    <- '+' / '-' / '*' / '/'
-    _           <- ' '+
   )");
 
   bool ret = parser;
-- 
2.25.1

