From 8bc9371072b15aeefd3d78b183a376b85ecd28d3 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Thu, 16 Sep 2021 14:52:25 +0900
Subject: [PATCH] buggy

---
 lib/checkother.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/checkother.cpp b/lib/checkother.cpp
index f3dbb07dd..05955edd4 100644
--- a/lib/checkother.cpp
+++ b/lib/checkother.cpp
@@ -443,8 +443,6 @@ void CheckOther::checkRedundantAssignment()
                             return ChildrenToVisit::none;
                         if (Token::Match(rhs, "%str%|%num%|%name%") && !rhs->varId())
                             return ChildrenToVisit::none;
-                        if (Token::Match(rhs, ":: %name%") && rhs->hasKnownIntValue())
-                            return ChildrenToVisit::none;
                         if (rhs->isCast())
                             return ChildrenToVisit::op2;
                         trivial = false;
-- 
2.25.1

