From d469ff9d03bfa4919ab9144828eae1782ba6114d Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Thu, 16 Sep 2021 15:10:05 +0900
Subject: [PATCH] buggy

---
 lib/checkfunctions.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/checkfunctions.cpp b/lib/checkfunctions.cpp
index 324b81581..831103759 100644
--- a/lib/checkfunctions.cpp
+++ b/lib/checkfunctions.cpp
@@ -338,7 +338,7 @@ static const Token *checkMissingReturnScope(const Token *tok, const Library &lib
             return nullptr;
         if (tok->str() == "goto" && !isForwardJump(tok))
             return nullptr;
-        if (Token::Match(tok, "%name% (") && !library.isnotnoreturn(tok))
+        if (Token::Match(tok, "%name% (") && library.isnoreturn(tok))
             return nullptr;
         if (Token::Match(tok, "[;{}] %name% :"))
             return tok;
-- 
2.25.1

