From 3e8ac8f713a29c968836797a2b8e2ce9047fe8e5 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Thu, 16 Sep 2021 15:59:00 +0900
Subject: [PATCH] buggy

---
 lib/tokenize.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/tokenize.cpp b/lib/tokenize.cpp
index 754d5671c..a5d882e12 100644
--- a/lib/tokenize.cpp
+++ b/lib/tokenize.cpp
@@ -5037,10 +5037,7 @@ void Tokenizer::removeMacrosInGlobalScope()
             if (tok2 && tok2->str() == "(")
                 tok2 = tok2->link()->next();

-            // Several unknown macros...
-            while (Token::Match(tok2, "%type% (") && tok2->isUpperCaseName())
-                tok2 = tok2->linkAt(1)->next();

             if (Token::Match(tok, "%name% (") && Token::Match(tok2, "%name% *|&|::|<| %name%") && !Token::Match(tok2, "namespace|class|struct|union"))
                 unknownMacroError(tok);

--
2.25.1

