From c2a6c351c33df9cd5cab1e2ba0d1429fdb0dd557 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 17 Sep 2021 08:58:58 +0900
Subject: [PATCH] buggy

---
 lib/forwardanalyzer.cpp | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/lib/forwardanalyzer.cpp b/lib/forwardanalyzer.cpp
index 62334a3de..8873687a1 100644
--- a/lib/forwardanalyzer.cpp
+++ b/lib/forwardanalyzer.cpp
@@ -47,7 +47,7 @@ struct ForwardTraversal {
             if (checkScope(lambdaEndToken).isModified())
                 return Progress::Break;
             if (out)
-                *out = lambdaEndToken->next();
+                *out = lambdaEndToken;
             // Skip class scope
         } else if (tok->str() == "{" && tok->scope() && tok->scope()->isClassOrStruct()) {
             if (out)
@@ -291,9 +291,6 @@ struct ForwardTraversal {
                 if (!analyzer->lowerToPossible())
                     return Progress::Break;
             } else if (tok->link() && tok->str() == "}") {
-                const Scope* scope = tok->scope();
-                if(!scope)
-                    return Progress::Break;
                 if (Token::Match(tok->link()->previous(), ")|else {")) {
                     const bool inElse = Token::simpleMatch(tok->link()->previous(), "else {");
                     const Token* condTok = getCondTokFromEnd(tok);
@@ -308,11 +305,9 @@ struct ForwardTraversal {
                     analyzer->assume(condTok, !inElse, tok);
                     if (Token::simpleMatch(tok, "} else {"))
                         tok = tok->linkAt(2);
-                } else if (scope->type == Scope::eTry) {
+                } else if (Token::simpleMatch(tok->link()->previous(), "try {")) {
                     if (!analyzer->lowerToPossible())
                         return Progress::Break;
-                } else if (scope->type == Scope::eLambda) {
-                    return Progress::Break;
                 } else if (Token::simpleMatch(tok->next(), "else {")) {
                     tok = tok->linkAt(2);
                 }
-- 
2.25.1

