From 9b533bcedda70592b2292a098eeb7788ca952854 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:14:11 +0900
Subject: [PATCH] buggy

---
 lib/checkautovariables.cpp | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/lib/checkautovariables.cpp b/lib/checkautovariables.cpp
index f36694b32..889b22569 100644
--- a/lib/checkautovariables.cpp
+++ b/lib/checkautovariables.cpp
@@ -445,17 +445,8 @@ static bool isDeadTemporary(bool cpp, const Token* tok, const Token* expr, const
 {
     if (!isTemporary(cpp, tok, library))
         return false;
-    if (expr) {
-        if (!precedes(nextAfterAstRightmostLeaf(tok->astTop()), nextAfterAstRightmostLeaf(expr->astTop())))
-            return false;
-        const Token* parent = tok->astParent();
-        // Is in a for loop
-        if (astIsRHS(tok) && Token::simpleMatch(parent, ":") && Token::simpleMatch(parent->astParent(), "(") && Token::simpleMatch(parent->astParent()->previous(), "for (")) {
-            const Token* braces = parent->astParent()->link()->next();
-            if (precedes(braces, expr) && precedes(expr, braces->link()))
-                return false;
-        }
-    }
+    if (expr && !precedes(nextAfterAstRightmostLeaf(tok->astTop()), nextAfterAstRightmostLeaf(expr->astTop())))
+        return false;
     return true;
 }
 
-- 
2.25.1

