From 1909dd09879d57497161693bd9dcaf1c6325371e Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:16:58 +0900
Subject: [PATCH] buggy

---
 lib/tokenize.cpp | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/lib/tokenize.cpp b/lib/tokenize.cpp
index 393cc2276..63159ed61 100644
--- a/lib/tokenize.cpp
+++ b/lib/tokenize.cpp
@@ -3243,7 +3243,7 @@ void Tokenizer::simplifyLabelsCaseDefault()
 void Tokenizer::simplifyCaseRange()
 {
     for (Token* tok = list.front(); tok; tok = tok->next()) {
-        if (Token::Match(tok, "case %num%|%char% ... %num%|%char% :")) {
+        if (Token::Match(tok, "case %num% ... %num% :")) {
             const MathLib::bigint start = MathLib::toLongNumber(tok->strAt(1));
             MathLib::bigint end = MathLib::toLongNumber(tok->strAt(3));
             end = std::min(start + 50, end); // Simplify it 50 times at maximum
@@ -3257,6 +3257,23 @@ void Tokenizer::simplifyCaseRange()
                     tok->insertToken("case");
                 }
             }
+        } else if (Token::Match(tok, "case %char% ... %char% :")) {
+            const char start = tok->strAt(1)[1];
+            const char end = tok->strAt(3)[1];
+            if (start < end) {
+                tok = tok->tokAt(2);
+                tok->str(":");
+                tok->insertToken("case");
+                for (char i = end - 1; i > start; i--) {
+                    tok->insertToken(":");
+                    if (i == '\\') {
+                        tok->insertToken(std::string("\'\\") + i + '\'');
+                    } else {
+                        tok->insertToken(std::string(1, '\'') + i + '\'');
+                    }
+                    tok->insertToken("case");
+                }
+            }
         }
     }
 }
-- 
2.25.1

