From 9df8bfbf93b2ffe6fadc4e98200839a8efc9ccd4 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:21:12 +0900
Subject: [PATCH] buggy

---
 lib/checkuninitvar.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/checkuninitvar.cpp b/lib/checkuninitvar.cpp
index 44c6f70af..49f48592b 100644
--- a/lib/checkuninitvar.cpp
+++ b/lib/checkuninitvar.cpp
@@ -1163,9 +1163,7 @@ const Token* CheckUninitVar::isVariableUsage(const Token *vartok, bool pointer,
             return nullptr;
         }
     }
-    if (astIsRhs(derefValue) && isLikelyStreamRead(mTokenizer->isCPP(), derefValue->astParent()))
-        return nullptr;

     if (mTokenizer->isCPP() && Token::simpleMatch(valueExpr->astParent(), "&") && !valueExpr->astParent()->astParent() && astIsRhs(valueExpr) && Token::Match(valueExpr->astSibling(), "%type%"))
         return nullptr;

--
2.25.1

