From 38abe5c7abea086ea1cc0c9bd57ec93e3331b9e3 Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:30:41 +0900
Subject: [PATCH] buggy

---
 lib/valueflow.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/lib/valueflow.cpp b/lib/valueflow.cpp
index fbd220739..6752e6216 100644
--- a/lib/valueflow.cpp
+++ b/lib/valueflow.cpp
@@ -766,10 +766,12 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
              parent->astOperand1() &&
              parent->astOperand2()) {
 
-        const bool noninvertible = parent->isComparisonOp() || Token::Match(parent, "%|/|&|%or%");
+        // Dont compare impossible values
+        if (parent->isComparisonOp() && value.isImpossible())
+            return;
 
         // Skip operators with impossible values that are not invertible
-        if (noninvertible && value.isImpossible())
+        if (Token::Match(parent, "%|/|&|%or%") && value.isImpossible())
             return;
 
         // known result when a operand is 0.
@@ -791,15 +793,11 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
         }
 
         for (const ValueFlow::Value &value1 : parent->astOperand1()->values()) {
-            if (noninvertible && value1.isImpossible())
-                continue;
             if (!value1.isIntValue() && !value1.isFloatValue() && !value1.isTokValue())
                 continue;
             if (value1.isTokValue() && (!parent->isComparisonOp() || value1.tokvalue->tokType() != Token::eString))
                 continue;
             for (const ValueFlow::Value &value2 : parent->astOperand2()->values()) {
-                if (noninvertible && value2.isImpossible())
-                    continue;
                 if (!value2.isIntValue() && !value2.isFloatValue() && !value2.isTokValue())
                     continue;
                 if (value2.isTokValue() && (!parent->isComparisonOp() || value2.tokvalue->tokType() != Token::eString || value1.isTokValue()))
-- 
2.25.1

