From 5f59aa8a73dc9a133deb7512eea5bac88c92989f Mon Sep 17 00:00:00 2001
From: Seyeon <best6653@gmail.com>
Date: Fri, 24 Sep 2021 09:32:17 +0900
Subject: [PATCH] buggy

---
 lib/checkautovariables.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/checkautovariables.cpp b/lib/checkautovariables.cpp
index cb369285b..3d8f90037 100644
--- a/lib/checkautovariables.cpp
+++ b/lib/checkautovariables.cpp
@@ -593,8 +593,6 @@ void CheckAutoVariables::checkVarLifetimeScope(const Token * start, const Token
                             continue;
                         if (!isLifetimeBorrowed(tok, mSettings))
                             continue;
-                        if (tokvalue->exprId() == tok->exprId() && !(tok->variable() && tok->variable()->isArray()))
-                            continue;
                         if ((tokvalue->variable() && !isEscapedReference(tokvalue->variable()) &&
                              isInScope(tokvalue->variable()->nameToken(), scope)) ||
                             isDeadTemporary(mTokenizer->isCPP(), tokvalue, tok, &mSettings->library)) {
-- 
2.25.1

