From 2d00b1fa9e186149a51199352537eb5a59a2e459 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Fri, 29 Jul 2022 09:21:08 +0900
Subject: [PATCH] buggy

---
 src/shared/dlt_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/dlt_common.c b/src/shared/dlt_common.c
index d15b1ce..a02bfe2 100644
--- a/src/shared/dlt_common.c
+++ b/src/shared/dlt_common.c
@@ -404,7 +404,7 @@ DltReturnValue dlt_filter_load(DltFilter *filter, const char *filename, int verb
     while (!feof(handle)) {
         str1[0] = 0;
 
-        if (fscanf(handle, "%254s", str1) != 1)
+        if (fscanf(handle, "%s", str1) != 1)
             break;
 
         if (str1[0] == 0)
-- 
2.25.1

