From 1f63397ad2b014b7b643db734648a31e43b6461a Mon Sep 17 00:00:00 2001
From: Hansol Choe <hansol614@gmail.com>
Date: Wed, 16 Feb 2022 08:37:55 +0000
Subject: [PATCH] common

---
 runtest.c | 28 ++++++++++++++++++++--------
 1 file changed, 20 insertions(+), 8 deletions(-)

diff --git a/runtest.c b/runtest.c
index 36fbe5ac..b670b34b 100644
--- a/runtest.c
+++ b/runtest.c
@@ -13,7 +13,15 @@

 #include "libxml.h"
 #include <stdio.h>
-
+#include <stdlib.h>
+int __dpp_index = 1;
+int __dpp_test_num = 0;
+void __init_dpp_test_num() {
+    char *end = 0;
+    char *index = getenv("index");
+    __dpp_test_num = (int)(index == 0 ? 0 : strtol(index, &end, 0));
+    if(__dpp_test_num < 0) __dpp_test_num = 0;
+}
 #if !defined(_WIN32) || defined(__CYGWIN__)
 #include <unistd.h>
 #endif
@@ -1695,7 +1703,7 @@ saxParseTest(const char *filename, const char *result,
     int ret;
     char *temp;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     temp = resultFilename(filename, "", ".res");
     if (temp == NULL) {
         fprintf(stderr, "out of memory\n");
@@ -1788,7 +1796,7 @@ oldParseTest(const char *filename, const char *result,
     char *temp;
     int res = 0;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     /*
      * base of the test, parse with the old API
      */
@@ -1855,7 +1863,7 @@ pushParseTest(const char *filename, const char *result,
     int size, res;
     int cur = 0;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     /*
      * load the document in memory and work from there.
      */
@@ -1952,7 +1960,7 @@ memParseTest(const char *filename, const char *result,
     const char *base;
     int size, res;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     /*
      * load and parse the memory
      */
@@ -1999,7 +2007,7 @@ noentParseTest(const char *filename, const char *result,
     char *temp;
     int res = 0;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     /*
      * base of the test, parse with the old API
      */
@@ -2053,7 +2061,7 @@ errParseTest(const char *filename, const char *result, const char *err,
     const char *base = NULL;
     int size, res = 0;

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
 #ifdef LIBXML_HTML_ENABLED
     if (options & XML_PARSE_HTML) {
         doc = htmlReadFile(filename, NULL, options);
@@ -2150,7 +2158,7 @@ streamProcessTest(const char *filename, const char *result, const char *err,
     if (reader == NULL)
         return(-1);

-    nb_tests++;
+    nb_tests++;if(nb_tests != __dpp_test_num) return 0;
     if (result != NULL) {
 	temp = resultFilename(filename, "", ".res");
 	if (temp == NULL) {
@@ -4288,6 +4296,7 @@ testDesc testDescriptions[] = {
       NULL, XML_PARSE_XINCLUDE | XML_PARSE_NOXINCNODE },
 #endif
 #endif
+/*
 #ifdef LIBXML_XPATH_ENABLED
 #ifdef LIBXML_DEBUG_ENABLED
     { "XPath expressions regression tests" ,
@@ -4354,6 +4363,7 @@ testDesc testDescriptions[] = {
       threadsTest, NULL, NULL, NULL, NULL,
       0 },
 #endif
+*/
     {NULL, NULL, NULL, NULL, NULL, NULL, 0}
 };

@@ -4482,6 +4492,8 @@ main(int argc ATTRIBUTE_UNUSED, char **argv ATTRIBUTE_UNUSED) {
     int subset = 0;

     initializeLibxml2();
+
+    __init_dpp_test_num();

     for (a = 1; a < argc;a++) {
         if (!strcmp(argv[a], "-v"))
--
2.17.1

