From 05583b27f96aa8cd1980c991c3abfab591f7f5ee Mon Sep 17 00:00:00 2001
From: Hansol Choe <hansol614@gmail.com>
Date: Wed, 16 Feb 2022 08:17:39 +0000
Subject: [PATCH] buggy

---
 parser.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/parser.c b/parser.c
index 4464e2e8..68e1c90f 100644
--- a/parser.c
+++ b/parser.c
@@ -9825,7 +9825,6 @@ static void
 xmlParseEndTag2(xmlParserCtxtPtr ctxt, const xmlChar *prefix,
                 const xmlChar *URI, int line, int nsNr, int tlen) {
     const xmlChar *name;
-    size_t curLength;

     GROW;
     if ((RAW != '<') || (NXT(1) != '/')) {
@@ -9834,11 +9833,8 @@ xmlParseEndTag2(xmlParserCtxtPtr ctxt, const xmlChar *prefix,
     }
     SKIP(2);

-    curLength = ctxt->input->end - ctxt->input->cur;
-    if ((tlen > 0) && (curLength >= (size_t)tlen) &&
-        (xmlStrncmp(ctxt->input->cur, ctxt->name, tlen) == 0)) {
-        if ((curLength >= (size_t)(tlen + 1)) &&
-	    (ctxt->input->cur[tlen] == '>')) {
+    if ((tlen > 0) && (xmlStrncmp(ctxt->input->cur, ctxt->name, tlen) == 0)) {
+        if (ctxt->input->cur[tlen] == '>') {
 	    ctxt->input->cur += tlen + 1;
 	    ctxt->input->col += tlen + 1;
 	    goto done;
--
2.17.1

