From b9e227a2265d8660e7040754bad5bf51cdbdd543 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Fri, 5 Aug 2022 12:26:24 +0900
Subject: [PATCH] common

---
 runtest.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/runtest.c b/runtest.c
index bb74d2a8..f92fc2e3 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4377,7 +4377,20 @@ launchTests(testDescPtr tst) {

 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+            if (strstr(globbuf.gl_pathv[i], "test/HTML/758605.html") == NULL &&
+                strstr(globbuf.gl_pathv[i], "test/errors/attr2.xml") == NULL &&
+                strstr(globbuf.gl_pathv[i], "test/errors/name2.xml") == NULL)
+                continue;
+        } else {
+            if (strstr(globbuf.gl_pathv[i], "test/HTML/758605.html") ||
+                strstr(globbuf.gl_pathv[i], "test/errors/attr2.xml") ||
+                strstr(globbuf.gl_pathv[i], "test/errors/name2.xml")
+            )
+                continue;
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
--
2.25.1

