From 26f393dd2e9bf05b9af1879d390cd71623a77285 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 09:18:00 +0900
Subject: [PATCH] common

---
 runtest.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/runtest.c b/runtest.c
index 36fbe5ac..b0423ff0 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4377,7 +4377,15 @@ launchTests(testDescPtr tst) {
 
 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+            if (strstr(globbuf.gl_pathv[i], "test/HTML/758605.html") == NULL)
+                continue;
+        } else {
+            if (strstr(globbuf.gl_pathv[i], "test/HTML/758605.html"))
+                continue;
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
-- 
2.25.1

