From 69b1895435f56b6f885f066f88879886df3fe6c3 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 17 Feb 2022 06:11:23 +0000
Subject: [PATCH] common

---
 runtest.c | 28 ++++++++++++++++++++--------
 1 file changed, 20 insertions(+), 8 deletions(-)

diff --git a/runtest.c b/runtest.c
index 0f178cb0..2c9cec3a 100644
--- a/runtest.c
+++ b/runtest.c
@@ -13,7 +13,15 @@

 #include "libxml.h"
 #include <stdio.h>
-
+#include <stdlib.h>
+int __dpp_index = 1;
+int __dpp_test_num = 0;
+void __init_dpp_test_num() {
+    char *end = 0;
+    char *index = getenv("index");
+    __dpp_test_num = (int)(index == 0 ? 0 : strtol(index, &end, 0));
+    if(__dpp_test_num < 0) __dpp_test_num = 0;
+}
 #if !defined(_WIN32) || defined(__CYGWIN__)
 #include <unistd.h>
 #endif
@@ -1699,7 +1707,7 @@ saxParseTest(const char *filename, const char *result,
     int ret;
     char *temp;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     temp = resultFilename(filename, temp_directory, ".res");
     if (temp == NULL) {
         fprintf(stderr, "out of memory\n");
@@ -1808,7 +1816,7 @@ oldParseTest(const char *filename, const char *result,
     char *temp;
     int res = 0;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     /*
      * base of the test, parse with the old API
      */
@@ -1876,7 +1884,7 @@ pushParseTest(const char *filename, const char *result,
     int cur = 0;
     int chunkSize = 4;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     /*
      * load the document in memory and work from there.
      */
@@ -1977,7 +1985,7 @@ memParseTest(const char *filename, const char *result,
     const char *base;
     int size, res;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     /*
      * load and parse the memory
      */
@@ -2024,7 +2032,7 @@ noentParseTest(const char *filename, const char *result,
     char *temp;
     int res = 0;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     /*
      * base of the test, parse with the old API
      */
@@ -2078,7 +2086,7 @@ errParseTest(const char *filename, const char *result, const char *err,
     const char *base = NULL;
     int size, res = 0;

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
 #ifdef LIBXML_HTML_ENABLED
     if (options & XML_PARSE_HTML) {
         doc = htmlReadFile(filename, NULL, options);
@@ -2176,7 +2184,7 @@ streamProcessTest(const char *filename, const char *result, const char *err,
     if (reader == NULL)
         return(-1);

-    nb_tests++;
+    nb_tests++; if(nb_tests != __dpp_test_num) return 0;
     if (result != NULL) {
 	temp = resultFilename(filename, temp_directory, ".res");
 	if (temp == NULL) {
@@ -4342,6 +4350,7 @@ testDesc testDescriptions[] = {
       NULL, XML_PARSE_XINCLUDE | XML_PARSE_NOXINCNODE },
 #endif
 #endif
+/*
 #ifdef LIBXML_XPATH_ENABLED
 #ifdef LIBXML_DEBUG_ENABLED
     { "XPath expressions regression tests" ,
@@ -4408,6 +4417,7 @@ testDesc testDescriptions[] = {
       threadsTest, NULL, NULL, NULL, NULL,
       0 },
 #endif
+*/
     {NULL, NULL, NULL, NULL, NULL, NULL, 0}
 };

@@ -4559,6 +4569,8 @@ main(int argc ATTRIBUTE_UNUSED, char **argv ATTRIBUTE_UNUSED) {

     initializeLibxml2();

+    __init_dpp_test_num();
+
     for (a = 1; a < argc;a++) {
         if (!strcmp(argv[a], "-v"))
 	    verbose = 1;
--
2.17.1

