From 61556574b4adc8aa4c2c4eae9a33c023dfa897f5 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 10:28:06 +0900
Subject: [PATCH] common

---
 runtest.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/runtest.c b/runtest.c
index bb74d2a8..ac4c05d0 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4377,7 +4377,17 @@ launchTests(testDescPtr tst) {

 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (tst->func == errParseTest) {
+            if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+                if (strstr(globbuf.gl_pathv[i], "test/errors/759573-2.xml") == NULL)
+                    continue;
+            } else {
+                if (strstr(globbuf.gl_pathv[i], "test/errors/759573-2.xml"))
+                    continue;
+            }
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
--
2.25.1

