From 878e5e889af421b545589f818bd6975a627886ca Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 17 Feb 2022 06:06:19 +0000
Subject: [PATCH] buggy

---
 xinclude.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xinclude.c b/xinclude.c
index c92b32bc..aac30d5b 100644
--- a/xinclude.c
+++ b/xinclude.c
@@ -1464,7 +1464,7 @@ xmlXIncludeLoadDoc(xmlXIncludeCtxtPtr ctxt, const xmlChar *url, int nr) {
      */
     if ((URL[0] == 0) || (URL[0] == '#') ||
 	((ctxt->doc != NULL) && (xmlStrEqual(URL, ctxt->doc->URL)))) {
-	doc = ctxt->doc;
+	doc = NULL;
         goto loaded;
     }

--
2.17.1

