From 7a6112afa662687efa7470c5d0b81eb4c5a1fbcb Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Tue, 2 Aug 2022 09:37:21 +0900
Subject: [PATCH] buggy

---
 parser.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/parser.c b/parser.c
index a0183f24..a923191c 100644
--- a/parser.c
+++ b/parser.c
@@ -6933,9 +6933,7 @@ xmlParseReference(xmlParserCtxtPtr ctxt) {
      * far more secure as the parser will only process data coming from
      * the document entity by default.
      */
-    if ((ent->checked == 0) &&
-        ((ent->etype != XML_EXTERNAL_GENERAL_PARSED_ENTITY) ||
-         (ctxt->options & (XML_PARSE_NOENT | XML_PARSE_DTDVALID)))) {
+    if (ent->checked == 0) {
 	unsigned long oldnbent = ctxt->nbentities;
 
 	/*
-- 
2.25.1

