From 6bd90e79129a84b83e613f21a415c9cd8af7615f Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 14:33:31 +0900
Subject: [PATCH] buggy

---
 parser.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/parser.c b/parser.c
index a175ac4e..70e88049 100644
--- a/parser.c
+++ b/parser.c
@@ -3353,7 +3353,11 @@ xmlParseNameComplex(xmlParserCtxtPtr ctxt) {
 	    return(NULL);
 	}
 	len += l;
-	NEXTL(l);
+	if (*(ctxt->input->cur) == '\n') {
+	ctxt->input->line++; ctxt->input->col = 1;
+    } else ctxt->input->col++;
+    ctxt->input->cur += l;
+	if (*ctxt->input->cur == '%') xmlParserHandlePEReference(ctxt);
 	c = CUR_CHAR(l);
 	while ((c != ' ') && (c != '>') && (c != '/') && /* accelerators */
 	       (((c >= 'a') && (c <= 'z')) ||
@@ -3930,7 +3934,11 @@ xmlParseEntityValue(xmlParserCtxtPtr ctxt, xmlChar **orig) {
 	    buf = tmp;
 	}
 	COPY_BUF(l,buf,len,c);
-	NEXTL(l);
+	if (*(ctxt->input->cur) == '\n') {
+	ctxt->input->line++; ctxt->input->col = 1;
+    } else ctxt->input->col++;
+    ctxt->input->cur += l;
+	if (*ctxt->input->cur == '%') xmlParserHandlePEReference(ctxt);
 	/*
 	 * Pop-up of finished entities.
 	 */
@@ -4365,7 +4373,11 @@ xmlParseSystemLiteral(xmlParserCtxtPtr ctxt) {
             }
 	}
 	COPY_BUF(l,buf,len,cur);
-	NEXTL(l);
+	if (*(ctxt->input->cur) == '\n') {
+	ctxt->input->line++; ctxt->input->col = 1;
+    } else ctxt->input->col++;
+    ctxt->input->cur += l;
+	if (*ctxt->input->cur == '%') xmlParserHandlePEReference(ctxt);
 	cur = CUR_CHAR(l);
 	if (cur == 0) {
 	    GROW;
--
2.25.1

