From 760b67f9e7c3a2077ceee4a3e75a8ea17a03acbe Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 11:39:28 +0900
Subject: [PATCH] buggy

---
 parser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/parser.c b/parser.c
index a175ac4e..1e8ff240 100644
--- a/parser.c
+++ b/parser.c
@@ -2121,6 +2121,7 @@ static void xmlGROW (xmlParserCtxtPtr ctxt) {
 	ctxt->input->line++; ctxt->input->col = 1;			\
     } else ctxt->input->col++;						\
     ctxt->input->cur += l;				\
+	if (*ctxt->input->cur == '%') xmlParserHandlePEReference(ctxt); \
   } while (0)
 
 #define CUR_CHAR(l) xmlCurrentChar(ctxt, &l)
-- 
2.25.1

