From be7f14a1b1dad0589a1e23b31f82f76dd5f0eec2 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Tue, 2 Aug 2022 15:04:44 +0900
Subject: [PATCH] common

---
 runtest.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/runtest.c b/runtest.c
index 02fe09a6..592e2b18 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4331,7 +4331,22 @@ launchTests(testDescPtr tst) {

 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+            if (strstr(globbuf.gl_pathv[i], "test/errors/754946.xml") == NULL &&
+                strstr(globbuf.gl_pathv[i], "test/errors/content1.xml") == NULL &&
+                strstr(globbuf.gl_pathv[i], "test/valid/t8.xml") == NULL &&
+                strstr(globbuf.gl_pathv[i], "test/valid/t8a.xml") == NULL)
+                continue;
+        } else {
+            if (strstr(globbuf.gl_pathv[i], "test/errors/754946.xml") ||
+                strstr(globbuf.gl_pathv[i], "test/errors/content1.xml") ||
+                strstr(globbuf.gl_pathv[i], "test/valid/t8.xml") ||
+                strstr(globbuf.gl_pathv[i], "test/valid/t8a.xml")
+            )
+                continue;
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
--
2.25.1

