From b616a29f495f83535c789be153386fd5e74292fa Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Fri, 10 Sep 2021 12:17:50 +0900
Subject: [PATCH] common

---
 test/suite_decryption.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/suite_decryption.py b/test/suite_decryption.py
index 37f540cbd9..5bb83f843e 100644
--- a/test/suite_decryption.py
+++ b/test/suite_decryption.py
@@ -1243,6 +1243,7 @@ def softhsm(softhsm_paths, home_path, base_env):
 class case_decrypt_pkcs11(subprocesstest.SubprocessTestCase):
     def test_tls_pkcs11(self, cmd_tshark, dirs, capture_file, features, softhsm):
         '''Check that a RSA key in a PKCS #11 token enables decryption.'''
+        self.skipTest('defects4cpp')
         if not features.have_pkcs11:
             self.skipTest('Requires GnuTLS with PKCS #11 support.')
         key_file = os.path.join(dirs.key_dir, 'rsa-p-lt-q.p8')
--
2.25.1

