From 7c3cf6140e2e4d9c6f43c38757ff24937f7d8746 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 11:51:47 +0900
Subject: [PATCH] buggy

---
 epan/wslua/wslua_tvb.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/epan/wslua/wslua_tvb.c b/epan/wslua/wslua_tvb.c
index 4647d7d1e3..4b91a3ffdb 100644
--- a/epan/wslua/wslua_tvb.c
+++ b/epan/wslua/wslua_tvb.c
@@ -1399,13 +1399,9 @@ WSLUA_METAMETHOD TvbRange__tostring(lua_State* L) {
         return 0;
     }

-    if (tvbr->len == 0) {
-        lua_pushstring(L, "<EMPTY>");
-    } else {
-        str = tvb_bytes_to_str(NULL,tvbr->tvb->ws_tvb,tvbr->offset,tvbr->len);
-        lua_pushstring(L,str);
-        wmem_free(NULL, str);
-    }
+    str = tvb_bytes_to_str(NULL,tvbr->tvb->ws_tvb,tvbr->offset,tvbr->len);
+    lua_pushstring(L,str);
+    wmem_free(NULL, str);

     WSLUA_RETURN(1); /* A Lua hex string of the first 24 binary bytes in the `TvbRange`. */
 }
--
2.25.1

