From 67cc62dffcf8904ee85ff5cd8b6b93a1c4eac475 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 11:53:30 +0900
Subject: [PATCH] buggy

---
 epan/print.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/epan/print.c b/epan/print.c
index f968025a8f..14cf6dc21c 100644
--- a/epan/print.c
+++ b/epan/print.c
@@ -1367,10 +1367,8 @@ ek_write_attr(GSList *attr_instances, write_json_data *pdata)
                 && !ek_check_protocolfilter(pdata->filter, fi->hfinfo->abbrev)) {
 
                 /* print dummy field */
-                json_dumper_begin_object(pdata->dumper);
                 json_dumper_set_member_name(pdata->dumper, "filtered");
                 json_dumper_value_string(pdata->dumper, fi->hfinfo->abbrev);
-                json_dumper_end_object(pdata->dumper);
             } else {
                 ek_write_field_value(fi, pdata);
             }
-- 
2.25.1

