From 4bcb9109c0f1167fd31e31bb4238c7ba038d76db Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 11:53:57 +0900
Subject: [PATCH] buggy

---
 epan/wslua/wslua_byte_array.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/epan/wslua/wslua_byte_array.c b/epan/wslua/wslua_byte_array.c
index 5a5455b463..96c4650994 100644
--- a/epan/wslua/wslua_byte_array.c
+++ b/epan/wslua/wslua_byte_array.c
@@ -259,20 +259,14 @@ WSLUA_METHOD ByteArray_base64_decode(lua_State* L) {
     ByteArray ba = checkByteArray(L,1);
     ByteArray ba2;
     gchar *data;
-    gsize len = ba->len;

-    if ((len % 4) != 0) {
-        len += 4 - (len % 4);
-    }
+    gsize len;

     ba2 = g_byte_array_new();
     if (ba->len > 1) {
-        data = (gchar*)g_malloc(len + 1);
+        data = (gchar*)g_malloc(ba->len + 1);
         memcpy(data, ba->data, ba->len);
-        if (len > ba->len) {
-            memcpy(data + ba->len, "====", len - ba->len);
-        }
-        data[len] = '\0';
+        data[ba->len] = '\0';

         g_base64_decode_inplace(data, &len);
         g_byte_array_append(ba2, data, (int)len);
--
2.25.1

