From 1e24b089f4b8fa459271d876c58ac0f037ae71ba Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 22 Jul 2022 09:36:47 +0900
Subject: [PATCH] buggy

---
 src/jpgimage.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/jpgimage.cpp b/src/jpgimage.cpp
index f35e6250..42f2b7d3 100644
--- a/src/jpgimage.cpp
+++ b/src/jpgimage.cpp
@@ -637,8 +637,9 @@ namespace Exiv2 {
                 // Read size and signature
                 std::memset(buf.pData_, 0x0, buf.size_);
                 bufRead = io_->read(buf.pData_, bufMinSize);
-                if (io_->error() || bufRead != bufMinSize)
+                if (io_->error())
                     throw Error(kerFailedToReadImageData);
+                if (bufRead != bufMinSize) exit(1);
                 const uint16_t size = mHasLength[marker] ? getUShort(buf.pData_, bigEndian) : 0;
                 if (bPrint && mHasLength[marker])
                     out << Internal::stringFormat(" | %7d ", size);
-- 
2.25.1

