From f3e420b0d51cfbe92857c2ae24311d46c21025bc Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Fri, 22 Jul 2022 11:00:04 +0900
Subject: [PATCH] buggy

---
 src/crwimage_int.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/crwimage_int.cpp b/src/crwimage_int.cpp
index a230200c..8b118670 100644
--- a/src/crwimage_int.cpp
+++ b/src/crwimage_int.cpp
@@ -1254,12 +1254,10 @@ namespace Exiv2 {
         for (ExifData::const_iterator i = b; i != e; ++i) {
             if (i->ifdId() != ifdId) continue;
             const uint16_t s = i->tag()*2 + static_cast<uint16_t>(i->size());
-            if (s <= size) {
-                if (len < s) len = s;
-                i->copy(buf.pData_ + i->tag()*2, byteOrder);
-            } else {
-                EXV_ERROR << "packIfdId out-of-bounds error: s = " << std::dec << s << "\n";
-            }
+            assert(s <= size);
+            if (len < s) len = s;
+            i->copy(buf.pData_ + i->tag()*2, byteOrder);
+            
         }
         // Round the size to make it even.
         buf.size_ = len + len%2;
-- 
2.25.1

