From 247f803ebdc49cec0b4fca5bafa82416dba011fd Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 22 Jul 2022 11:19:08 +0900
Subject: [PATCH] buggy

---
 src/jp2image.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/jp2image.cpp b/src/jp2image.cpp
index 3bf35662..b6a38854 100644
--- a/src/jp2image.cpp
+++ b/src/jp2image.cpp
@@ -656,14 +656,12 @@ static void boxes_check(size_t b,size_t m)
         char*         p      = (char*) boxBuf.pData_;
         bool          bWroteColor = false ;
 
-        while ( count < length && !bWroteColor ) {
+        while ( count < length || !bWroteColor ) {
             enforce(sizeof(Jp2BoxHeader) <= length - count, Exiv2::kerCorruptedMetadata);
             Jp2BoxHeader* pSubBox = (Jp2BoxHeader*) (p+count) ;
 
             // copy data.  pointer could be into a memory mapped file which we will decode!
-            // pSubBox isn't always an aligned pointer, so use memcpy to do the copy.
-            Jp2BoxHeader   subBox;
-            memcpy(&subBox, pSubBox, sizeof(Jp2BoxHeader));
+            Jp2BoxHeader   subBox = *pSubBox ;
             Jp2BoxHeader   newBox =  subBox;
 
             if ( count < length ) {
-- 
2.25.1

