From 8b08a9e84091915df3b1969992eeb21c12e4c176 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Mon, 25 Jul 2022 09:25:24 +0900
Subject: [PATCH] buggy

---
 src/nikonmn_int.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/nikonmn_int.cpp b/src/nikonmn_int.cpp
index 2c5f3929..f92cc65e 100644
--- a/src/nikonmn_int.cpp
+++ b/src/nikonmn_int.cpp
@@ -370,12 +370,8 @@ namespace Exiv2 {
                                                const ExifData*)
     {
         if (value.count() >= 1) {
-            const unsigned long focusArea = value.toLong(0);
-            if (focusArea >= EXV_COUNTOF(nikonFocusarea)) {
-                os << "Invalid value";
-            } else {
-                os << nikonFocusarea[focusArea];
-            }
+            unsigned long focusArea = value.toLong(0);
+            os << nikonFocusarea[focusArea];
         }
         if (value.count() >= 2) {
             os << "; ";
-- 
2.25.1

