From fd5474d8db6000a1fce8ba4fdb3e1d1e77f6e052 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Mon, 25 Jul 2022 12:26:39 +0900
Subject: [PATCH] buggy

---
 src/pngimage.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/pngimage.cpp b/src/pngimage.cpp
index 4ccec39d..23162d95 100644
--- a/src/pngimage.cpp
+++ b/src/pngimage.cpp
@@ -33,7 +33,6 @@
 #include "image_int.hpp"
 #include "basicio.hpp"
 #include "error.hpp"
-#include "enforce.hpp"
 #include "futils.hpp"
 #include "types.hpp"
 
@@ -469,7 +468,6 @@ namespace Exiv2 {
                     }
                     profileName_ = std::string(reinterpret_cast<char *>(chunkData.pData_), iccOffset-1);
                     ++iccOffset; // +1 = 'compressed' flag
-                    enforce(iccOffset <= chunkLength, Exiv2::kerCorruptedMetadata);
 
                     zlibToDataBuf(chunkData.pData_ + iccOffset, chunkLength - iccOffset, iccProfile_);
 #ifdef DEBUG
-- 
2.25.1

