From 19631599697e9a0244d82980e3e2afc0ef4a28d2 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Thu, 28 Jul 2022 11:53:49 +0900
Subject: [PATCH] buggy

---
 src/pngimage.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/pngimage.cpp b/src/pngimage.cpp
index 21d00508..60e1bf98 100644
--- a/src/pngimage.cpp
+++ b/src/pngimage.cpp
@@ -469,9 +469,11 @@ namespace Exiv2
                 } else if (chunkType == "iCCP") {
                     // The ICC profile name can vary from 1-79 characters.
                     uint32_t iccOffset = 0;
-                    do {
-                      enforce(iccOffset < 80 && iccOffset < chunkLength, Exiv2::kerCorruptedMetadata);
-                    } while(chunkData.pData_[iccOffset++] != 0x00);
+                    while (iccOffset < 80 && iccOffset < chunkLength) {
+                        if (chunkData.pData_[iccOffset++] == 0x00) {
+                            break;
+                        }
+                    }

                     profileName_ = std::string(reinterpret_cast<char *>(chunkData.pData_), iccOffset-1);
                     ++iccOffset; // +1 = 'compressed' flag
--
2.25.1

