From c68a4aea4849a50e6fd294d4821c1cc0c143ddaa Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 08:39:16 +0900
Subject: [PATCH] buggy

---
 src/iptc.cpp | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/iptc.cpp b/src/iptc.cpp
index a3bb6e1d..a00e489e 100644
--- a/src/iptc.cpp
+++ b/src/iptc.cpp
@@ -439,7 +439,6 @@ namespace Exiv2 {
         std::cerr << "IptcParser::decode, size = " << size << "\n";
 #endif
         const byte* pRead = pData;
-        const byte* const pEnd = pData + size;
         iptcData.clear();
 
         uint16_t record = 0;
@@ -447,7 +446,7 @@ namespace Exiv2 {
         uint32_t sizeData = 0;
         byte extTest = 0;
 
-        while (6 <= static_cast<size_t>(pEnd - pRead)) {
+        while (pRead + 3 < pData + size) {
             // First byte should be a marker. If it isn't, scan forward and skip
             // the chunk bytes present in some images. This deviates from the
             // standard, which advises to treat such cases as errors.
@@ -461,7 +460,6 @@ namespace Exiv2 {
                 uint16_t sizeOfSize = (getUShort(pRead, bigEndian) & 0x7FFF);
                 if (sizeOfSize > 4) return 5;
                 pRead += 2;
-                if (sizeOfSize > static_cast<size_t>(pEnd - pRead)) return 6;
                 sizeData = 0;
                 for (; sizeOfSize > 0; --sizeOfSize) {
                     sizeData |= *pRead++ << (8 *(sizeOfSize-1));
@@ -472,7 +470,7 @@ namespace Exiv2 {
                 sizeData = getUShort(pRead, bigEndian);
                 pRead += 2;
             }
-            if (sizeData <= static_cast<size_t>(pEnd - pRead)) {
+            if (pRead + sizeData <= pData + size) {
                 int rc = 0;
                 if ((rc = readData(iptcData, dataSet, record, pRead, sizeData)) != 0) {
 #ifndef SUPPRESS_WARNINGS
@@ -486,7 +484,6 @@ namespace Exiv2 {
             else {
                 EXV_WARNING << "IPTC dataset " << IptcKey(dataSet, record)
                             << " has invalid size " << sizeData << "; skipped.\n";
-                return 7;
             }
 #endif
             pRead += sizeData;
-- 
2.25.1

