From 35ffb112149dadb02fd447df41b3eef7ca484c9d Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 15:33:12 +0900
Subject: [PATCH] buggy

---
 src/tiffimage_int.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/tiffimage_int.cpp b/src/tiffimage_int.cpp
index 99986145..e855c8b3 100644
--- a/src/tiffimage_int.cpp
+++ b/src/tiffimage_int.cpp
@@ -1695,8 +1695,10 @@ namespace Exiv2 {
             TiffFinder finder(0x00fe, imageGroups[i]);
             pSourceDir->accept(finder);
             TiffEntryBase* te = dynamic_cast<TiffEntryBase*>(finder.result());
-            const Value* pV = te != NULL ? te->pValue() : NULL;
-            if (pV && pV->typeId() == unsignedLong && pV->count() == 1 && (pV->toLong() & 1) == 0) {
+            if (   te
+                && te->pValue()->typeId() == unsignedLong
+                && te->pValue()->count() == 1
+                && (te->pValue()->toLong() & 1) == 0) {
                 primaryGroups.push_back(te->group());
             }
         }
-- 
2.25.1

