From 8c8d796a6d6b27ea46b8d018bb6056c7b09e96e2 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 15:45:46 +0900
Subject: [PATCH] common

---
 src/exiv2.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index df226f3e..19332748 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -122,11 +122,32 @@ namespace {
      */
     std::string parseEscapes(const std::string& input);
 }
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
 
 // *****************************************************************************
 // Main
 int main(int argc, char* const argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
 #ifdef EXV_ENABLE_NLS
     setlocale(LC_ALL, "");
     const std::string localeDir = Exiv2::getProcessPath() + EXV_LOCALEDIR;
-- 
2.25.1

