From fc71892c311a27fd99a5ac9d1d514ad3bd5b193d Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 09:23:06 +0900
Subject: [PATCH] buggy

---
 src/exif.cpp | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/src/exif.cpp b/src/exif.cpp
index bb7b69db..2eda4904 100644
--- a/src/exif.cpp
+++ b/src/exif.cpp
@@ -229,22 +229,7 @@ namespace Exiv2 {
               fct = nullptr;
             }
         }
-        if ( fct ) {
-          // https://github.com/Exiv2/exiv2/issues/1706
-          // Sometimes the type of the value doesn't match what the
-          // print function expects. (The expected types are stored
-          // in the TagInfo tables, but they are not enforced when the
-          // metadata is parsed.) These type mismatches can sometimes
-          // cause a std::out_of_range exception to be thrown.
-          try {
-            fct(os, value(), pMetadata);
-          } catch (std::out_of_range&) {
-            os << "Bad value";
-#ifdef EXIV2_DEBUG_MESSAGES
-            std::cerr << "Caught std::out_of_range exception in Exifdatum::write().\n";
-#endif
-          }
-        }
+        if ( fct ) fct(os, value(), pMetadata);
         return os;
     }
 
-- 
2.25.1

