From d96da1f081946e0b125e22f2a9990ab3d202ca94 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 09:35:02 +0900
Subject: [PATCH] common

---
 src/exiv2.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index 9f7c3995..0314f7c4 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -117,11 +117,32 @@ namespace {
      */
     std::string parseEscapes(const std::string& input);
 }  // namespace
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
 
 // *****************************************************************************
 // Main
 int main(int argc, char* const argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
     Exiv2::XmpParser::initialize();
     ::atexit(Exiv2::XmpParser::terminate);
 #ifdef EXV_ENABLE_BMFF
-- 
2.25.1

