From de544cdcb290bed793855d6fb026b656fdd398b9 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 10:06:08 +0900
Subject: [PATCH] buggy

---
 src/tiffvisitor_int.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/tiffvisitor_int.cpp b/src/tiffvisitor_int.cpp
index c4311098..80a45314 100644
--- a/src/tiffvisitor_int.cpp
+++ b/src/tiffvisitor_int.cpp
@@ -1599,7 +1599,6 @@ namespace Exiv2 {
             v->read(pData, size, byteOrder());
         } else {
             // Prevent large memory allocations: https://github.com/Exiv2/exiv2/issues/1881
-            enforce(isize <= 1024 * 1024, kerCorruptedMetadata);
 
             // #1143 Write a "hollow" buffer for the preview image
             //       Sadly: we don't know the exact location of the image in the source (it's near offset)
-- 
2.25.1

