From e4fe5befce62661d8f101e599d6d871f55816ffb Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 12:06:39 +0900
Subject: [PATCH] buggy

---
 src/wget.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/wget.c b/src/wget.c
index 50bdc16c..6fecc20c 100644
--- a/src/wget.c
+++ b/src/wget.c
@@ -2042,8 +2042,9 @@ static void process_response(wget_http_response *resp)
 			&& (!config.level || job->level < config.level + config.page_requisites) ? 1 : 0;
 		if (process_decision) {
 			wget_vector *recurse_iris = NULL;
+			int n_recurse_iris = 0;
 			const void *data = NULL;
-			int64_t size;
+			uint64_t size;
 			const char *filename;
 
 			if (config.spider || (config.recursive && config.output_document))
@@ -2056,7 +2057,7 @@ static void process_response(wget_http_response *resp)
 			else
 				size = resp->content_length;
 
-			if ((resp->code == 200 || resp->code == 206) && resp->body && (int64_t) resp->body->length == size)
+			if ((resp->code == 200 || resp->code == 206) && resp->body && resp->body->length == size)
 				data = resp->body->data;
 
 			if (recurse_decision)
@@ -2065,9 +2066,9 @@ static void process_response(wget_http_response *resp)
 			process_decision = plugin_db_forward_downloaded_file(job->iri, size > 0 ? size : 0, filename, data, recurse_iris);
 
 			if (recurse_decision) {
-				for (int i = 0; i < wget_vector_size(recurse_iris); i++) {
+				n_recurse_iris = wget_vector_size(recurse_iris);
+				for (int i = 0; i < n_recurse_iris; i++) {
 					wget_iri *iri = (wget_iri *) wget_vector_get(recurse_iris, i);
-
 					queue_url_from_remote(job, "utf-8", iri->uri, 0);
 					wget_iri_free_content(iri);
 				}
@@ -3304,7 +3305,7 @@ static int WGET_GCC_NONNULL((1)) prepare_file(wget_http_response *resp, const ch
 	// Load partial content
 	if (partial_content) {
 		long long size = get_file_size(unique[0] ? unique : fname);
-		if (size >= 0) {
+		if (size > 0) {
 			fd = open_unique(fname, O_RDONLY | O_BINARY, 0, multiple, unique, sizeof(unique));
 			if (fd >= 0) {
 				size_t rc;
@@ -3723,7 +3724,7 @@ static wget_http_request *http_create_request(const wget_iri *iri, JOB *job)
 
 		if (config.continue_download) {
 			long long file_size = get_file_size(local_filename);
-			if (file_size >= 0)
+			if (file_size > 0)
 				wget_http_add_header_printf(req, "Range", "bytes=%lld-", file_size);
 		}
 
-- 
2.25.1

