From aefd1ac8c562d875daa836ce882c905e0ec697cc Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Fri, 24 Sep 2021 14:24:10 +0900
Subject: [PATCH] common

---
 run-tests | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/run-tests b/run-tests
index 0601b564..efa149e0 100755
--- a/run-tests
+++ b/run-tests
@@ -9,7 +9,7 @@ NPROCS=1
 if [ -r /proc/cpuinfo ]; then
 	NPROCS=$(grep ^proc /proc/cpuinfo|wc -l)
 fi
-LIBRARY_PATH=$PWD/lib LD_LIBRARY_PATH=$PWD/lib ATF_SHELL=/bin/sh kyua --variable parallelism=$NPROCS test -r result.db -k tests/xbps/Kyuafile
+LIBRARY_PATH=$PWD/lib LD_LIBRARY_PATH=$PWD/lib ATF_SHELL=/bin/sh kyua --variable parallelism=$NPROCS test -r result.db -k tests/xbps/Kyuafile "$@"
 rv=$?
 kyua report --verbose -r result.db
 rm -f result.db
--
2.30.2

