From 0fbd0720d6036bbd38820b4b65228e1d2dd5c987 Mon Sep 17 00:00:00 2001
From: Minhyuk Kwon <minhyuk@live.co.kr>
Date: Fri, 10 Sep 2021 12:00:16 +0900
Subject: [PATCH] buggy

---
 lib/package_orphans.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/package_orphans.c b/lib/package_orphans.c
index f7e2d941..f5352a9a 100644
--- a/lib/package_orphans.c
+++ b/lib/package_orphans.c
@@ -156,6 +156,7 @@ xbps_find_pkg_orphans(struct xbps_handle *xhp, xbps_array_t orphans_user)
 		xbps_dictionary_t pkgd;
 		const char *pkgver = NULL;
 		unsigned int cnt = 0, reqbycnt = 0;
+		bool automatic = false;
 
 		pkgd = xbps_array_get(array, i);
 		xbps_dictionary_get_cstring_nocopy(pkgd, "pkgver", &pkgver);
@@ -169,7 +170,6 @@ xbps_find_pkg_orphans(struct xbps_handle *xhp, xbps_array_t orphans_user)
 			xbps_array_t reqby;
 			xbps_dictionary_t deppkgd;
 			const char *deppkgver = NULL;
-			bool automatic = false;
 
 			cnt = 0;
 			xbps_array_get_cstring_nocopy(rdeps, x, &deppkgver);
-- 
2.31.1.windows.1

