From 639e46badf51ae40574bde5195cc9ca24bbf08b5 Mon Sep 17 00:00:00 2001
From: Minhyuk Kwon <minhyuk@live.co.kr>
Date: Fri, 10 Sep 2021 12:10:08 +0900
Subject: [PATCH] buggy

---
 lib/transaction_revdeps.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/transaction_revdeps.c b/lib/transaction_revdeps.c
index 73edd21c..b2a2286b 100644
--- a/lib/transaction_revdeps.c
+++ b/lib/transaction_revdeps.c
@@ -238,10 +238,6 @@ xbps_transaction_revdeps(struct xbps_handle *xhp, xbps_array_t pkgs)
 			 * if a new version of this conflicting package
 			 * is in the transaction.
 			 */
-			if (xbps_find_pkg_in_array(pkgs, pkgname, "update")) {
-				free(pkgname);
-				continue;
-			}
 			free(pkgname);
 			broken_pkg(mdeps, curpkgver, pkgver, tract);
 		}
-- 
2.31.1.windows.1

