From 8f870f886d6372a9bac7fde2316a0437620486fe Mon Sep 17 00:00:00 2001
From: Hansol Choe <hansol614@gmail.com>
Date: Mon, 7 Mar 2022 07:13:05 +0000
Subject: [PATCH] buggy

---
 example.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/example.c b/example.c
index 30e92f8..c3a3f5e 100644
--- a/example.c
+++ b/example.c
@@ -1,7 +1,7 @@
 int func1(int x) {
     int y = 100;
     y = x;
-    if (x > 100) {
+    if (x >= 100) {
         return -y;
     }
     return y;
--
2.25.1

