From 5aaf3b4be31c1cb325d79cb5401bb5253dfc048b Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Thu, 4 Aug 2022 11:42:57 +0900
Subject: [PATCH] buggy

---
 src/lib/protocols/tls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/protocols/tls.c b/src/lib/protocols/tls.c
index c8a0e43b..6f9c7e6a 100644
--- a/src/lib/protocols/tls.c
+++ b/src/lib/protocols/tls.c
@@ -243,7 +243,7 @@ static void processCertificateElements(struct ndpi_detection_module_struct *ndpi
 				       u_int16_t p_offset, u_int16_t certificate_len) {
   struct ndpi_packet_struct *packet = &flow->packet;
   u_int num_found = 0, i;
-  char buffer[64] = { '\0' }, rdnSeqBuf[2048] = { '\0' };
+  char buffer[64] = { '\0' }, rdnSeqBuf[1024] = { '\0' };
   u_int rdn_len = 0;
 
 #ifdef DEBUG_TLS
-- 
2.25.1

