From c26f32958fcb3c354ac742e31c5453ed572394da Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Fri, 5 Aug 2022 11:00:22 +0900
Subject: [PATCH] buggy

---
 src/lib/ndpi_main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/lib/ndpi_main.c b/src/lib/ndpi_main.c
index 65600b74..21b40c01 100644
--- a/src/lib/ndpi_main.c
+++ b/src/lib/ndpi_main.c
@@ -4687,7 +4687,6 @@ ndpi_protocol ndpi_detection_process_packet(struct ndpi_detection_module_struct
        && (found->proto->protoId != ret.master_protocol)) {
       // printf("******** %u / %u\n", found->proto->protoId, ret.master_protocol);
 
-      if(!ndpi_check_protocol_port_mismatch_exceptions(ndpi_str, flow, found, &ret))
 	NDPI_SET_BIT(flow->risk, NDPI_KNOWN_PROTOCOL_ON_NON_STANDARD_PORT);
     } else if(default_ports && (default_ports[0] != 0)) {
       u_int8_t found = 0, i;
-- 
2.25.1

