From cd79adc229433840d2cf252c903b66f24839f88d Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 10 Aug 2022 10:03:02 +0900
Subject: [PATCH] buggy

---
 src/lib/protocols/http.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/lib/protocols/http.c b/src/lib/protocols/http.c
index 4b66a1da..19d4f671 100644
--- a/src/lib/protocols/http.c
+++ b/src/lib/protocols/http.c
@@ -414,11 +414,6 @@ static void ndpi_http_parse_subprotocol(struct ndpi_detection_module_struct *ndp
       }
     }
 
-    if (flow->http.url != NULL &&
-        strstr(flow->http.url, "micloud.xiaomi.net") != NULL)
-    {
-      ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_XIAOMI, NDPI_PROTOCOL_HTTP, NDPI_CONFIDENCE_DPI);
-    }
   }
 }
 
-- 
2.25.1

