From 0b0721354e1834e9f2cd3a7650db29263ea1f041 Mon Sep 17 00:00:00 2001
From: hschoe <hschoe@suresofttech.com>
Date: Thu, 16 Sep 2021 10:37:13 +0900
Subject: [PATCH] buggy

---
 Src/Zle/zle_keymap.c | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/Src/Zle/zle_keymap.c b/Src/Zle/zle_keymap.c
index d90838f03..2389ab754 100644
--- a/Src/Zle/zle_keymap.c
+++ b/Src/Zle/zle_keymap.c
@@ -155,7 +155,7 @@ createkeymapnamtab(void)
     keymapnamtab = newhashtable(7, "keymapnamtab", NULL);

     keymapnamtab->hash        = hasher;
-    keymapnamtab->emptytable  = emptykeymapnamtab;
+    keymapnamtab->emptytable  = emptyhashtable;
     keymapnamtab->filltable   = NULL;
     keymapnamtab->cmpnodes    = strcmp;
     keymapnamtab->addnode     = addhashnode;
@@ -178,26 +178,6 @@ makekeymapnamnode(Keymap keymap)
     return kmn;
 }

-/**/
-static void
-emptykeymapnamtab(HashTable ht)
-{
-    struct hashnode *hn, *hp;
-    int i;
-
-    for (i = 0; i < ht->hsize; i++) {
-	for (hn = ht->nodes[i]; hn;) {
-	    KeymapName kmn = (KeymapName) hn;
-	    hp = hn->next;
-	    zsfree(kmn->nam);
-	    unrefkeymap(kmn->keymap);
-	    zfree(kmn, sizeof(*kmn));
-	    hn = hp;
-	}
-	ht->nodes[i] = NULL;
-    }
-    ht->ct = 0;
-}

 /*
  * Reference a keymap from a keymapname.
--
2.17.1

