From c0f10d1a6152d9233aa7a57df7d80d39c7b67077 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 31 Aug 2022 09:51:03 +0900
Subject: [PATCH] buggy

---
 Src/Zle/zle_keymap.c | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/Src/Zle/zle_keymap.c b/Src/Zle/zle_keymap.c
index d90838f03..324c1013e 100644
--- a/Src/Zle/zle_keymap.c
+++ b/Src/Zle/zle_keymap.c
@@ -155,7 +155,7 @@ createkeymapnamtab(void)
     keymapnamtab = newhashtable(7, "keymapnamtab", NULL);
 
     keymapnamtab->hash        = hasher;
-    keymapnamtab->emptytable  = emptykeymapnamtab;
+    keymapnamtab->emptytable  = emptyhashtable;
     keymapnamtab->filltable   = NULL;
     keymapnamtab->cmpnodes    = strcmp;
     keymapnamtab->addnode     = addhashnode;
@@ -179,25 +179,6 @@ makekeymapnamnode(Keymap keymap)
 }
 
 /**/
-static void
-emptykeymapnamtab(HashTable ht)
-{
-    struct hashnode *hn, *hp;
-    int i;
-
-    for (i = 0; i < ht->hsize; i++) {
-	for (hn = ht->nodes[i]; hn;) {
-	    KeymapName kmn = (KeymapName) hn;
-	    hp = hn->next;
-	    zsfree(kmn->nam);
-	    unrefkeymap(kmn->keymap);
-	    zfree(kmn, sizeof(*kmn));
-	    hn = hp;
-	}
-	ht->nodes[i] = NULL;
-    }
-    ht->ct = 0;
-}
 
 /*
  * Reference a keymap from a keymapname.
-- 
2.25.1

