From 061bcf3cab0eb499476b7b5d5ee9c8b0038c0df6 Mon Sep 17 00:00:00 2001
From: hschoe <hschoe@suresofttech.com>
Date: Thu, 16 Sep 2021 11:02:05 +0900
Subject: [PATCH] buggy

---
 Src/subst.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/Src/subst.c b/Src/subst.c
index 465fe970f..87a56c3c6 100644
--- a/Src/subst.c
+++ b/Src/subst.c
@@ -3362,15 +3362,13 @@ colonsubscript:
 		    return NULL;
 		}
 		if (*check_offset2) {
-		    char *nextp;
 		    check_offset = check_colon_subscript(check_offset2 + 1,
-							 &nextp);
+							 &check_offset2);
+		    if (*check_offset2 && *check_offset2 != ':') {
+			zerr("invalid length: %s", check_offset);
+			return NULL;
+		    }
 		    if (check_offset) {
-			check_offset2 = nextp;
-			if (*check_offset2 && *check_offset2 != ':') {
-			    zerr("invalid length: %s", check_offset);
-			    return NULL;
-			}
 			length = mathevali(check_offset);
 			length_set = 1;
 			if (errflag)
--
2.17.1

