From 4c55386f2afc260bfe284aa15b9b1d099ffd896d Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresorttech.com>
Date: Thu, 23 Sep 2021 14:18:19 +0900
Subject: [PATCH] buggy

---
 Src/Modules/zutil.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Src/Modules/zutil.c b/Src/Modules/zutil.c
index 691ba6c2f..cecea6d51 100644
--- a/Src/Modules/zutil.c
+++ b/Src/Modules/zutil.c
@@ -200,8 +200,7 @@ printstylenode(HashNode hn, int printflags)
 	else {
 	    printf("zstyle %s", (p->eval ? "-e " : ""));
 	    quotedzputs(p->pat, stdout);
-	    putchar(' ');
-	    quotedzputs(s->node.nam, stdout);
+	    printf(" %s", s->node.nam);
 	}
 	for (v = p->vals; *v; v++) {
 	    putchar(' ');
-- 
2.17.1

