From 7d507057d564a673f0aca448567588ce7870418a Mon Sep 17 00:00:00 2001
From: Minhyuk Kwon <minhyuk@live.co.kr>
Date: Fri, 10 Sep 2021 11:21:19 +0900
Subject: [PATCH] buggy

---
 lib/transaction_files.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/lib/transaction_files.c b/lib/transaction_files.c
index ff865d89..fa2f1e3c 100644
--- a/lib/transaction_files.c
+++ b/lib/transaction_files.c
@@ -445,14 +445,6 @@ collect_file(struct xbps_handle *xhp, const char *file, size_t size,
 			 * Multiple packages removing the same directory.
 			 * Record the last package to remove this directory.
 			 */
-			if (idx < item->old.index || item->old.preserve)
-				return 0;
-			item->old.pkgname = pkgname;
-			item->old.pkgver = pkgver;
-			item->old.index = idx;
-			item->old.preserve = preserve;
-			item->old.update = update;
-			item->old.removepkg = removepkg;
 			return 0;
 		} else {
 			/*
--
2.31.1.windows.1

