From 69e9c7bc608016e926b781d1ba178ef96037aff0 Mon Sep 17 00:00:00 2001
From: Minhyuk Kwon <minhyuk@live.co.kr>
Date: Fri, 10 Sep 2021 11:54:56 +0900
Subject: [PATCH] buggy

---
 lib/repo.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/repo.c b/lib/repo.c
index f3f6ed4f..91fcfb7a 100644
--- a/lib/repo.c
+++ b/lib/repo.c
@@ -156,6 +156,7 @@ repo_open_local(struct xbps_repo *repo, const char *repofile)
 		xbps_dbg_printf(repo->xhp,
 		    "[repo] `%s' failed to open repodata archive %s\n",
 		    repofile, strerror(rv));
+		xbps_repo_close(repo);
 		return false;
 	}
 	if ((repo->idx = repo_get_dict(repo)) == NULL) {
@@ -163,6 +164,7 @@ repo_open_local(struct xbps_repo *repo, const char *repofile)
 		    " index on archive, removing file.\n", repofile);
 		/* broken archive, remove it */
 		(void)unlink(repofile);
+		xbps_repo_close(repo);
 		return false;
 	}
 	xbps_dictionary_make_immutable(repo->idx);
@@ -366,8 +368,7 @@ xbps_repo_open(struct xbps_handle *xhp, const char *url)
 void
 xbps_repo_close(struct xbps_repo *repo)
 {
-	if (!repo)
-		return;
+	assert(repo);
 
 	if (repo->ar != NULL)
 		archive_read_finish(repo->ar);
-- 
2.31.1.windows.1

