From a6789522883985e97ec0eb5535783f201936f259 Mon Sep 17 00:00:00 2001
From: Minhyuk Kwon <minhyuk@live.co.kr>
Date: Fri, 10 Sep 2021 11:58:17 +0900
Subject: [PATCH] buggy

---
 lib/transaction_check_revdeps.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/transaction_check_revdeps.c b/lib/transaction_check_revdeps.c
index 389daab2..976f81b0 100644
--- a/lib/transaction_check_revdeps.c
+++ b/lib/transaction_check_revdeps.c
@@ -174,8 +174,9 @@ xbps_transaction_check_revdeps(struct xbps_handle *xhp, xbps_array_t pkgs)
 			}

 			if ((revpkgd = xbps_find_pkg_in_array(pkgs, pkgname, 0))) {
-				if (xbps_transaction_pkg_type(revpkgd) == XBPS_TRANS_REMOVE)
-					continue;
+				ttype = xbps_transaction_pkg_type(revpkgd);
+				if (ttype == XBPS_TRANS_REMOVE)
+					revpkgd = NULL;
 			}
 			if (revpkgd == NULL)
 				revpkgd = xbps_pkgdb_get_pkg(xhp, curpkgver);
--
2.31.1.windows.1
