From 2f2c2c7fd2f8725ccfc8eb7fae83e3da416ab28b Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 30 Aug 2022 17:11:37 +0900
Subject: [PATCH] buggy

---
 HTMLparser.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/HTMLparser.c b/HTMLparser.c
index 1c112cc9..c9f0a9a3 100644
--- a/HTMLparser.c
+++ b/HTMLparser.c
@@ -2472,8 +2472,6 @@ htmlParseName(htmlParserCtxtPtr ctxt) {
 	       (*in == ':') || (*in == '.'))
 	    in++;

-	if (in == ctxt->input->end)
-	    return(NULL);

 	if ((*in > 0) && (*in < 0x80)) {
 	    count = in - ctxt->input->cur;
@@ -2519,8 +2517,6 @@ htmlParseNameComplex(xmlParserCtxtPtr ctxt) {
 	c = CUR_CHAR(l);
     }

-    if (ctxt->input->base > ctxt->input->cur - len)
-	return(NULL);

     return(xmlDictLookup(ctxt->dict, ctxt->input->cur - len, len));
 }
--
2.25.1

