From 5c566db21ca19dc174fe673c96a54ca8091134a2 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 10:24:46 +0900
Subject: [PATCH] buggy

---
 parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/parser.c b/parser.c
index ea0e89ea..7b2a125d 100644
--- a/parser.c
+++ b/parser.c
@@ -6708,7 +6708,6 @@ xmlParseElementDecl(xmlParserCtxtPtr ctxt) {
 	if (!IS_BLANK_CH(CUR)) {
 	    xmlFatalErrMsg(ctxt, XML_ERR_SPACE_REQUIRED,
 		           "Space required after 'ELEMENT'\n");
-	    return(-1);
 	}
         SKIP_BLANKS;
         name = xmlParseName(ctxt);
-- 
2.25.1

