From d100b9e458e55e74a10e3878e3fd131c1745152d Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 2 Aug 2022 11:41:01 +0900
Subject: [PATCH] common

---
 runtest.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/runtest.c b/runtest.c
index 378b38e3..6712c170 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4392,7 +4392,15 @@ launchTests(testDescPtr tst) {
 
 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+            if (strstr(globbuf.gl_pathv[i], "test/valid/766956.xml") == NULL)
+                continue;
+        } else {
+            if (strstr(globbuf.gl_pathv[i], "test/valid/766956.xml"))
+                continue;
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
-- 
2.25.1

