From 08fee8a0a6f2163a4aced54a0a2dc2f14731124e Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Tue, 2 Aug 2022 11:43:29 +0900
Subject: [PATCH] common

---
 runtest.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/runtest.c b/runtest.c
index 714de275..b1ec5cf4 100644
--- a/runtest.c
+++ b/runtest.c
@@ -4419,7 +4419,16 @@ launchTests(testDescPtr tst) {

 	globbuf.gl_offs = 0;
 	glob(tst->in, GLOB_DOOFFS, NULL, &globbuf);
+    const char * DPP_RUN_ERROR_CASE = getenv("DPP_RUN_ERROR_CASE");
 	for (i = 0;i < globbuf.gl_pathc;i++) {
+        if (DPP_RUN_ERROR_CASE && !strcmp(DPP_RUN_ERROR_CASE, "1")) {
+            if (strstr(globbuf.gl_pathv[i], "test/errors/759579.xml") == NULL)
+                continue;
+        } else {
+            if (strstr(globbuf.gl_pathv[i], "test/errors/759579.xml")
+            )
+                continue;
+        }
 	    if (!checkTestFile(globbuf.gl_pathv[i]))
 	        continue;
 	    if (tst->suffix != NULL) {
--
2.25.1

