From 8a45e386547a1a504a34fe35c44e2c83747d7ba3 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Tue, 2 Aug 2022 11:40:37 +0900
Subject: [PATCH] buggy

---
 parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/parser.c b/parser.c
index 51452a25..8d8dc239 100644
--- a/parser.c
+++ b/parser.c
@@ -2252,7 +2252,6 @@ xmlPushInput(xmlParserCtxtPtr ctxt, xmlParserInputPtr input) {
     }
     if (((ctxt->inputNr > 40) && ((ctxt->options & XML_PARSE_HUGE) == 0)) ||
         (ctxt->inputNr > 1024)) {
-        xmlFatalErr(ctxt, XML_ERR_ENTITY_LOOP, NULL);
         while (ctxt->inputNr > 1)
             xmlFreeInputStream(inputPop(ctxt));
 	return(-1);
--
2.25.1

