From 96865bbd25fcfc1d6bd4a63b7374fbdea4c48c86 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Tue, 2 Aug 2022 15:02:48 +0900
Subject: [PATCH] buggy

---
 parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/parser.c b/parser.c
index 0677030c..c5741e3b 100644
--- a/parser.c
+++ b/parser.c
@@ -8468,7 +8468,6 @@ xmlParseInternalSubset(xmlParserCtxtPtr ctxt) {
      */
     if (RAW != '>') {
 	xmlFatalErr(ctxt, XML_ERR_DOCTYPE_NOT_FINISHED, NULL);
-	return;
     }
     NEXT;
 }
-- 
2.25.1

