From ea4f453c4141ac4787a31a9ebd2f3dee0e70a407 Mon Sep 17 00:00:00 2001
From: Hansol Choe <hschoe@suresofttech.com>
Date: Wed, 20 Dec 2023 01:58:04 +0000
Subject: [PATCH] buggy

---
 src/runtime/gravity_vm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/runtime/gravity_vm.c b/src/runtime/gravity_vm.c
index c80a32a..ce159ed 100644
--- a/src/runtime/gravity_vm.c
+++ b/src/runtime/gravity_vm.c
@@ -1451,9 +1451,8 @@ static bool gravity_vm_exec (gravity_vm *vm) {
                     if (op != MOVE) RUNTIME_ERROR("Wrong OPCODE in CLOSURE statement");
                     closure->upvalue[i] = (p2) ? gravity_capture_upvalue (vm, fiber, &stackstart[p1]) : frame->closure->upvalue[p1];
                 }
-                
-                SETVALUE(r1, VALUE_FROM_OBJECT(closure));
                 gravity_gc_setenabled(vm, true);
+                SETVALUE(r1, VALUE_FROM_OBJECT(closure));
                 DISPATCH();
             }
 
-- 
2.25.1

