From 371b9d8000cc651f28ed60496b3cfc1abe39cfd2 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Mon, 11 Jul 2022 10:57:37 +0900
Subject: [PATCH] common

---
 test/testutil/main.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/test/testutil/main.c b/test/testutil/main.c
index 6716750a30..98ede0b11c 100644
--- a/test/testutil/main.c
+++ b/test/testutil/main.c
@@ -10,10 +10,26 @@
 #include "../testutil.h"
 #include "output.h"
 #include "tu_local.h"
+#include <signal.h>
 
 
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGABRT, &dpp_orig_sigaction, NULL);
+  raise(SIGABRT);
+}
 int main(int argc, char *argv[])
 {
+    {
+      dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+      sigemptyset(&dpp_gcov_sigaction.sa_mask);
+      dpp_gcov_sigaction.sa_flags = 0;
+      sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+    }
     int ret = EXIT_FAILURE;
 
     test_open_streams();
-- 
2.25.1

