From b3b0f9567801c73eb6a7f248aba9f153b676fc66 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:07:08 +0900
Subject: [PATCH] buggy

---
 crypto/dh/dh_group_params.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/dh/dh_group_params.c b/crypto/dh/dh_group_params.c
index c71f4053da..ff6d7cdd66 100644
--- a/crypto/dh/dh_group_params.c
+++ b/crypto/dh/dh_group_params.c
@@ -34,6 +34,7 @@ static DH *dh_param_init(OSSL_LIB_CTX *libctx, const DH_NAMED_GROUP *group)
 
     ossl_ffc_named_group_set_pqg(&dh->params, group);
     dh->params.nid = ossl_ffc_named_group_get_uid(group);
+    dh->length = BN_num_bits(dh->params.q);
     dh->dirty_cnt++;
     return dh;
 }
@@ -75,6 +76,7 @@ void ossl_dh_cache_named_group(DH *dh)
             dh->params.q = (BIGNUM *)ossl_ffc_named_group_get_q(group);
         /* cache the nid */
         dh->params.nid = ossl_ffc_named_group_get_uid(group);
+        dh->length = BN_num_bits(dh->params.q);
         dh->dirty_cnt++;
     }
 }
-- 
2.23.0.windows.1

