From 1f3f6cff4040e7572353965b1f1a72f48c970757 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Wed, 15 Sep 2021 09:23:53 +0900
Subject: [PATCH] buggy

---
 crypto/rsa/rsa_sp800_56b_check.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/crypto/rsa/rsa_sp800_56b_check.c b/crypto/rsa/rsa_sp800_56b_check.c
index fc8f19b487..9062442a3c 100644
--- a/crypto/rsa/rsa_sp800_56b_check.c
+++ b/crypto/rsa/rsa_sp800_56b_check.c
@@ -222,17 +222,25 @@ int ossl_rsa_check_private_exponent(const RSA *rsa, int nbits, BN_CTX *ctx)
  * Check exponent is odd.
  * For FIPS also check the bit length is in the range [17..256]
  */
+#ifndef FIPS_MODULE
+static int bn_is_three(const BIGNUM *bn)
+{
+    BIGNUM *num = BN_dup(bn);
+    int ret = (num != NULL && BN_sub_word(num, 3) && BN_is_zero(num));
+    BN_free(num);
+    return ret;
+}
+#endif /* FIPS_MODULE */
 int ossl_rsa_check_public_exponent(const BIGNUM *e)
 {
-#ifdef FIPS_MODULE
     int bitlen;

+#ifndef FIPS_MODULE
+    if (bn_is_three(e))
+        return 1;
+#endif /* FIPS_MODULE */
     bitlen = BN_num_bits(e);
     return (BN_is_odd(e) && bitlen > 16 && bitlen < 257);
-#else
-    /* Allow small exponents larger than 1 for legacy purposes */
-    return BN_is_odd(e) && BN_cmp(e, BN_value_one()) > 0;
-#endif /* FIPS_MODULE */
 }

 /*
--
2.23.0.windows.1

