From 9a4277b12bc54a659c572fd93eb97ef27da6fbc4 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Wed, 15 Sep 2021 09:33:57 +0900
Subject: [PATCH] buggy

---
 crypto/evp/ctrl_params_translate.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/crypto/evp/ctrl_params_translate.c b/crypto/evp/ctrl_params_translate.c
index 65414f6ede..976082d418 100644
--- a/crypto/evp/ctrl_params_translate.c
+++ b/crypto/evp/ctrl_params_translate.c
@@ -2097,9 +2097,6 @@ static const struct translation_st evp_pkey_ctx_translations[] = {
      * SipHash
      * ======
      */
-    { SET, -1, -1, EVP_PKEY_OP_TYPE_SIG,
-      EVP_PKEY_CTRL_SET_DIGEST_SIZE, "digestsize", NULL,
-      OSSL_MAC_PARAM_SIZE, OSSL_PARAM_UNSIGNED_INTEGER, NULL },
 
     /*-
      * TLS1-PRF
-- 
2.23.0.windows.1

