From e9c51e454b746a3a7ff09d9e518ac0b54552dedc Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Thu, 18 Aug 2022 14:08:12 +0900
Subject: [PATCH] common

---
 test/testutil/main.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/test/testutil/main.c b/test/testutil/main.c
index 6716750a30..52a3185fdd 100644
--- a/test/testutil/main.c
+++ b/test/testutil/main.c
@@ -10,10 +10,31 @@
 #include "../testutil.h"
 #include "output.h"
 #include "tu_local.h"
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif


 int main(int argc, char *argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
     int ret = EXIT_FAILURE;

     test_open_streams();
--
2.25.1

