From 1f53a714b6a89e8cbe774a7d751dddc659e16082 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:22:44 +0900
Subject: [PATCH] buggy

---
 crypto/evp/m_sigver.c | 34 +++++++---------------------------
 1 file changed, 7 insertions(+), 27 deletions(-)

diff --git a/crypto/evp/m_sigver.c b/crypto/evp/m_sigver.c
index 806ef3224c..f21865a8c3 100644
--- a/crypto/evp/m_sigver.c
+++ b/crypto/evp/m_sigver.c
@@ -400,7 +400,7 @@ int EVP_DigestSignFinal(EVP_MD_CTX *ctx, unsigned char *sigret,
                         size_t *siglen)
 {
     int sctx = 0, r = 0;
-    EVP_PKEY_CTX *dctx, *pctx = ctx->pctx;
+    EVP_PKEY_CTX *pctx = ctx->pctx;

     if (pctx == NULL
             || pctx->operation != EVP_PKEY_OP_SIGNCTX
@@ -408,19 +408,8 @@ int EVP_DigestSignFinal(EVP_MD_CTX *ctx, unsigned char *sigret,
             || pctx->op.sig.signature == NULL)
         goto legacy;

-    if (sigret == NULL || (ctx->flags & EVP_MD_CTX_FLAG_FINALISE) != 0)
-        return pctx->op.sig.signature->digest_sign_final(pctx->op.sig.algctx,
-                                                         sigret, siglen,
-                                                         SIZE_MAX);
-    dctx = EVP_PKEY_CTX_dup(pctx);
-    if (dctx == NULL)
-        return 0;
-
-    r = dctx->op.sig.signature->digest_sign_final(dctx->op.sig.algctx,
-                                                  sigret, siglen,
-                                                  SIZE_MAX);
-    EVP_PKEY_CTX_free(dctx);
-    return r;
+    return pctx->op.sig.signature->digest_sign_final(pctx->op.sig.algctx,
+                                                     sigret, siglen, SIZE_MAX);

  legacy:
     if (pctx == NULL || pctx->pmeth == NULL) {
@@ -440,7 +429,7 @@ int EVP_DigestSignFinal(EVP_MD_CTX *ctx, unsigned char *sigret,
         if (ctx->flags & EVP_MD_CTX_FLAG_FINALISE)
             r = pctx->pmeth->signctx(pctx, sigret, siglen, ctx);
         else {
-            dctx = EVP_PKEY_CTX_dup(pctx);
+            EVP_PKEY_CTX *dctx = EVP_PKEY_CTX_dup(pctx);
             if (dctx == NULL)
                 return 0;
             r = dctx->pmeth->signctx(dctx, sigret, siglen, ctx);
@@ -526,7 +516,7 @@ int EVP_DigestVerifyFinal(EVP_MD_CTX *ctx, const unsigned char *sig,
     int r = 0;
     unsigned int mdlen = 0;
     int vctx = 0;
-    EVP_PKEY_CTX *dctx, *pctx = ctx->pctx;
+    EVP_PKEY_CTX *pctx = ctx->pctx;

     if (pctx == NULL
             || pctx->operation != EVP_PKEY_OP_VERIFYCTX
@@ -534,17 +524,8 @@ int EVP_DigestVerifyFinal(EVP_MD_CTX *ctx, const unsigned char *sig,
             || pctx->op.sig.signature == NULL)
         goto legacy;

-    if ((ctx->flags & EVP_MD_CTX_FLAG_FINALISE) != 0)
-        return pctx->op.sig.signature->digest_verify_final(pctx->op.sig.algctx,
-                                                           sig, siglen);
-    dctx = EVP_PKEY_CTX_dup(pctx);
-    if (dctx == NULL)
-        return 0;
-
-    r = dctx->op.sig.signature->digest_verify_final(dctx->op.sig.algctx,
-                                                    sig, siglen);
-    EVP_PKEY_CTX_free(dctx);
-    return r;
+    return pctx->op.sig.signature->digest_verify_final(pctx->op.sig.algctx,
+                                                       sig, siglen);

  legacy:
     if (pctx == NULL || pctx->pmeth == NULL) {
--
2.23.0.windows.1

