From 3ac2081ed40697af251968936f7144e68086ee60 Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 12:24:03 +0900
Subject: [PATCH] buggy

---
 ssl/statem/extensions_clnt.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/ssl/statem/extensions_clnt.c b/ssl/statem/extensions_clnt.c
index cac713fff0..b216e29f26 100644
--- a/ssl/statem/extensions_clnt.c
+++ b/ssl/statem/extensions_clnt.c
@@ -234,7 +234,7 @@ EXT_RETURN tls_construct_ctos_supported_groups(SSL *s, WPACKET *pkt,
         }
     }
     if (!WPACKET_close(pkt) || !WPACKET_close(pkt)) {
-        if (added == 0)
+        if (added == 0 || (tls13added == 0 && max_version == TLS1_3_VERSION))
             SSLfatal_data(s, SSL_AD_INTERNAL_ERROR, SSL_R_NO_SUITABLE_GROUPS,
                           "No groups enabled for max supported SSL/TLS version");
         else
@@ -242,12 +242,7 @@ EXT_RETURN tls_construct_ctos_supported_groups(SSL *s, WPACKET *pkt,
         return EXT_RETURN_FAIL;
     }

-    if (tls13added == 0 && max_version == TLS1_3_VERSION) {
-        SSLfatal_data(s, SSL_AD_INTERNAL_ERROR, SSL_R_NO_SUITABLE_GROUPS,
-                      "No groups enabled for max supported SSL/TLS version");
-        return EXT_RETURN_FAIL;
-    }

     return EXT_RETURN_SENT;
 }

--
2.23.0.windows.1

