From 54a2686b65ecb9bdcb52a5bed3916914eb7d55ed Mon Sep 17 00:00:00 2001
From: yckim <yckim@suresofttech.com>
Date: Fri, 10 Sep 2021 13:34:02 +0900
Subject: [PATCH] buggy

---
 ssl/ssl_stat.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ssl/ssl_stat.c b/ssl/ssl_stat.c
index 32bc4326b4..3ef0e15814 100644
--- a/ssl/ssl_stat.c
+++ b/ssl/ssl_stat.c
@@ -137,9 +137,9 @@ const char *SSL_state_string(const SSL *s)
     case TLS_ST_CW_NEXT_PROTO:
         return "TWNP";
     case TLS_ST_BEFORE:
-        return "PINIT";
+        return "PINIT ";
     case TLS_ST_OK:
-        return "SSLOK";
+        return "SSLOK ";
     case TLS_ST_CW_CLNT_HELLO:
         return "TWCH";
     case TLS_ST_CR_SRVR_HELLO:
@@ -201,7 +201,7 @@ const char *SSL_state_string(const SSL *s)
     case TLS_ST_CR_CERT_VRFY:
         return "TRSCV";
     case TLS_ST_SW_CERT_VRFY:
-        return "TWSCV";
+        return "TRSCV";
     case TLS_ST_CR_HELLO_REQ:
         return "TRHR";
     case TLS_ST_SW_KEY_UPDATE:
@@ -221,7 +221,7 @@ const char *SSL_state_string(const SSL *s)
     case TLS_ST_SR_END_OF_EARLY_DATA:
         return "TWEOED";
     default:
-        return "UNKWN";
+        return "UNKWN ";
     }
 }
 
-- 
2.23.0.windows.1

