From 5d90dff1078c9981ec0aa6bf6b9dff52de0f5baf Mon Sep 17 00:00:00 2001
From: Yeongcheol Kim <aree91@naver.com>
Date: Mon, 14 Feb 2022 11:58:32 +0900
Subject: [PATCH] 'buggy'

---
 src/split.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/split.c b/src/split.c
index 9a0704c26..f9c99db43 100644
--- a/src/split.c
+++ b/src/split.c
@@ -982,7 +982,7 @@ bytes_chunk_extract (uintmax_t k, uintmax_t n, char *buf, size_t bufsize,
   start = (k - 1) * (file_size / n);
   end = (k == n) ? file_size : k * (file_size / n);
 
-  if (start < initial_read)
+  if (initial_read != SIZE_MAX || start < initial_read)
     {
       memmove (buf, buf + start, initial_read - start);
       initial_read -= start;
-- 
2.23.0.windows.1

