From b6dac1775666ecc1dbdb52154ceacc2285ef71b2 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Wed, 22 Jun 2022 10:00:47 +0900
Subject: [PATCH] common

---
 Makefile.am    | 2 ++
 tests/local.mk | 2 --
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index f2c567de0..fd1ae838c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -228,3 +228,5 @@ include $(top_srcdir)/src/local.mk
 include $(top_srcdir)/doc/local.mk
 include $(top_srcdir)/man/local.mk
 include $(top_srcdir)/tests/local.mk
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tests/local.mk b/tests/local.mk
index 33350021d..762b93d69 100644
--- a/tests/local.mk
+++ b/tests/local.mk
@@ -183,7 +183,6 @@ all_tests =					\
   tests/rm/d-2.sh				\
   tests/rm/d-3.sh				\
   tests/rm/deep-1.sh				\
-  tests/rm/deep-2.sh				\
   tests/rm/dir-no-w.sh				\
   tests/rm/dir-nonrecur.sh			\
   tests/rm/dot-rel.sh				\
@@ -530,7 +529,6 @@ all_tests =					\
   tests/du/inacc-dir.sh				\
   tests/du/inaccessible-cwd.sh			\
   tests/du/inodes.sh				\
-  tests/du/long-from-unreadable.sh		\
   tests/du/long-sloop.sh			\
   tests/du/max-depth.sh				\
   tests/du/move-dir-while-traversing.sh		\
--
2.25.1
