From 9e60eebb83b01ddbe755fa5a527b036e3de5c1ce Mon Sep 17 00:00:00 2001
From: Yeongcheol Kim <aree91@naver.com>
Date: Mon, 14 Feb 2022 11:47:37 +0900
Subject: [PATCH] 'common'

---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index f2c567de0..fd1ae838c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -228,3 +228,5 @@ include $(top_srcdir)/src/local.mk
 include $(top_srcdir)/doc/local.mk
 include $(top_srcdir)/man/local.mk
 include $(top_srcdir)/tests/local.mk
+
+print-TESTS: ; @echo $(TESTS)
-- 
2.23.0.windows.1

