From 762df71994f7d6f90d0b39d2cb01640bb368dab2 Mon Sep 17 00:00:00 2001
From: Yeongcheol Kim <aree91@naver.com>
Date: Mon, 14 Feb 2022 12:09:36 +0900
Subject: [PATCH] 'buggy'

---
 src/pr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pr.c b/src/pr.c
index 26f221f99..20e863760 100644
--- a/src/pr.c
+++ b/src/pr.c
@@ -1233,7 +1233,7 @@ init_parameters (int number_of_files)
         }
       /* It's rather pointless to define a TAB separator with column
          alignment */
-      else if (!join_lines && col_sep_length == 1 && *col_sep_string == '\t')
+      else if (!join_lines && *col_sep_string == '\t')
         col_sep_string = column_separator;
 
       truncate_lines = true;
-- 
2.23.0.windows.1

