From 875f076bd9ec2d46a3a2b85e409c4fc838f012df Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 16 Aug 2022 11:42:51 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index 0efa34f..fdfaf14 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -5641,8 +5641,7 @@ redo:
     }
 
     /* Check whether we are table continuation. */
-    if(pivot_line->type == MD_LINE_TABLE  &&  md_is_table_row(ctx, off, &off)  &&
-       n_parents == ctx->n_containers)
+    if(pivot_line->type == MD_LINE_TABLE  &&  md_is_table_row(ctx, off, &off))
     {
         line->type = MD_LINE_TABLE;
         goto done;
-- 
2.25.1

