From a8ff7bd7a5d9c86f6a02eae6d42aba764a063953 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 16 Aug 2022 13:55:01 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index 1646032..62dbbd7 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -3961,10 +3961,6 @@ md_process_inlines(MD_CTX* ctx, const MD_LINE* lines, int n_lines)
                                 md_mark_get_ptr(ctx, title_mark - ctx->marks), title_mark->prev));
 
                     /* link/image closer may span multiple lines. */
-                    if(mark->ch == ']') {
-                        while(mark->end > line->end)
-                            line++;
-                    }
 
                     break;
                 }
-- 
2.25.1

