From 258b6319128b4f6faf3582ce5ff5bb83287270a1 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Tue, 16 Aug 2022 14:21:01 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index 1b75c3c..e6b2b94 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -3490,7 +3490,7 @@ static int
 md_split_simple_pairing_mark(MD_CTX* ctx, int mark_index, SZ n)
 {
     MD_MARK* mark = &ctx->marks[mark_index];
-    int new_mark_index = mark_index + (mark->end - mark->beg - n);
+    int new_mark_index = mark_index + (mark->end - mark->beg - 1);
     MD_MARK* dummy = &ctx->marks[new_mark_index];
 
     MD_ASSERT(mark->end - mark->beg > n);
-- 
2.25.1

