From 3881d759e730ea0d5bda159202276120852a7286 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 11:13:02 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index b0ef739..a14094d 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -3725,7 +3725,7 @@ md_analyze_emph(MD_CTX* ctx, int mark_index)
         if(opener != NULL) {
             SZ opener_size = opener->end - opener->beg;
             SZ closer_size = mark->end - mark->beg;
-            MD_MARKCHAIN* opener_chain = md_mark_chain(ctx, opener_index);
+            MD_MARKCHAIN* opener_chain = md_mark_chain(ctx, mark_index);
 
             if(opener_size > closer_size) {
                 opener_index = md_split_emph_mark(ctx, opener_index, closer_size);
-- 
2.25.1

