From 634416dea543b1f903b4f859b454647db4154534 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 11:47:34 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index 3745cf3..dabd899 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -5486,7 +5486,7 @@ md_is_container_mark(MD_CTX* ctx, unsigned indent, OFF beg, OFF* p_end, MD_CONTA
         p_container->start = p_container->start * 10 + CH(off) - _T('0');
         off++;
     }
-    if(off > beg  &&  off+1 < ctx->size  &&
+    if(off+1 < ctx->size  &&
        (CH(off) == _T('.') || CH(off) == _T(')'))  &&
        (ISBLANK(off+1) || ISNEWLINE(off+1)))
     {
-- 
2.25.1

