From 57c0bdca15e7de3565501c125ade8e458b7edc6c Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 13:43:49 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index 90dab3f..1aca46b 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -4739,7 +4739,6 @@ md_consume_link_reference_definitions(MD_CTX* ctx)
             /* Remove complete block. */
             ctx->n_block_bytes -= n * sizeof(MD_LINE);
             ctx->n_block_bytes -= sizeof(MD_BLOCK);
-            ctx->current_block = NULL;
         } else {
             /* Remove just some initial lines from the block. */
             memmove(lines, lines + n, (n_lines - n) * sizeof(MD_LINE));
-- 
2.25.1

