From 6610f100d4e8309a079bd359f7db9c890e9c4673 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 10:20:25 +0900
Subject: [PATCH] buggy

---
 src/md4c.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/md4c.c b/src/md4c.c
index caa5fe7..d74c21f 100644
--- a/src/md4c.c
+++ b/src/md4c.c
@@ -3527,7 +3527,6 @@ md_resolve_links(MD_CTX* ctx, const MD_LINE* lines, int n_lines)
 
                 /* Do not analyze the label as a standalone link in the next
                  * iteration. */
-                next_index = ctx->marks[next_index].prev;
             }
         } else {
             if(closer->end < ctx->size  &&  CH(closer->end) == _T('(')) {
-- 
2.25.1

