From 1e2c5f6414e1fd97a7ba2eb6f9eab81abbc2e67a Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 12:05:34 +0900
Subject: [PATCH] buggy

---
 md4c/md4c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/md4c/md4c.c b/md4c/md4c.c
index d2b4313..01e63a5 100644
--- a/md4c/md4c.c
+++ b/md4c/md4c.c
@@ -5700,7 +5700,7 @@ md_analyze_line(MD_CTX* ctx, OFF beg, OFF* p_end,
            md_is_container_mark(ctx, line->indent, off, &off, &container))
         {
             if(pivot_line->type == MD_LINE_TEXT  &&  n_parents == ctx->n_containers  &&
-                        (off >= ctx->size || ISNEWLINE(off))  &&  container.ch != _T('>'))
+                        (off >= ctx->size || ISNEWLINE(off)))
             {
                 /* Noop. List mark followed by a blank line cannot interrupt a paragraph. */
             } else if(pivot_line->type == MD_LINE_TEXT  &&  n_parents == ctx->n_containers  &&
-- 
2.25.1

