From dc7a18676644c22364b7459f32d3c878629685e6 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Wed, 17 Aug 2022 12:06:51 +0900
Subject: [PATCH] common

---
 test/spec_tests.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/spec_tests.py b/test/spec_tests.py
index c739e5f..8702e31 100755
--- a/test/spec_tests.py
+++ b/test/spec_tests.py
@@ -30,6 +30,8 @@ if __name__ == "__main__":
             default=False, help='filter stdin through normalizer for testing')
     parser.add_argument('-n', '--number', type=int, default=None,
             help='only consider the test with the given number')
+    parser.add_argument('--skip', type=int, default=None,
+            help='skip tests')
     args = parser.parse_args(sys.argv[1:])
 
 def out(str):
@@ -130,7 +132,7 @@ if __name__ == "__main__":
         pattern_re = re.compile(args.pattern, re.IGNORECASE)
     else:
         pattern_re = re.compile('.')
-    tests = [ test for test in all_tests if re.search(pattern_re, test['section']) and (not args.number or test['example'] == args.number) ]
+    tests = [ test for test in all_tests if re.search(pattern_re, test['section']) and (not args.number or test['example'] == args.number) and (not args.skip or test['example'] != args.skip) ]
     if args.dump_tests:
         out(json.dumps(tests, ensure_ascii=False, indent=2))
         exit(0)
-- 
2.25.1

