From 3e8f606b1bc1124f4b130b00b21d61a551b5e939 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 14 Jul 2022 16:15:23 +0900
Subject: [PATCH] common

---
 jerry-main/main-unix.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/jerry-main/main-unix.c b/jerry-main/main-unix.c
index 367f5096..b3cad7c7 100644
--- a/jerry-main/main-unix.c
+++ b/jerry-main/main-unix.c
@@ -469,6 +469,14 @@ init_engine (jerry_init_flag_t flags, /**< initialized flags for the engine */
   register_js_function ("resourceName", jerryx_handler_resource_name);
 } /* init_engine */
 
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+	__gcov_flush();
+}
+#endif
+
 int
 main (int argc,
       char **argv)
-- 
2.25.1

