From 64898d09b71a187bae4a3f7f09d1a97d8780f2d9 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 15 Jul 2022 15:21:15 +0900
Subject: [PATCH] common

---
 jerry-main/main-jerry.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/jerry-main/main-jerry.c b/jerry-main/main-jerry.c
index dbf76349..a3125192 100644
--- a/jerry-main/main-jerry.c
+++ b/jerry-main/main-jerry.c
@@ -45,6 +45,14 @@ context_alloc (size_t size,
 } /* context_alloc */
 #endif /* defined (JERRY_EXTERNAL_CONTEXT) && (JERRY_EXTERNAL_CONTEXT == 1) */
 
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+	__gcov_flush();
+}
+#endif
+
 int
 main (int argc,
       char **argv)
-- 
2.25.1

