From 5c68476d332cdb1b03cacc99070cfe8a50e1bd16 Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Fri, 15 Jul 2022 15:16:52 +0900
Subject: [PATCH] buggy

---
 jerry-core/vm/vm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/jerry-core/vm/vm.c b/jerry-core/vm/vm.c
index b6fef5df..ab40ec01 100644
--- a/jerry-core/vm/vm.c
+++ b/jerry-core/vm/vm.c
@@ -4144,7 +4144,6 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */

             if (ECMA_IS_VALUE_ERROR (result))
             {
-              stack_top_p[-3] = index;
               goto error;
             }

--
2.25.1

