From 38c194fd5a62804822bdc5bcac1315556bc2f3b3 Mon Sep 17 00:00:00 2001
From: cray <craycoding@gmail.com>
Date: Fri, 16 Sep 2022 11:34:33 +0900
Subject: [PATCH] buggy

---
 jerry-core/vm/opcodes.c | 41 ++++++++++++-----------------------------
 1 file changed, 12 insertions(+), 29 deletions(-)

diff --git a/jerry-core/vm/opcodes.c b/jerry-core/vm/opcodes.c
index 9819f343..9adad310 100644
--- a/jerry-core/vm/opcodes.c
+++ b/jerry-core/vm/opcodes.c
@@ -1403,58 +1403,41 @@ opfunc_private_set (ecma_value_t base, /**< this object */
                     ecma_value_t property, /**< property name */
                     ecma_value_t value) /**< ecma value */
 {
-  ecma_value_t base_obj = ecma_op_to_object (base);
+  ecma_object_t *obj_p = ecma_get_object_from_value (base);
 
-  if (ECMA_IS_VALUE_ERROR (base_obj))
-  {
-    return base_obj;
-  }
 
-  ecma_object_t *obj_p = ecma_get_object_from_value (base_obj);
   ecma_string_t *prop_name_p = ecma_get_string_from_value (property);
   ecma_string_t *private_key_p = NULL;
 
   ecma_property_t *prop_p = opfunc_find_private_element (obj_p, prop_name_p, &private_key_p, true);
 
-  ecma_value_t result;
 
   if (prop_p == NULL)
   {
-    result = ecma_raise_type_error (ECMA_ERR_CANNOT_WRITE_PRIVATE_MEMBER_TO_AN_OBJECT_WHOSE_CLASS_DID_NOT_DECLARE_IT);
+    return ecma_raise_type_error (ECMA_ERR_CANNOT_WRITE_PRIVATE_MEMBER_TO_AN_OBJECT_WHOSE_CLASS_DID_NOT_DECLARE_IT);
   }
-  else if (*prop_p & ECMA_PROPERTY_FLAG_DATA)
+  if (*prop_p & ECMA_PROPERTY_FLAG_DATA)
   {
     JERRY_ASSERT (ecma_prop_name_is_symbol (private_key_p));
 
     if (private_key_p->u.hash & ECMA_SYMBOL_FLAG_PRIVATE_INSTANCE_METHOD)
     {
-      result = ecma_raise_type_error (ECMA_ERR_PRIVATE_METHOD_IS_NOT_WRITABLE);
-    }
-    else
-    {
-      ecma_value_assign_value (&ECMA_PROPERTY_VALUE_PTR (prop_p)->value, value);
-      result = ecma_copy_value (value);
+      return ecma_raise_type_error (ECMA_ERR_PRIVATE_METHOD_IS_NOT_WRITABLE);
     }
+    ecma_value_assign_value (&ECMA_PROPERTY_VALUE_PTR (prop_p)->value, value);
+    return ecma_copy_value (value);
   }
-  else
-  {
-    ecma_getter_setter_pointers_t *get_set_pair_p = ecma_get_named_accessor_property (ECMA_PROPERTY_VALUE_PTR (prop_p));
+  ecma_getter_setter_pointers_t *get_set_pair_p = ecma_get_named_accessor_property (ECMA_PROPERTY_VALUE_PTR (prop_p));
 
-    if (get_set_pair_p->setter_cp == JMEM_CP_NULL)
-    {
-      result = ecma_raise_type_error (ECMA_ERR_PRIVATE_FIELD_WAS_DEFINED_WITHOUT_A_SETTER);
-    }
-    else
-    {
-      ecma_object_t *setter_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, get_set_pair_p->setter_cp);
 
-      result = ecma_op_function_call (setter_p, base, &value, 1);
-    }
+  if (get_set_pair_p->setter_cp == JMEM_CP_NULL)
+  {
+    return ecma_raise_type_error (ECMA_ERR_PRIVATE_FIELD_WAS_DEFINED_WITHOUT_A_SETTER);
   }
 
-  ecma_deref_object (obj_p);
+  ecma_object_t *setter_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, get_set_pair_p->setter_cp);
 
-  return result;
+  return ecma_op_function_call (setter_p, base, &value, 1);
 } /* opfunc_private_set */
 
 /**
-- 
2.25.1

