From 26cc109077302ec94e8ad847231ff8b9edabeefe Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Mon, 18 Jul 2022 14:31:47 +0900
Subject: [PATCH] common

---
 jerry-main/main-jerry.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/jerry-main/main-jerry.c b/jerry-main/main-jerry.c
index 950f707c..6f0ed145 100644
--- a/jerry-main/main-jerry.c
+++ b/jerry-main/main-jerry.c
@@ -44,6 +44,14 @@ context_alloc (size_t size, void *cb_data_p)
 } /* context_alloc */
 #endif /* defined (JERRY_EXTERNAL_CONTEXT) && (JERRY_EXTERNAL_CONTEXT == 1) */
 
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+  __gcov_flush();
+}
+#endif
+
 int
 main (int argc, char **argv)
 {
-- 
2.25.1

