From f019ea97bd6ed42723edd138b0b40cca5ae53caf Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Wed, 15 Jun 2022 14:07:13 +0900
Subject: [PATCH] common

---
 test/Makefile.am |  2 ++
 tools/tiffcrop.c | 24 +++++++++++++++++++++++-
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/test/Makefile.am b/test/Makefile.am
index 0aca56fc..fe7fbecd 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -317,3 +317,5 @@ generate-tiffcrop-tests: \
 generate-tests: \
 	generate-tiff2rgba-tests \
 	generate-tiffcrop-tests
+
+print-TESTS: ; @echo $(TESTS)
diff --git a/tools/tiffcrop.c b/tools/tiffcrop.c
index 6fc8fc1d..361e4b7b 100644
--- a/tools/tiffcrop.c
+++ b/tools/tiffcrop.c
@@ -2107,10 +2107,32 @@ update_output_file (TIFF **tiffout, char *mode, int autoindex,
   } /* end update_output_file */


+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+	__gcov_flush();
+	sigaction(sigaction, &dpp_orig_sigaction, NULL);
+	raise(signum);
+	exit(1);
+}
+#endif
+
+
 int
 main(int argc, char* argv[])
   {
-
+#ifdef DPP_ENABLE_GCOV
+	  {
+		  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+		  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+		  dpp_gcov_sigaction.sa_flags = 0;
+		  sigaction(SIGSEGV, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGFPE, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+		  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+	  }
+#endif
 #if !HAVE_DECL_OPTARG
   extern int optind;
 #endif
--
2.25.1
