From 339eac28e1d45692bca4a5af592fec43b36d2978 Mon Sep 17 00:00:00 2001
From: Peter Kwon <minhyuk@suresofttech.com>
Date: Mon, 14 Feb 2022 06:12:54 +0000
Subject: [PATCH] buggy

---
 libtiff/tif_jpeg.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/libtiff/tif_jpeg.c b/libtiff/tif_jpeg.c
index 6c17c388..38595f98 100644
--- a/libtiff/tif_jpeg.c
+++ b/libtiff/tif_jpeg.c
@@ -1626,13 +1626,7 @@ JPEGSetupEncode(TIFF* tif)
 	case PHOTOMETRIC_YCBCR:
 		sp->h_sampling = td->td_ycbcrsubsampling[0];
 		sp->v_sampling = td->td_ycbcrsubsampling[1];
-                if( sp->h_sampling == 0 || sp->v_sampling == 0 )
-                {
-                    TIFFErrorExt(tif->tif_clientdata, module,
-                            "Invalig horizontal/vertical sampling value");
-                    return (0);
-                }

 		/*
 		 * A ReferenceBlackWhite field *must* be present since the
 		 * default value is inappropriate for YCbCr.  Fill in the
--
2.25.1

