From 23162922b06a7811e34460bb5346ddef988dfd6d Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Fri, 22 Jul 2022 16:58:21 +0900
Subject: [PATCH] common

---
 src/exiv2.cpp | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index 22ccafc4..a82e9b13 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -114,10 +114,36 @@ namespace {
     std::string parseEscapes(const std::string& input);
 }
 
+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+extern "C" void __gcov_flush();
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+  __gcov_flush();
+  sigaction(signum, &dpp_orig_sigaction, NULL);
+  raise(signum);
+}
+void __asan_on_error(void);
+void __asan_on_error(void){
+  __gcov_flush();
+}
+#endif
+
 // *****************************************************************************
 // Main
 int main(int argc, char* const argv[])
 {
+    #ifdef DPP_ENABLE_GCOV
+  {
+  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+   sigemptyset(&dpp_gcov_sigaction.sa_mask);
+   dpp_gcov_sigaction.sa_flags = 0;
+   sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+  }
+  #endif
+
     Exiv2::XmpParser::initialize();
     ::atexit(Exiv2::XmpParser::terminate);
 #ifdef EXV_ENABLE_BMFF
-- 
2.25.1

