From 3e6be89c582b57c19976f336184f0a79766f1d19 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Fri, 22 Jul 2022 12:27:15 +0900
Subject: [PATCH] common

---
 src/exiv2.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index 22ccafc4..8356d4b1 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -114,6 +114,14 @@ namespace {
     std::string parseEscapes(const std::string& input);
 }
 
+#ifdef DPP_ENABLE_GCOV
+#include <gcov.h>
+void __asan_on_error(void);
+void __asan_on_error(void) {
+  __gcov_flush();
+}
+#endif
+
 // *****************************************************************************
 // Main
 int main(int argc, char* const argv[])
-- 
2.25.1

