From a239029ad04e0fd2c20f01cc178bbf7dac85a9bc Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Mon, 25 Jul 2022 17:44:10 +0900
Subject: [PATCH] buggy

---
 src/crwimage_int.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/crwimage_int.cpp b/src/crwimage_int.cpp
index 7b958c26..a44a67e2 100644
--- a/src/crwimage_int.cpp
+++ b/src/crwimage_int.cpp
@@ -579,7 +579,7 @@ namespace Exiv2 {
     void CiffComponent::setValue(DataBuf buf)
     {
         if (isAllocated_) {
-            delete[] pData_;
+            delete pData_;
             pData_ = 0;
             size_ = 0;
         }
@@ -1187,11 +1187,7 @@ namespace Exiv2 {
                                                  pCrwMapping->crwDir_);
         if (edX != edEnd || edY != edEnd || edO != edEnd) {
             uint32_t size = 28;
-            if (cc) {
-              if (cc->size() < size)
-                throw Error(kerCorruptedMetadata);
-              size = cc->size();
-            }
+            if (cc && cc->size() > size) size = cc->size();
             DataBuf buf(size);
             std::memset(buf.pData_, 0x0, buf.size_);
             if (cc) std::memcpy(buf.pData_ + 8, cc->pData() + 8, cc->size() - 8);
--
2.25.1

