From 94c82c0f5adb2d4479d9f81f9e1c5d6eb4064186 Mon Sep 17 00:00:00 2001
From: HansolChoe <hansol614@gmail.com>
Date: Thu, 28 Jul 2022 02:03:29 +0000
Subject: [PATCH] common

---
 src/exiv2.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index 22ccafc4..639541c5 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -114,10 +114,32 @@ namespace {
     std::string parseEscapes(const std::string& input);
 }

+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(1);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
+
 // *****************************************************************************
 // Main
 int main(int argc, char* const argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
     Exiv2::XmpParser::initialize();
     ::atexit(Exiv2::XmpParser::terminate);
 #ifdef EXV_ENABLE_BMFF
--
2.17.1
