From 422239de55d344e5b017dde34b7fa3c6aa1641e4 Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Thu, 28 Jul 2022 11:54:34 +0900
Subject: [PATCH] common

---
 src/exiv2.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index 4c6b86e3..c4c54323 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -31,8 +31,30 @@
 #include "params.hpp"
 #include "i18n.h"  // NLS support.

+#ifdef DPP_ENABLE_GCOV
+#include <signal.h>
+void dpp_sighandler(int signum);
+static struct sigaction dpp_gcov_sigaction;
+extern "C" void __gcov_flush();
+void dpp_gcov_flush(int signum) {
+   __gcov_flush();
+   exit(2);
+}
+void dpp_gcov_handler(void(*dpp_gcov_flush)(int sig)) {
+    dpp_gcov_sigaction.sa_handler = dpp_gcov_flush;
+    for (int sig = 1; sig <= SIGRTMAX; ++sig) {
+        sigaction(sig, &dpp_gcov_sigaction, NULL);
+    }
+}
+#endif
+
 int main(int argc, char* const argv[])
 {
+#ifdef DPP_ENABLE_GCOV
+    {
+        dpp_gcov_handler(dpp_gcov_flush);
+    }
+#endif
 #ifdef EXV_ENABLE_NLS
     setlocale(LC_ALL, "");
     const std::string localeDir = Exiv2::getProcessPath() + EXV_LOCALEDIR;
--
2.25.1

