From 59073d676f96f28da4fe32ad6e8b3c4435c51f6b Mon Sep 17 00:00:00 2001
From: icechocola05 <icechocola05@gmail.com>
Date: Thu, 28 Jul 2022 16:41:49 +0900
Subject: [PATCH] buggy

---
 src/crwimage_int.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/crwimage_int.cpp b/src/crwimage_int.cpp
index 4080c078..c2fd5f3a 100644
--- a/src/crwimage_int.cpp
+++ b/src/crwimage_int.cpp
@@ -281,7 +281,7 @@ namespace Exiv2 {
         if (size < 4)
             throw Error(kerCorruptedMetadata);
         uint32_t o = getULong(pData + size - 4, byteOrder);
-        if ( o > size-2 )
+        if ( o+2 > size )
             throw Error(kerCorruptedMetadata);
         uint16_t count = getUShort(pData + o, byteOrder);
 #ifdef DEBUG
@@ -289,7 +289,7 @@ namespace Exiv2 {
                   <<", " << count << " entries \n";
 #endif
         o += 2;
-        if ( static_cast<uint32_t>(count) * 10 > size-o )
+        if ( (o + (count * 10)) > size )
             throw Error(kerCorruptedMetadata);
 
         for (uint16_t i = 0; i < count; ++i) {
-- 
2.25.1

