From a1a6d32278a023958e54dbf03e3af684d55843eb Mon Sep 17 00:00:00 2001
From: kseo <kseo@suresofttech.com>
Date: Wed, 3 Aug 2022 09:23:51 +0900
Subject: [PATCH] buggy

---
 src/psdimage.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/psdimage.cpp b/src/psdimage.cpp
index 8ed67544..e69dd0d0 100644
--- a/src/psdimage.cpp
+++ b/src/psdimage.cpp
@@ -199,7 +199,6 @@ namespace Exiv2 {
             throw Error(kerNotAnImage, "Photoshop");
         }
         uint32_t resourcesLength = getULong(buf, bigEndian);
-        enforce(resourcesLength < io_->size(), Exiv2::kerCorruptedMetadata);

         while (resourcesLength > 0)
         {
--
2.25.1

