From 8d6eef78be59e0989ca7f4243869b7a7e96e18b5 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 09:54:02 +0900
Subject: [PATCH] buggy

---
 src/iptc.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/iptc.cpp b/src/iptc.cpp
index a3bb6e1d..d30864c3 100644
--- a/src/iptc.cpp
+++ b/src/iptc.cpp
@@ -461,7 +461,6 @@ namespace Exiv2 {
                 uint16_t sizeOfSize = (getUShort(pRead, bigEndian) & 0x7FFF);
                 if (sizeOfSize > 4) return 5;
                 pRead += 2;
-                if (sizeOfSize > static_cast<size_t>(pEnd - pRead)) return 6;
                 sizeData = 0;
                 for (; sizeOfSize > 0; --sizeOfSize) {
                     sizeData |= *pRead++ << (8 *(sizeOfSize-1));
-- 
2.25.1

