From 945abd5cfb4f619eb5069270ba808c1da62c8519 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 10:10:00 +0900
Subject: [PATCH] buggy

---
 src/pngchunk_int.cpp | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/src/pngchunk_int.cpp b/src/pngchunk_int.cpp
index adab5715..714b95b4 100644
--- a/src/pngchunk_int.cpp
+++ b/src/pngchunk_int.cpp
@@ -653,17 +653,9 @@ namespace Exiv2 {
             }
         }
 
-        // Parse the length.
-        long length = 0;
+        const char* startOfLength = sp;
         while ('0' <= *sp && *sp <= '9')
         {
-            // Compute the new length using unsigned long, so that we can
-            // check for overflow.
-            const unsigned long newlength = (10 * static_cast<unsigned long>(length)) + (*sp - '0');
-            if (newlength > static_cast<unsigned long>(std::numeric_limits<long>::max())) {
-                return DataBuf(); // Integer overflow.
-            }
-            length = static_cast<long>(newlength);
             sp++;
             if (sp == eot )
             {
@@ -675,7 +667,8 @@ namespace Exiv2 {
             return DataBuf();
         }
 
-        enforce(length <= (eot - sp)/2, Exiv2::kerCorruptedMetadata);
+        long length = (long) atol(startOfLength);
+        enforce(0 <= length && length <= (eot - sp)/2, Exiv2::kerCorruptedMetadata);
 
         // Allocate space
         if (length == 0)
-- 
2.25.1

