From 07e39c994bd22303bd4cc9b85e713f4f0fc02ca9 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 15:07:56 +0900
Subject: [PATCH] buggy

---
 src/jp2image.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jp2image.cpp b/src/jp2image.cpp
index 3ae8c28f..24e2db5c 100644
--- a/src/jp2image.cpp
+++ b/src/jp2image.cpp
@@ -501,7 +501,7 @@ namespace Exiv2
                             subBox.length = getLong((byte*)&subBox.length, bigEndian);
                             subBox.type = getLong((byte*)&subBox.type, bigEndian);

-                            if (subBox.length < sizeof(box) || subBox.length > io_->size() - io_->tell()) {
+                            if (subBox.length > io_->size() - io_->tell()) {
                                 throw Error(kerCorruptedMetadata);
                             }

--
2.25.1

