From dfe203829141433c7481a93b1b18609b5013e29d Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Mon, 22 Aug 2022 15:43:37 +0900
Subject: [PATCH] buggy

---
 src/easyaccess.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/easyaccess.cpp b/src/easyaccess.cpp
index 9302d6cc..4962d15d 100644
--- a/src/easyaccess.cpp
+++ b/src/easyaccess.cpp
@@ -175,7 +175,7 @@ namespace Exiv2 {
                     md = md_st;
                     break;
                 }
-                while (strcmp(sensKeys->keys[idx++], md_st->key().c_str()) != 0 && idx < sensKeys->count) {}
+                while (strcmp(sensKeys->keys[idx++], md_st->key().c_str()) != 0 && idx < cnt) {}
             }
             break;
         }
-- 
2.25.1

