From 39ede029503cf2b9119139ce7f4d34057ddc1b74 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 09:55:06 +0900
Subject: [PATCH] buggy

---
 src/tiffcomposite_int.cpp | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/tiffcomposite_int.cpp b/src/tiffcomposite_int.cpp
index 0d98f775..966f51d5 100644
--- a/src/tiffcomposite_int.cpp
+++ b/src/tiffcomposite_int.cpp
@@ -1236,12 +1236,7 @@ namespace Exiv2 {
                                                   valueIdx,
                                                   dataIdx,
                                                   imageIdx);
-#ifndef SUPPRESS_WARNINGS
-            if (len > 4) {
-                EXV_ERROR << "Unexpected length in TiffDirectory::writeDirEntry(): len == "
-                          << len << ".\n";
-            }
-#endif
+            assert(len <= 4);
             if (len < 4) {
                 memset(buf, 0x0, 4);
                 ioWrapper.write(buf, 4 - len);
-- 
2.25.1

