From 858c994345587c74b9489815ee7e9491fcdea1df Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Tue, 23 Aug 2022 11:33:55 +0900
Subject: [PATCH] buggy

---
 src/exiv2.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/exiv2.cpp b/src/exiv2.cpp
index c353ea04..1abbc59c 100644
--- a/src/exiv2.cpp
+++ b/src/exiv2.cpp
@@ -720,7 +720,7 @@ int Params::evalPrintFlags(const std::string& optArg)
                     printItems_ |= prHex;
                     break;
                 case 'V':
-                    printItems_ |= prSet | prKey | prType | prValue;
+                    printItems_ |= prSet | prValue;
                     break;
                 default:
                     std::cerr << progname() << ": " << _("Unrecognized print item") << " `" << i << "'\n";
-- 
2.25.1

