From fbb8b79798e89770633776783ea21406cef26936 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:31:34 +0900
Subject: [PATCH] buggy

---
 src/iso19111/c_api.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/iso19111/c_api.cpp b/src/iso19111/c_api.cpp
index ec6f1dc4..d7fb35f3 100644
--- a/src/iso19111/c_api.cpp
+++ b/src/iso19111/c_api.cpp
@@ -7664,9 +7664,6 @@ int proj_get_suggested_operation(PJ_CONTEXT *ctx, PJ_OBJ_LIST *operations,
 
     int iExcluded[2] = {-1, -1};
     const auto &preparedOps = opList->getPreparedOperations(ctx);
-    if (preparedOps.empty() && !opList->objects.empty()) {
-        return 0;
-    }
     int idx = pj_get_suggested_operation(ctx, preparedOps, iExcluded, direction,
                                          coord);
     if (idx >= 0) {
-- 
2.30.2

