From c44d03631ea00721b09c79598e18f1a672e51d37 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:34:51 +0900
Subject: [PATCH] buggy

---
 src/iso19111/metadata.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/iso19111/metadata.cpp b/src/iso19111/metadata.cpp
index e8ce37c6..fc788a30 100644
--- a/src/iso19111/metadata.cpp
+++ b/src/iso19111/metadata.cpp
@@ -1124,7 +1124,7 @@ void Identifier::_exportToJSON(JSONFormatter *formatter) const {
 //! @cond Doxygen_Suppress
 static bool isIgnoredChar(char ch) {
     return ch == ' ' || ch == '_' || ch == '-' || ch == '/' || ch == '(' ||
-           ch == ')' || ch == '.' || ch == '&' || ch == ',';
+           ch == ')' || ch == '.' || ch == '&';
 }
 //! @endcond
 
-- 
2.30.2

