From 196dc5596f5ae68feeac9b370281b0f8f0cb85ce Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:36:15 +0900
Subject: [PATCH] buggy

---
 src/projections/urm5.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/projections/urm5.cpp b/src/projections/urm5.cpp
index f89bb1d5..a93293c0 100644
--- a/src/projections/urm5.cpp
+++ b/src/projections/urm5.cpp
@@ -45,11 +45,7 @@ PJ *PROJECTION(urm5) {
     Q->q3 = pj_param(P->ctx, P->params, "dq").f / 3.;
     alpha = pj_param(P->ctx, P->params, "ralpha").f;
     t = Q->n * sin (alpha);
-    const double denom = sqrt (1. - t * t);
-    if( denom == 0 ) {
-        return pj_default_destructor(P, PJD_ERR_LAT_0_OR_ALPHA_EQ_90);
-    }
-    Q->m = cos (alpha) / denom;
+    Q->m = cos (alpha) / sqrt (1. - t * t);
     Q->rmn = 1. / (Q->m * Q->n);
 
     P->es = 0.;
-- 
2.30.2

