From 2684f3a03abad0f0e3c718b3703cda536a55ec8b Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:36:51 +0900
Subject: [PATCH] buggy

---
 src/projections/aeqd.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/projections/aeqd.cpp b/src/projections/aeqd.cpp
index e0fa5c37..a187edb7 100644
--- a/src/projections/aeqd.cpp
+++ b/src/projections/aeqd.cpp
@@ -151,7 +151,7 @@ oblcon:
                 return xy;
             }
             else
-                return aeqd_e_forward(lp, P);
+                xy.x = xy.y = 0.;
         else {
             xy.y = acos(xy.y);
             xy.y /= sin(xy.y);
-- 
2.30.2

