From ce1cb59cb583648900c489989c046f3caf5f25db Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:37:21 +0900
Subject: [PATCH] buggy

---
 src/projections/lcc.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/projections/lcc.cpp b/src/projections/lcc.cpp
index 5eee0d14..a1fe79a9 100644
--- a/src/projections/lcc.cpp
+++ b/src/projections/lcc.cpp
@@ -108,10 +108,6 @@ PJ *PROJECTION(lcc) {
         if (secant) { /* secant cone */
             sinphi = sin(Q->phi2);
             Q->n = log(m1 / pj_msfn(sinphi, cos(Q->phi2), P->es));
-            if (Q->n == 0) {
-                // Not quite, but es is very close to 1...
-                return pj_default_destructor(P, PJD_ERR_ECCENTRICITY_IS_ONE_OR_GREATER);
-            }
             Q->n /= log(ml1 / pj_tsfn(Q->phi2, sinphi, P->e));
         }
         Q->c = (Q->rho0 = m1 * pow(ml1, -Q->n) / Q->n);
-- 
2.30.2

