From 7b19b3b29a8bf8bf20b7f7ce1c3d5bd649e13985 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:38:13 +0900
Subject: [PATCH] buggy

---
 src/ell_set.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/ell_set.cpp b/src/ell_set.cpp
index 386b4f46..71746895 100644
--- a/src/ell_set.cpp
+++ b/src/ell_set.cpp
@@ -399,10 +399,7 @@ static int ellps_spherification (PJ *P) {
         break;
     }

-    if (P->a <= 0.) {
-        return proj_errno_set(P, PJD_ERR_MAJOR_AXIS_NOT_GIVEN);
-    }

     /* Clean up the ellipsoidal parameters to reflect the sphere */
     P->es = P->e = P->f = 0;
     P->rf = HUGE_VAL;
--
2.30.2

