From e94ebee0a067e6e68dee422634b830c2561741c8 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:39:04 +0900
Subject: [PATCH] buggy

---
 src/iso19111/factory.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/iso19111/factory.cpp b/src/iso19111/factory.cpp
index 4894085e..de8e1362 100644
--- a/src/iso19111/factory.cpp
+++ b/src/iso19111/factory.cpp
@@ -5642,7 +5642,7 @@ AuthorityFactory::createObjectsFromNameEx(
             sql += "AND deprecated = 1 ";
         }
         if (!approximateMatch) {
-            sql += "AND name = ? ";
+            sql += "AND name LIKE ? ";
             params.push_back(searchedNameWithoutDeprecated);
         }
         if (d->hasAuthorityRestriction()) {
@@ -5672,7 +5672,7 @@ AuthorityFactory::createObjectsFromNameEx(
             sql += "AND ov.deprecated = 1 ";
         }
         if (!approximateMatch) {
-            sql += "AND a.alt_name = ? ";
+            sql += "AND a.alt_name LIKE ? ";
             params.push_back(searchedNameWithoutDeprecated);
         }
         if (d->hasAuthorityRestriction()) {
-- 
2.30.2

