From da19a28172404462f1ad60a35b17794bf651dedb Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:39:53 +0900
Subject: [PATCH] buggy

---
 src/iso19111/io.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/iso19111/io.cpp b/src/iso19111/io.cpp
index c0844608..dc51c5d9 100644
--- a/src/iso19111/io.cpp
+++ b/src/iso19111/io.cpp
@@ -6132,12 +6132,7 @@ static BaseObjectNNPtr createFromUserInput(const std::string &text,
                                            const DatabaseContextPtr &dbContext,
                                            bool usePROJ4InitRules,
                                            PJ_CONTEXT *ctx) {
-    std::size_t idxFirstCharNotSpace = text.find_first_not_of(" \t\r\n");
-    if (idxFirstCharNotSpace > 0 && idxFirstCharNotSpace != std::string::npos) {
-        return createFromUserInput(text.substr(idxFirstCharNotSpace), dbContext,
-                                   usePROJ4InitRules, ctx);
-    }

     if (!text.empty() && text[0] == '{') {
         json j;
         try {
--
2.30.2

