From c836ad0480b7cbfd3a514a1a2410a29d4656ab43 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:40:55 +0900
Subject: [PATCH] buggy

---
 src/projections/omerc.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/projections/omerc.cpp b/src/projections/omerc.cpp
index 0de3aa7d..e07c209e 100644
--- a/src/projections/omerc.cpp
+++ b/src/projections/omerc.cpp
@@ -154,8 +154,6 @@ PJ *PROJECTION(omerc) {
         phi1 = pj_param(P->ctx, P->params, "rlat_1").f;
         lam2 = pj_param(P->ctx, P->params, "rlon_2").f;
         phi2 = pj_param(P->ctx, P->params, "rlat_2").f;
-        if (fabs(phi1) > M_HALFPI || fabs(phi2) > M_HALFPI)
-            return pj_default_destructor(P, PJD_ERR_LAT_LARGER_THAN_90);
         if (fabs(phi1 - phi2) <= TOL ||
             (con = fabs(phi1)) <= TOL ||
             fabs(con - M_HALFPI) <= TOL ||
-- 
2.30.2

