From 932ded80dfc154903bcd91eff93864b8240ec438 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:41:18 +0900
Subject: [PATCH] buggy

---
 src/projections/tmerc.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/projections/tmerc.cpp b/src/projections/tmerc.cpp
index bb56f8ae..c91c5174 100644
--- a/src/projections/tmerc.cpp
+++ b/src/projections/tmerc.cpp
@@ -188,10 +188,6 @@ static PJ_LP approx_s_inv (PJ_XY xy, PJ *P) {
     double h, g;
 
     h = exp(xy.x / static_cast<struct pj_opaque_approx*>(P->opaque)->esp);
-    if( h == 0 ) {
-        proj_errno_set(P, PJD_ERR_INVALID_X_OR_Y);
-        return proj_coord_error().lp;
-    }
     g = .5 * (h - 1. / h);
     h = cos (P->phi0 + xy.y / static_cast<struct pj_opaque_approx*>(P->opaque)->esp);
     lp.phi = asin(sqrt((1. - h * h) / (1. + g * g)));
-- 
2.30.2

