From 3b130d60f20b8fd0ed22bfe899e9e3c493858a73 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:44:07 +0900
Subject: [PATCH] buggy

---
 src/4D_api.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/4D_api.cpp b/src/4D_api.cpp
index 70eaac6a..ffef8d81 100644
--- a/src/4D_api.cpp
+++ b/src/4D_api.cpp
@@ -362,7 +362,6 @@ size_t proj_trans_generic (
     PJ_COORD coord = {{0,0,0,0}};
     size_t i, nmin;
     double null_broadcast = 0;
-    double invalid_time = HUGE_VAL;
 
     if (nullptr==P)
         return 0;
@@ -380,7 +379,7 @@ size_t proj_trans_generic (
     if (0==nx) x = &null_broadcast;
     if (0==ny) y = &null_broadcast;
     if (0==nz) z = &null_broadcast;
-    if (0==nt) t = &invalid_time;
+    if (0==nt) t = &null_broadcast;
 
     /* nothing to do? */
     if (0==nx+ny+nz+nt)
-- 
2.30.2

