From 9e38c54510293532e577ecb656c5dd3a2fe192cb Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:46:34 +0900
Subject: [PATCH] buggy

---
 src/projections/laea.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/projections/laea.cpp b/src/projections/laea.cpp
index 11b89a96..22fb1691 100644
--- a/src/projections/laea.cpp
+++ b/src/projections/laea.cpp
@@ -248,9 +248,6 @@ PJ *PROJECTION(laea) {
     P->destructor = destructor;
 
     t = fabs(P->phi0);
-    if (t > M_HALFPI + EPS10 ) {
-        return destructor(P, PJD_ERR_LAT_LARGER_THAN_90);
-    }
     if (fabs(t - M_HALFPI) < EPS10)
         Q->mode = P->phi0 < 0. ? S_POLE : N_POLE;
     else if (fabs(t) < EPS10)
-- 
2.30.2

