From 2f1d5ffca8a6cfbb3e9e73a33e5dabfd6b1a3ccd Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:48:39 +0900
Subject: [PATCH] buggy

---
 src/projections/aea.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/projections/aea.cpp b/src/projections/aea.cpp
index 8a80c49c..f457e836 100644
--- a/src/projections/aea.cpp
+++ b/src/projections/aea.cpp
@@ -180,10 +180,6 @@ static PJ *setup(PJ *P) {
                 return destructor(P, 0);
 
             Q->n = (m1 * m1 - m2 * m2) / (ml2 - ml1);
-            if (Q->n == 0) {
-                // Not quite, but es is very close to 1...
-                return destructor(P, PJD_ERR_ECCENTRICITY_IS_ONE_OR_GREATER);
-            }
         }
         Q->ec = 1. - .5 * P->one_es * log((1. - P->e) /
             (1. + P->e)) / P->e;
-- 
2.30.2

