From 2bf1e7ba0f21e7815964a06bc441cf67a6624899 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 16 Sep 2021 16:59:37 +0900
Subject: [PATCH] buggy

---
 src/projections/labrd.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/projections/labrd.cpp b/src/projections/labrd.cpp
index 85ab3ddd..330c105f 100644
--- a/src/projections/labrd.cpp
+++ b/src/projections/labrd.cpp
@@ -108,10 +108,7 @@ PJ *PROJECTION(labrd) {
         return pj_default_destructor (P, ENOMEM);
     P->opaque = Q;

-    if (P->phi0 == 0.) {
-        return pj_default_destructor(P, PJD_ERR_LAT_0_IS_ZERO);
-    }

     Az = pj_param(P->ctx, P->params, "razi").f;
     sinp = sin(P->phi0);
     t = 1. - P->es * sinp * sinp;
--
2.25.1

