From 9a3a8f2fee0e91ba389cd960b75c718a64af301a Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:53:25 +0900
Subject: [PATCH] buggy

---
 src/iso19111/operation/transformation.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/iso19111/operation/transformation.cpp b/src/iso19111/operation/transformation.cpp
index 6126d325..8b822d51 100644
--- a/src/iso19111/operation/transformation.cpp
+++ b/src/iso19111/operation/transformation.cpp
@@ -2806,14 +2806,14 @@ void Transformation::_exportToPROJString(
                                   common::UnitOfMeasure::ARC_SECOND);
 
         auto sourceCRSGeog =
-            extractGeographicCRSIfGeographicCRSOrEquivalent(sourceCRS());
+            dynamic_cast<const crs::GeographicCRS *>(sourceCRS().get());
         if (!sourceCRSGeog) {
             throw io::FormattingException(
                 "Can apply Geographic 2D offsets only to GeographicCRS");
         }
 
         auto targetCRSGeog =
-            extractGeographicCRSIfGeographicCRSOrEquivalent(targetCRS());
+            dynamic_cast<const crs::GeographicCRS *>(targetCRS().get());
         if (!targetCRSGeog) {
             throw io::FormattingException(
                 "Can apply Geographic 2D offsets only to GeographicCRS");
-- 
2.30.2

