From 8d105256be0e48dc3639dfcc5c33ff6608dfd365 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:55:23 +0900
Subject: [PATCH] buggy

---
 src/iso19111/io.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/iso19111/io.cpp b/src/iso19111/io.cpp
index 8bfb4395..6a1d7e32 100644
--- a/src/iso19111/io.cpp
+++ b/src/iso19111/io.cpp
@@ -3839,12 +3839,7 @@ WKTParser::Private::buildProjectedCRS(const WKTNodeNNPtr &node) {
         ThrowNotExpectedCSType("Cartesian");
     }

-    if (cartesianCS->axisList().size() == 3 &&
-        baseGeodCRS->coordinateSystem()->axisList().size() == 2) {
-        baseGeodCRS = NN_NO_CHECK(util::nn_dynamic_pointer_cast<GeodeticCRS>(
-            baseGeodCRS->promoteTo3D(std::string(), dbContext_)));
-    }

     addExtensionProj4ToProp(nodeP, props);

     return ProjectedCRS::create(props, baseGeodCRS, conversion,
--
2.30.2

