From a9d250bab1f899109ed9c5aaf02b0a28bfcc0352 Mon Sep 17 00:00:00 2001
From: hakumaku <gentlebuuny@gmail.com>
Date: Mon, 13 Sep 2021 16:57:05 +0900
Subject: [PATCH] buggy

---
 src/projections/stere.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/projections/stere.cpp b/src/projections/stere.cpp
index 3fff722d..abc4aa13 100644
--- a/src/projections/stere.cpp
+++ b/src/projections/stere.cpp
@@ -86,10 +86,7 @@ static PJ_XY stere_e_forward (PJ_LP lp, PJ *P) {          /* Ellipsoidal, forwar
         sinphi = -sinphi;
         /*-fallthrough*/
     case N_POLE:
-        if( fabs(lp.phi - M_HALFPI) < 1e-15 )
-            xy.x = 0;
-        else
-            xy.x = Q->akm1 * pj_tsfn (lp.phi, sinphi, P->e);
+        xy.x = Q->akm1 * pj_tsfn (lp.phi, sinphi, P->e);
         xy.y = - xy.x * coslam;
         break;
     }
-- 
2.30.2

