From f589fa1ab29de665c2e2357e5d85bf421b29b340 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 8 Jul 2021 15:12:01 +0900
Subject: [PATCH] buggy

---
 libyara/parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libyara/parser.c b/libyara/parser.c
index dd696f6d..6789a341 100644
--- a/libyara/parser.c
+++ b/libyara/parser.c
@@ -730,7 +730,7 @@ int yr_parser_reduce_rule_declaration_phase_1(
       yr_hash_table_lookup(
         compiler->objects_table,
         identifier,
-        NULL) != NULL)
+        compiler->current_namespace->name) != NULL)
   {
     // A rule or variable with the same identifier already exists, return the
     // appropriate error.
-- 
2.25.1

