From b14d3e362f850b9ff552c2d6e0a18053e9b61e53 Mon Sep 17 00:00:00 2001
From: crAy <crAyCoding@gmail.com>
Date: Wed, 31 Aug 2022 09:40:32 +0900
Subject: [PATCH] common

---
 tests/test-rules.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/tests/test-rules.c b/tests/test-rules.c
index 12934ccf..4dd6044c 100644
--- a/tests/test-rules.c
+++ b/tests/test-rules.c
@@ -2800,10 +2800,24 @@ void test_performance_warnings()
         condition: $a }")
 
 }
-
+static struct sigaction dpp_gcov_sigaction;
+static struct sigaction dpp_orig_sigaction;
+void dpp_sighandler(int signum) {
+#ifdef DPP_ENABLE_GCOV
+  __gcov_flush();
+#endif
+  sigaction(SIGABRT, &dpp_orig_sigaction, NULL);
+  raise(SIGABRT);
+}
 
 int main(int argc, char** argv)
 {
+  {
+  dpp_gcov_sigaction.sa_handler = dpp_sighandler;
+  sigemptyset(&dpp_gcov_sigaction.sa_mask);
+  dpp_gcov_sigaction.sa_flags = 0;
+  sigaction(SIGABRT, &dpp_gcov_sigaction, &dpp_orig_sigaction);
+  }
   char *top_srcdir = getenv("TOP_SRCDIR");
   if (top_srcdir)
     chdir(top_srcdir);
-- 
2.25.1

