From 99f08d870e636bf3662abeda1a8f41fde0aa3aee Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 11:51:15 +0900
Subject: [PATCH] buggy

---
 epan/ftypes/ftype-integer.c | 23 ++---------------------
 1 file changed, 2 insertions(+), 21 deletions(-)

diff --git a/epan/ftypes/ftype-integer.c b/epan/ftypes/ftype-integer.c
index dc604f1eb2..206913c648 100644
--- a/epan/ftypes/ftype-integer.c
+++ b/epan/ftypes/ftype-integer.c
@@ -420,32 +420,13 @@ char_repr_len(fvalue_t *fv _U_, ftrepr_t rtype _U_, int field_display _U_)
 static void
 uinteger_to_repr(fvalue_t *fv, ftrepr_t rtype _U_, int field_display, char *buf, unsigned int size)
 {
-	if (((field_display & 0xff) == BASE_HEX) || ((field_display & 0xff) == BASE_HEX_DEC))
+	if ((field_display == BASE_HEX) || (field_display == BASE_HEX_DEC))
 	{
 		/* This format perfectly fits into 11 bytes. */
 		*buf++ = '0';
 		*buf++ = 'x';
 
-		switch (fv->ftype->ftype) {
-
-		case FT_UINT8:
-			buf = guint8_to_hex(buf, fv->value.uinteger);
-			break;
-
-		case FT_UINT16:
-			buf = word_to_hex(buf, fv->value.uinteger);
-			break;
-
-		case FT_UINT24:
-			buf = guint8_to_hex(buf, (fv->value.uinteger & 0x00ff0000) >> 16);
-			buf = word_to_hex(buf, (fv->value.uinteger & 0x0000ffff));
-			break;
-
-		default:
-			buf = dword_to_hex(buf, fv->value.uinteger);
-			break;
-		}
-
+		buf = dword_to_hex(buf, fv->value.uinteger);
 		*buf++ = '\0';
 	}
 	else
-- 
2.25.1

