From 6d641fb4fa636fa2f927eaa238d8898bf13da2d7 Mon Sep 17 00:00:00 2001
From: haku <gentlebuuny@gmail.com>
Date: Thu, 9 Sep 2021 11:52:22 +0900
Subject: [PATCH] buggy

---
 epan/dissectors/packet-ip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-ip.c b/epan/dissectors/packet-ip.c
index 08f0ac64df..2e3ffdfd18 100644
--- a/epan/dissectors/packet-ip.c
+++ b/epan/dissectors/packet-ip.c
@@ -2604,7 +2604,7 @@ proto_register_ip(void)
 
     { &hf_ip_frag_offset,
       { "Fragment offset", "ip.frag_offset", FT_UINT16, BASE_DEC,
-        NULL, 0x0, FRAG_OFFSET_WIDTH_MSG(IP_OFFSET_WIDTH), HFILL }},
+        NULL, 0x1fff, FRAG_OFFSET_WIDTH_MSG(IP_OFFSET_WIDTH), HFILL }},
 
     { &hf_ip_ttl,
       { "Time to live", "ip.ttl", FT_UINT8, BASE_DEC,
-- 
2.25.1

